--- a/net/minecraft/network/syncher/EntityDataSerializer.java
+++ b/net/minecraft/network/syncher/EntityDataSerializer.java
@@ -18,11 +18,11 @@
    static <T> EntityDataSerializer<T> m_238095_(final FriendlyByteBuf.Writer<T> p_238096_, final FriendlyByteBuf.Reader<T> p_238097_) {
       return new EntityDataSerializer.ForValueType<T>() {
          public void m_6856_(FriendlyByteBuf p_238109_, T p_238110_) {
-            p_238096_.accept((T)p_238109_, p_238110_);
+            p_238096_.accept(p_238109_, p_238110_);
          }
 
          public T m_6709_(FriendlyByteBuf p_238107_) {
-            return p_238097_.apply((T)p_238107_);
+            return p_238097_.apply(p_238107_);
          }
       };
    }
