--- a/net/minecraft/world/level/block/ButtonBlock.java
+++ b/net/minecraft/world/level/block/ButtonBlock.java
@@ -166,7 +166,7 @@
          p_51122_.m_7731_(p_51123_, p_51121_.m_61124_(f_51045_, Boolean.valueOf(flag)), 3);
          this.m_51124_(p_51121_, p_51122_, p_51123_);
          this.m_51067_((Player)null, p_51122_, p_51123_, flag);
-         p_51122_.m_142346_(list.stream().findFirst().orElse((Entity)null), flag ? GameEvent.f_215589_ : GameEvent.f_214625_, p_51123_);
+         p_51122_.m_142346_(list.stream().findFirst().orElse(null), flag ? GameEvent.f_215589_ : GameEvent.f_214625_, p_51123_);
       }
 
       if (flag) {
