--- a/net/minecraft/client/gui/GuiOverlayDebug.java
+++ b/net/minecraft/client/gui/GuiOverlayDebug.java
@@ -88,6 +88,7 @@
 
    }
 
+   @SuppressWarnings("incomplete-switch")
    protected List<String> call() {
       BlockPos blockpos = new BlockPos(this.field_175242_a.func_175606_aa().field_70165_t, this.field_175242_a.func_175606_aa().func_174813_aQ().field_72338_b, this.field_175242_a.func_175606_aa().field_70161_v);
       if (this.field_175242_a.func_189648_am()) {
@@ -171,7 +172,7 @@
             for(UnmodifiableIterator unmodifiableiterator = iblockstate.func_177228_b().entrySet().iterator(); unmodifiableiterator.hasNext(); list.add(iproperty.func_177701_a() + ": " + s)) {
                Entry<IProperty<?>, Comparable<?>> entry = (Entry)unmodifiableiterator.next();
                iproperty = (IProperty)entry.getKey();
-               T t = entry.getValue();
+               T t = (T)entry.getValue();
                s = iproperty.func_177702_a(t);
                if (Boolean.TRUE.equals(t)) {
                   s = TextFormatting.GREEN + s;
