--- a/net/minecraft/nbt/TagParser.java
+++ b/net/minecraft/nbt/TagParser.java
@@ -251,11 +251,11 @@
             }
 
             if (p_129363_ == ByteTag.f_128255_) {
-               list.add((T)((NumericTag)tag).m_7063_());
+               list.add((T)(Byte)((NumericTag)tag).m_7063_());
             } else if (p_129363_ == LongTag.f_128873_) {
-               list.add((T)((NumericTag)tag).m_7046_());
+               list.add((T)(Long)((NumericTag)tag).m_7046_());
             } else {
-               list.add((T)((NumericTag)tag).m_7047_());
+               list.add((T)(Integer)((NumericTag)tag).m_7047_());
             }
 
             if (this.m_129376_()) {
