--- a/net/minecraft/client/multiplayer/ClientPacketListener.java
+++ b/net/minecraft/client/multiplayer/ClientPacketListener.java
@@ -776,7 +776,7 @@
     @Override
     public void handleRemoveEntities(final ClientboundRemoveEntitiesPacket packet) {
         PacketUtils.ensureRunningOnSameThread(packet, this, this.minecraft.packetProcessor());
-        packet.getEntityIds().forEach(entityId -> {
+        packet.getEntityIds().forEach((int entityId) -> {
             Entity entity = this.level.getEntity(entityId);
             if (entity != null) {
                 if (entity.hasIndirectPassenger(this.minecraft.player)) {
@@ -2095,7 +2095,7 @@
     public void handleGameRuleValues(final ClientboundGameRuleValuesPacket packet) {
         PacketUtils.ensureRunningOnSameThread(packet, this, this.minecraft.packetProcessor());
         if (this.minecraft.screen instanceof InWorldGameRulesScreen inWorldGameRulesScreen) {
-            inWorldGameRulesScreen.onGameRuleValuesUpdated(packet.values());
+            inWorldGameRulesScreen.onGameRuleValuesUpdated((Map)packet.values());
         }
     }
 
