--- a/net/minecraft/util/ClassInheritanceMultiMap.java
+++ b/net/minecraft/util/ClassInheritanceMultiMap.java
@@ -59,7 +59,7 @@
          List<T> list = this.field_180218_a.computeIfAbsent(p_219790_1_, (p_219791_1_) -> {
             return this.field_181745_e.stream().filter(p_219791_1_::isInstance).collect(Collectors.toList());
          });
-         return Collections.unmodifiableCollection(list);
+         return (Collection<S>) Collections.unmodifiableCollection(list);
       }
    }
 
