--- a/net/minecraft/data/registries/RegistriesDatapackGenerator.java
+++ b/net/minecraft/data/registries/RegistriesDatapackGenerator.java
@@ -50,7 +50,7 @@
                     PackOutput.PathProvider pathProvider = this.output.createRegistryElementsPathProvider(registryKey);
                     return CompletableFuture.allOf(
                         registry.listElements()
-                            .map(e -> dumpValue(pathProvider.json(e.key().identifier()), cache, writeOps, v.elementCodec(), e.value()))
+                            .<CompletableFuture<?>>map(e -> dumpValue(pathProvider.json(e.key().identifier()), cache, writeOps, v.elementCodec(), e.value()))
                             .toArray(CompletableFuture[]::new)
                     );
                 }
