--- a/net/minecraft/client/shader/ShaderGroup.java
+++ b/net/minecraft/client/shader/ShaderGroup.java
@@ -40,7 +40,7 @@
    private float field_148036_j;
    private float field_148037_k;
 
-   public ShaderGroup(TextureManager p_i1050_1_, IResourceManager p_i1050_2_, Framebuffer p_i1050_3_, ResourceLocation p_i1050_4_) throws JsonSyntaxException, IOException, JsonException {
+   public ShaderGroup(TextureManager p_i1050_1_, IResourceManager p_i1050_2_, Framebuffer p_i1050_3_, ResourceLocation p_i1050_4_) throws JsonException, IOException, JsonSyntaxException {
       this.field_148033_b = p_i1050_2_;
       this.field_148035_a = p_i1050_3_;
       this.field_148036_j = 0.0F;
@@ -52,7 +52,7 @@
       this.func_152765_a(p_i1050_1_, p_i1050_4_);
    }
 
-   public void func_152765_a(TextureManager p_152765_1_, ResourceLocation p_152765_2_) throws JsonSyntaxException, IOException, JsonException {
+   public void func_152765_a(TextureManager p_152765_1_, ResourceLocation p_152765_2_) throws JsonException, IOException, JsonSyntaxException {
       JsonParser jsonparser = new JsonParser();
       IResource iresource = null;
 
@@ -119,7 +119,7 @@
 
    }
 
-   private void func_152764_a(TextureManager p_152764_1_, JsonElement p_152764_2_) throws IOException, JsonException {
+   private void func_152764_a(TextureManager p_152764_1_, JsonElement p_152764_2_) throws JsonException, IOException {
       JsonObject jsonobject = JsonUtils.func_151210_l(p_152764_2_, "pass");
       String s = JsonUtils.func_151200_h(jsonobject, "name");
       String s1 = JsonUtils.func_151200_h(jsonobject, "intarget");
@@ -269,7 +269,7 @@
       this.field_148031_d.clear();
    }
 
-   public Shader func_148023_a(String p_148023_1_, Framebuffer p_148023_2_, Framebuffer p_148023_3_) throws IOException, JsonException {
+   public Shader func_148023_a(String p_148023_1_, Framebuffer p_148023_2_, Framebuffer p_148023_3_) throws JsonException, IOException {
       Shader shader = new Shader(this.field_148033_b, p_148023_1_, p_148023_2_, p_148023_3_);
       this.field_148031_d.add(this.field_148031_d.size(), shader);
       return shader;
