--- a/net/minecraft/world/entity/SpawnPlacements.java
+++ b/net/minecraft/world/entity/SpawnPlacements.java
@@ -61,7 +61,7 @@
 
    public static <T extends Entity> boolean m_229494_(EntityType<T> p_229495_, ServerLevelAccessor p_229496_, MobSpawnType p_229497_, BlockPos p_229498_, RandomSource p_229499_) {
       SpawnPlacements.Data spawnplacements$data = f_21750_.get(p_229495_);
-      return spawnplacements$data == null || spawnplacements$data.f_21769_.m_21780_(p_229495_, p_229496_, p_229497_, p_229498_, p_229499_);
+      return spawnplacements$data == null || spawnplacements$data.f_21769_.m_21780_((EntityType)p_229495_, p_229496_, p_229497_, p_229498_, p_229499_);
    }
 
    static {
