--- a/net/minecraft/util/datafix/fixes/FixProjectileStoredItem.java
+++ b/net/minecraft/util/datafix/fixes/FixProjectileStoredItem.java
@@ -46,7 +46,7 @@
         final String entityName, final FixProjectileStoredItem.SubFixer<?> fixer, final Type<?> inputEntityChoiceType, final Type<T> outputEntityChoiceType
     ) {
         OpticFinder<?> opticfinder = DSL.namedChoice(entityName, inputEntityChoiceType);
-        return input -> input.updateTyped(opticfinder, outputEntityChoiceType, typed -> fixer.fix(typed, outputEntityChoiceType));
+        return input -> input.updateTyped(opticfinder, outputEntityChoiceType, typed -> fixer.fix((Typed)typed, (Type)outputEntityChoiceType));
     }
 
     private static <T> Typed<T> fixArrow(final Typed<?> typed, final Type<T> outputType) {
