--- a/net/minecraft/util/datafix/fixes/ItemShulkerBoxColorFix.java
+++ b/net/minecraft/util/datafix/fixes/ItemShulkerBoxColorFix.java
@@ -58,7 +58,7 @@
                             Dynamic<?> dynamic = typed1.get(DSL.remainderFinder());
                             int i = dynamic.get("Color").asInt(0);
                             dynamic.remove("Color");
-                            return input.set(opticfinder1, typed.set(opticfinder2, typed1.set(DSL.remainderFinder(), dynamic)))
+                            return input.set(opticfinder1, (Typed<?>)typed.set(opticfinder2, typed1.set(DSL.remainderFinder(), dynamic)))
                                 .set(opticfinder, Pair.of(References.ITEM_NAME.typeName(), NAMES_BY_COLOR[i % 16]));
                         }
                     }
