--- a/net/minecraft/util/datafix/fixes/TextComponentHoverAndClickEventFix.java
+++ b/net/minecraft/util/datafix/fixes/TextComponentHoverAndClickEventFix.java
@@ -101,8 +101,8 @@
                     : copyFields(dynamic.remove("contents"), dynamic1, "id", "count", "components");
             }
             case "show_entity" -> {
-                Dynamic<?> dynamic = dynamic.get("contents").orElseEmptyMap();
-                yield copyFields(dynamic.remove("contents"), dynamic, "id", "type", "name").renameField("id", "uuid").renameField("type", "id");
+                Dynamic<?> contents = dynamic.get("contents").orElseEmptyMap();
+                yield copyFields(dynamic.remove("contents"), contents, "id", "type", "name").renameField("id", "uuid").renameField("type", "id");
             }
             default -> dynamic;
         };
