--- a/net/minecraft/util/datafix/fixes/ChunkBedBlockEntityInjecterFix.java
+++ b/net/minecraft/util/datafix/fixes/ChunkBedBlockEntityInjecterFix.java
@@ -40,8 +40,8 @@
         return TypeRewriteRule.seq(
             this.fixTypeEverywhere(
                 "InjectBedBlockEntityType",
-                this.getInputSchema().findChoiceType(References.BLOCK_ENTITY),
-                this.getOutputSchema().findChoiceType(References.BLOCK_ENTITY),
+                (com.mojang.datafixers.types.templates.TaggedChoice.TaggedChoiceType<String>)this.getInputSchema().findChoiceType(References.BLOCK_ENTITY),
+                (com.mojang.datafixers.types.templates.TaggedChoice.TaggedChoiceType<String>)this.getOutputSchema().findChoiceType(References.BLOCK_ENTITY),
                 ops -> v -> v
             ),
             this.fixTypeEverywhereTyped(
