--- a/net/minecraft/client/gui/recipebook/GuiRecipeBook.java
+++ b/net/minecraft/client/gui/recipebook/GuiRecipeBook.java
@@ -147,7 +147,7 @@
       if (!s.isEmpty()) {
          ObjectSet<RecipeList> objectset = new ObjectLinkedOpenHashSet<RecipeList>(this.field_191888_F.func_193987_a(SearchTreeManager.field_194012_b).func_194038_a(s.toLowerCase(Locale.ROOT)));
          list1.removeIf((p_193947_1_) -> {
-            return !p_193947_0_.contains(p_193947_1_);
+            return !objectset.contains(p_193947_1_);
          });
       }
 
