--- a/net/minecraft/item/IItemPropertyGetter.java
+++ b/net/minecraft/item/IItemPropertyGetter.java
@@ -1,4 +1,9 @@
 package net.minecraft.item;
 
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.world.World;
+import javax.annotation.Nullable;
+
 public interface IItemPropertyGetter {
+    float call(ItemStack p_call_1_, @Nullable World p_call_2_, @Nullable LivingEntity p_call_3_); // MCP: add client-only method to server
 }
