--- a/net/minecraft/core/HolderLookup.java
+++ b/net/minecraft/core/HolderLookup.java
@@ -89,7 +89,7 @@
          }));
          return new HolderLookup.Provider() {
             public <T> Optional<HolderLookup.RegistryLookup<T>> m_254861_(ResourceKey<? extends Registry<? extends T>> p_255663_) {
-               return Optional.ofNullable(map.get(p_255663_));
+               return Optional.ofNullable((HolderLookup.RegistryLookup<T>) map.get(p_255663_));
             }
          };
       }
