--- a/net/minecraft/server/rcon/thread/RconClient.java
+++ b/net/minecraft/server/rcon/thread/RconClient.java
@@ -39,8 +39,9 @@
    }
 
    public void run() {
-      while(true) {
+      {
          try {
+         while(true) {
             if (!this.f_11515_) {
                return;
             }
@@ -91,6 +92,8 @@
                      continue;
                }
             }
+            return;
+         }
          } catch (IOException ioexception) {
             return;
          } catch (Exception exception1) {
@@ -101,8 +104,6 @@
             f_11579_.info("Thread {} shutting down", (Object)this.f_11516_);
             this.f_11515_ = false;
          }
-
-         return;
       }
    }
 
