--- a/net/minecraft/client/OptionInstance.java
+++ b/net/minecraft/client/OptionInstance.java
@@ -107,7 +107,7 @@
    }
 
    public AbstractWidget m_231507_(Options p_231508_, int p_231509_, int p_231510_, int p_231511_) {
-      OptionInstance.TooltipSupplier<T> tooltipsupplier = this.f_231474_.apply((T)Minecraft.m_91087_());
+      OptionInstance.TooltipSupplier<T> tooltipsupplier = this.f_231474_.apply(Minecraft.m_91087_());
       return this.f_231476_.m_213823_(tooltipsupplier, p_231508_, p_231509_, p_231510_, p_231511_).apply(this);
    }
 
@@ -264,7 +264,7 @@
       default <R> OptionInstance.SliderableValueSet<R> m_231657_(final IntFunction<? extends R> p_231658_, final ToIntFunction<? super R> p_231659_) {
          return new OptionInstance.SliderableValueSet<R>() {
             public Optional<R> m_214064_(R p_231674_) {
-               return IntRangeBase.this.m_214064_((T)Integer.valueOf(p_231659_.applyAsInt(p_231674_))).map(p_231658_::apply);
+               return IntRangeBase.this.m_214064_(Integer.valueOf(p_231659_.applyAsInt(p_231674_))).map(p_231658_::apply);
             }
 
             public double m_213640_(R p_231678_) {
