--- a/net/minecraft/client/gui/font/FontSet.java
+++ b/net/minecraft/client/gui/font/FontSet.java
@@ -52,7 +52,7 @@
       }
 
       Set<GlyphProvider> set = Sets.newHashSet();
-      intset.forEach((p_216544_) -> {
+      intset.forEach((int p_216544_) -> {
          for(GlyphProvider glyphprovider1 : p_95072_) {
             GlyphInfo glyphinfo = glyphprovider1.m_7823_(p_216544_);
             if (glyphinfo != null) {
