--- a/net/minecraft/world/inventory/AbstractFurnaceMenu.java
+++ b/net/minecraft/world/inventory/AbstractFurnaceMenu.java
@@ -145,7 +145,7 @@
    }
 
    protected boolean m_38977_(ItemStack p_38978_) {
-      return this.f_38954_.m_7465_().m_44015_(this.f_38957_, new SimpleContainer(p_38978_), this.f_38954_).isPresent();
+      return this.f_38954_.m_7465_().m_44015_((RecipeType<AbstractCookingRecipe>)this.f_38957_, new SimpleContainer(p_38978_), this.f_38954_).isPresent();
    }
 
    protected boolean m_38988_(ItemStack p_38989_) {
