--- a/net/minecraft/util/valueproviders/ClampedNormalInt.java
+++ b/net/minecraft/util/valueproviders/ClampedNormalInt.java
@@ -8,7 +8,7 @@
 import net.minecraft.util.Mth;
 
 public class ClampedNormalInt extends IntProvider {
-   public static final Codec<ClampedNormalInt> f_185867_ = RecordCodecBuilder.create((p_185887_) -> {
+   public static final Codec<ClampedNormalInt> f_185867_ = RecordCodecBuilder.<ClampedNormalInt>create((p_185887_) -> {
       return p_185887_.group(Codec.FLOAT.fieldOf("mean").forGetter((p_185905_) -> {
          return p_185905_.f_185868_;
       }), Codec.FLOAT.fieldOf("deviation").forGetter((p_185903_) -> {
