--- a/net/minecraft/util/datafix/fixes/ItemStackEnchantmentFix.java
+++ b/net/minecraft/util/datafix/fixes/ItemStackEnchantmentFix.java
@@ -68,7 +68,7 @@
    }
 
    private Dynamic<?> func_209627_a(Dynamic<?> p_209627_1_) {
-      Optional optional1 = p_209627_1_.get("ench").flatMap(Dynamic::getStream).map((p_209626_0_) -> {
+      Optional<java.util.stream.Stream<Dynamic<?>>> optional1 = p_209627_1_.get("ench").flatMap(Dynamic::getStream).map((p_209626_0_) -> {
          return p_209626_0_.map((p_209628_0_) -> {
             return p_209628_0_.set("id", p_209628_0_.createString(field_208047_a.getOrDefault(Integer.valueOf(p_209628_0_.getInt("id")), "null")));
          });
@@ -80,7 +80,7 @@
       }
 
       return p_209627_1_.update("StoredEnchantments", (p_209625_0_) -> {
-         Optional optional2 = p_209625_0_.getStream().map((p_209623_0_) -> {
+         Optional<java.util.stream.Stream<Dynamic<?>>>  optional2 = p_209625_0_.getStream().map((p_209623_0_) -> {
             return p_209623_0_.map((p_209624_0_) -> {
                return p_209624_0_.set("id", p_209624_0_.createString(field_208047_a.getOrDefault(Integer.valueOf(p_209624_0_.getInt("id")), "null")));
             });
