--- a/net/minecraft/util/datafix/fixes/JukeboxRecordItem.java
+++ b/net/minecraft/util/datafix/fixes/JukeboxRecordItem.java
@@ -28,7 +28,7 @@
             Dynamic<?> dynamic1 = dynamic.emptyMap();
             dynamic1 = dynamic1.set("id", dynamic1.createString(s));
             dynamic1 = dynamic1.set("Count", dynamic1.createByte((byte)1));
-            return p_207419_1_.set(opticfinder, (Typed)((Optional)type1.readTyped(dynamic1).getSecond()).orElseThrow(() -> {
+            return p_207419_1_.set(opticfinder, (Typed)(type1.readTyped(dynamic1).getSecond()).orElseThrow(() -> {
                return new IllegalStateException("Could not create record item stack.");
             })).set(DSL.remainderFinder(), dynamic);
          }
