--- a/net/minecraft/util/datafix/fixes/TeamDisplayName.java
+++ b/net/minecraft/util/datafix/fixes/TeamDisplayName.java
@@ -29,7 +29,7 @@
             return (p_211877_0_) -> {
                return p_211877_0_.mapSecond((p_211875_0_) -> {
                   return p_211875_0_.update("DisplayName", (p_211878_1_) -> {
-                     Optional optional = p_211878_1_.getStringValue().map((p_211879_0_) -> {
+                     Optional<String> optional = p_211878_1_.getStringValue().map((p_211879_0_) -> {
                         return ITextComponent.Serializer.func_150696_a(new TextComponentString(p_211879_0_));
                      });
                      p_211875_0_.getClass();
