--- a/net/minecraft/world/entity/monster/hoglin/HoglinAi.java
+++ b/net/minecraft/world/entity/monster/hoglin/HoglinAi.java
@@ -78,7 +78,7 @@
    }
 
    private static void m_34615_(Brain<Hoglin> p_34616_) {
-      p_34616_.m_21895_(Activity.f_37991_, 10, ImmutableList.of(SetWalkTargetAwayFrom.m_257370_(MemoryModuleType.f_26383_, 1.3F, 15, false), m_34571_(), SetEntityLookTargetSometimes.m_257458_(8.0F, UniformInt.m_146622_(30, 60)), EraseMemoryIf.<PathfinderMob>m_258093_(HoglinAi::m_34617_, MemoryModuleType.f_26383_)), MemoryModuleType.f_26383_);
+      p_34616_.m_21895_(Activity.f_37991_, 10, ImmutableList.of(SetWalkTargetAwayFrom.m_257370_(MemoryModuleType.f_26383_, 1.3F, 15, false), m_34571_(), SetEntityLookTargetSometimes.m_257458_(8.0F, UniformInt.m_146622_(30, 60)), EraseMemoryIf.m_258093_(HoglinAi::m_34617_, MemoryModuleType.f_26383_)), MemoryModuleType.f_26383_);
    }
 
    private static RunOne<Hoglin> m_34571_() {
