--- a/net/minecraft/tags/TagKey.java
+++ b/net/minecraft/tags/TagKey.java
@@ -31,7 +31,7 @@
    }
 
    public static <T> TagKey<T> m_203882_(ResourceKey<? extends Registry<T>> p_203883_, ResourceLocation p_203884_) {
-      return f_203869_.intern(new TagKey<>(p_203883_, p_203884_));
+      return (TagKey<T>)f_203869_.intern(new TagKey<>(p_203883_, p_203884_));
    }
 
    public boolean m_207645_(ResourceKey<? extends Registry<?>> p_207646_) {
@@ -39,7 +39,7 @@
    }
 
    public <E> Optional<TagKey<E>> m_207647_(ResourceKey<? extends Registry<E>> p_207648_) {
-      return this.m_207645_(p_207648_) ? Optional.of(this) : Optional.empty();
+      return this.m_207645_(p_207648_) ? Optional.of((TagKey<E>)this) : Optional.empty();
    }
 
    public String toString() {
