--- a/net/minecraft/world/level/levelgen/SurfaceRules.java
+++ b/net/minecraft/world/level/levelgen/SurfaceRules.java
@@ -158,7 +158,7 @@
       }
    }
 
-   static record BlockRuleSource(BlockState f_189512_, SurfaceRules.StateRule f_189513_) implements SurfaceRules.RuleSource {
+   private static record BlockRuleSource(BlockState f_189512_, SurfaceRules.StateRule f_189513_) implements SurfaceRules.RuleSource {
       static final Codec<SurfaceRules.BlockRuleSource> f_189514_ = BlockState.f_61039_.xmap(SurfaceRules.BlockRuleSource::new, SurfaceRules.BlockRuleSource::f_189512_).fieldOf("result_state").codec();
 
       BlockRuleSource(BlockState p_189517_) {
