--- a/net/minecraft/util/datafix/fixes/ChunkProtoTickListFix.java
+++ b/net/minecraft/util/datafix/fixes/ChunkProtoTickListFix.java
@@ -179,7 +179,7 @@
         int k = p_185050_ >>> 8 & 15;
         String s = p_185051_.apply(p_185046_ != null ? p_185046_.get().m_185090_(i, j, k) : null);
         return p_185045_.createMap(
-            ImmutableMap.builder()
+            ImmutableMap.<Dynamic<?>, Dynamic<?>>builder()
                 .put(p_185045_.createString("i"), p_185045_.createString(s))
                 .put(p_185045_.createString("x"), p_185045_.createInt(p_185047_ * 16 + i))
                 .put(p_185045_.createString("y"), p_185045_.createInt(p_185048_ * 16 + j))
