--- a/net/minecraft/resources/RegistryFileCodec.java
+++ b/net/minecraft/resources/RegistryFileCodec.java
@@ -75,7 +75,7 @@
                      return "Failed to get element " + resourcekey;
                   });
                }).map((p_255658_) -> {
-                  return Pair.of(p_255658_, pair.getSecond());
+                  return Pair.of((Holder<E>)p_255658_, pair.getSecond());
                }).setLifecycle(Lifecycle.stable());
             }
          }
