--- a/net/minecraft/world/World.java
+++ b/net/minecraft/world/World.java
@@ -1568,7 +1568,7 @@
    }
 
    public Stream<VoxelShape> func_212392_a(@Nullable Entity p_212392_1_, VoxelShape p_212392_2_, VoxelShape p_212392_3_, Set<Entity> p_212392_4_) {
-      Stream<VoxelShape> stream = super.func_212392_a(p_212392_1_, p_212392_2_, p_212392_3_, p_212392_4_);
+      Stream<VoxelShape> stream = IWorldReader.super.func_212392_a(p_212392_1_, p_212392_2_, p_212392_3_, p_212392_4_);
       return p_212392_1_ == null ? stream : Stream.concat(stream, this.func_211155_a(p_212392_1_, p_212392_2_, p_212392_4_));
    }
 
