--- a/net/minecraft/command/impl/ForceLoadCommand.java
+++ b/net/minecraft/command/impl/ForceLoadCommand.java
@@ -83,7 +83,7 @@
       ServerWorld serverworld = p_212722_0_.func_197023_e();
       RegistryKey<World> registrykey = serverworld.func_234923_W_();
       LongSet longset = serverworld.func_217469_z();
-      longset.forEach((p_212720_1_) -> {
+      longset.forEach((long p_212720_1_) -> {
          serverworld.func_217458_b(ChunkPos.func_212578_a(p_212720_1_), ChunkPos.func_212579_b(p_212720_1_), false);
       });
       p_212722_0_.func_197030_a(new TranslationTextComponent("commands.forceload.removed.all", registrykey.func_240901_a_()), true);
