--- a/net/minecraft/world/level/chunk/PalettedContainer.java
+++ b/net/minecraft/world/level/chunk/PalettedContainer.java
@@ -50,7 +50,7 @@
    }
 
    public static <T> Codec<PalettedContainer<T>> m_188054_(IdMap<T> p_188055_, Codec<T> p_188056_, PalettedContainer.Strategy p_188057_, T p_188058_) {
-      return RecordCodecBuilder.create((p_188047_) -> {
+      return RecordCodecBuilder.<PalettedContainer.DiscData<T>>create((p_188047_) -> {
          return p_188047_.group(p_188056_.mapResult(ExtraCodecs.m_184381_(p_188058_)).listOf().fieldOf("palette").forGetter(PalettedContainer.DiscData::f_188123_), Codec.LONG_STREAM.optionalFieldOf("data").forGetter(PalettedContainer.DiscData::f_188124_)).apply(p_188047_, PalettedContainer.DiscData::new);
       }).comapFlatMap((p_188081_) -> {
          return m_188067_(p_188055_, p_188057_, p_188081_);
@@ -276,7 +276,7 @@
       void m_63144_(T p_63145_, int p_63146_);
    }
 
-   static record Data(PalettedContainer.Configuration<T> f_188100_, BitStorage f_188101_, Palette<T> f_188102_) {
+   static record Data<T>(PalettedContainer.Configuration<T> f_188100_, BitStorage f_188101_, Palette<T> f_188102_) {
       Data(PalettedContainer.Configuration<T> f_188100_, BitStorage f_188101_, Palette<T> f_188102_) {
          this.f_188100_ = f_188100_;
          this.f_188101_ = f_188101_;
@@ -305,7 +305,7 @@
       }
    }
 
-   static record DiscData(List<T> f_188123_, Optional<LongStream> f_188124_) {
+   static record DiscData<T>(List<T> f_188123_, Optional<LongStream> f_188124_) {
       DiscData(List<T> f_188123_, Optional<LongStream> f_188124_) {
          this.f_188123_ = f_188123_;
          this.f_188124_ = f_188124_;
