--- a/net/minecraft/network/protocol/configuration/ClientboundUpdateEnabledFeaturesPacket.java
+++ b/net/minecraft/network/protocol/configuration/ClientboundUpdateEnabledFeaturesPacket.java
@@ -8,7 +8,7 @@
 
 public record ClientboundUpdateEnabledFeaturesPacket(Set<ResourceLocation> f_290345_) implements Packet<ClientConfigurationPacketListener> {
    public ClientboundUpdateEnabledFeaturesPacket(FriendlyByteBuf p_299340_) {
-      this(p_299340_.m_236838_(HashSet::new, FriendlyByteBuf::m_130281_));
+      this(p_299340_.<ResourceLocation, Set<ResourceLocation>>m_236838_(HashSet::new, FriendlyByteBuf::m_130281_));
    }
 
    public void m_5779_(FriendlyByteBuf p_297257_) {
