--- a/net/minecraft/client/gui/font/FontManager.java
+++ b/net/minecraft/client/gui/font/FontManager.java
@@ -95,7 +95,7 @@
                intset.addAll(glyphprovider1.m_6990_());
             }
 
-            intset.forEach((p_95034_) -> {
+            intset.forEach((int p_95034_) -> {
                if (p_95034_ != 32) {
                   for(GlyphProvider glyphprovider2 : Lists.reverse(list)) {
                      if (glyphprovider2.m_214022_(p_95034_) != null) {
