--- a/net/minecraft/world/gen/feature/LiquidsConfig.java
+++ b/net/minecraft/world/gen/feature/LiquidsConfig.java
@@ -19,7 +19,7 @@
          return p_236653_0_.field_227364_c_;
       }), Codec.INT.fieldOf("hole_count").orElse(1).forGetter((p_236652_0_) -> {
          return p_236652_0_.field_227365_d_;
-      }), Registry.field_212618_g.listOf().fieldOf("valid_blocks").xmap(ImmutableSet::copyOf, ImmutableList::copyOf).forGetter((p_236651_0_) -> {
+      }), Registry.field_212618_g.listOf().fieldOf("valid_blocks").<Set<Block>>xmap(ImmutableSet::copyOf, ImmutableList::copyOf).forGetter((p_236651_0_) -> {
          return p_236651_0_.field_227366_f_;
       })).apply(p_236650_0_, LiquidsConfig::new);
    });
