--- a/net/minecraft/client/gui/screens/PresetFlatWorldScreen.java
+++ b/net/minecraft/client/gui/screens/PresetFlatWorldScreen.java
@@ -270,7 +270,7 @@
 
          public Entry(Holder<FlatLevelGeneratorPreset> p_232758_) {
             this.f_169357_ = p_232758_.m_203334_();
-            this.f_232755_ = p_232758_.m_203543_().map((p_232760_) -> {
+            this.f_232755_ = p_232758_.m_203543_().<Component>map((p_232760_) -> {
                return Component.m_237115_(p_232760_.m_135782_().m_214296_("flat_world_preset"));
             }).orElse(PresetFlatWorldScreen.f_232751_);
          }
