--- a/net/minecraft/client/renderer/entity/EntityRenderDispatcher.java
+++ b/net/minecraft/client/renderer/entity/EntityRenderDispatcher.java
@@ -91,9 +91,9 @@
     public <T extends Entity> EntityRenderer<? super T, ?> getRenderer(final T entity) {
         return (EntityRenderer<? super T, ?>)(switch (entity) {
             case AbstractClientPlayer abstractclientplayer -> this.getAvatarRenderer(
-                (Map<PlayerModelType, AvatarRenderer<T>>)this.playerRenderers, (T)abstractclientplayer
+                this.playerRenderers, abstractclientplayer
             );
-            case ClientMannequin clientmannequin -> this.getAvatarRenderer((Map<PlayerModelType, AvatarRenderer<T>>)this.mannequinRenderers, (T)clientmannequin);
+            case ClientMannequin clientmannequin -> this.getAvatarRenderer(this.mannequinRenderers, clientmannequin);
             default -> (EntityRenderer)this.renderers.get(entity.getType());
         });
     }
