--- a/net/minecraft/data/info/RegistryDumpReport.java
+++ b/net/minecraft/data/info/RegistryDumpReport.java
@@ -37,7 +37,7 @@
          jsonobject.addProperty("default", resourcelocation.toString());
       }
 
-      int j = Registry.f_122897_.m_7447_(p_124059_);
+      int j = ((Registry)Registry.f_122897_).m_7447_(p_124059_);
       jsonobject.addProperty("protocol_id", j);
       JsonObject jsonobject1 = new JsonObject();
 
