--- a/net/minecraft/client/resources/model/ModelBakery.java
+++ b/net/minecraft/client/resources/model/ModelBakery.java
@@ -173,7 +173,7 @@
 
         @Override
         public <T> T m_383903_(ModelBaker.SharedOperationKey<T> p_393371_) {
-            return (T)this.f_382156_.computeIfAbsent(p_393371_, this.f_382151_);
+            return (T)this.f_382156_.computeIfAbsent((ModelBaker.SharedOperationKey)p_393371_, this.f_382151_);
         }
     }
 }
