--- a/net/minecraft/util/SortedArraySet.java
+++ b/net/minecraft/util/SortedArraySet.java
@@ -22,7 +22,7 @@
    }
 
    public static <T extends Comparable<T>> SortedArraySet<T> func_226172_a_(int p_226172_0_) {
-      return new SortedArraySet<>(p_226172_0_, Comparator.naturalOrder());
+      return new SortedArraySet<>(p_226172_0_, Comparator.<T>naturalOrder());
    }
 
    private static <T> T[] func_226177_a_(Object[] p_226177_0_) {
@@ -96,7 +96,7 @@
    }
 
    public boolean remove(Object p_remove_1_) {
-      int i = this.func_226182_c_(p_remove_1_);
+      int i = this.func_226182_c_((T)p_remove_1_);
       if (i >= 0) {
          this.func_226183_d_(i);
          return true;
@@ -110,7 +110,7 @@
    }
 
    public boolean contains(Object p_contains_1_) {
-      int i = this.func_226182_c_(p_contains_1_);
+      int i = this.func_226182_c_((T)p_contains_1_);
       return i >= 0;
    }
 
