--- a/net\minecraft\util\datafix\fixes\PistonPushedBlock.java
+++ b/net\minecraft\util\datafix\fixes\PistonPushedBlock.java
@@ -28,7 +28,7 @@
       Typed<?> typed = (Typed)type.pointTyped(p_207419_1_.getOps()).orElseThrow(() -> {
          return new IllegalStateException("Could not create new piston block entity.");
       });
-      return typed.set(DSL.remainderFinder(), dynamic).set(opticfinder, (Typed)((Optional)type1.readTyped(dynamic1).getSecond()).orElseThrow(() -> {
+      return typed.set(DSL.remainderFinder(), dynamic).set(opticfinder, (Typed)(type1.readTyped(dynamic1).getSecond()).orElseThrow(() -> {
          return new IllegalStateException("Could not parse newly created block state tag.");
       }));
    }
