--- a/net/minecraft/util/valueproviders/ClampedNormalFloat.java
+++ b/net/minecraft/util/valueproviders/ClampedNormalFloat.java
@@ -8,7 +8,7 @@
 import net.minecraft.util.RandomSource;
 
 public class ClampedNormalFloat extends FloatProvider {
-   public static final Codec<ClampedNormalFloat> f_146411_ = RecordCodecBuilder.create((p_146431_) -> {
+   public static final Codec<ClampedNormalFloat> f_146411_ = RecordCodecBuilder.<ClampedNormalFloat>create((p_146431_) -> {
       return p_146431_.group(Codec.FLOAT.fieldOf("mean").forGetter((p_146449_) -> {
          return p_146449_.f_146412_;
       }), Codec.FLOAT.fieldOf("deviation").forGetter((p_146447_) -> {
