--- a/net/minecraft/client/gui/font/FontSet.java
+++ b/net/minecraft/client/gui/font/FontSet.java
@@ -104,7 +104,7 @@
         }
 
         Set<GlyphProvider> set = Sets.newHashSet();
-        intset.forEach(codepoint -> {
+        intset.forEach((int codepoint) -> {
             for (GlyphProvider glyphprovider : list) {
                 UnbakedGlyph unbakedglyph = glyphprovider.getGlyph(codepoint);
                 if (unbakedglyph != null) {
