--- a/net/minecraft/world/entity/monster/breeze/Breeze.java
+++ b/net/minecraft/world/entity/monster/breeze/Breeze.java
@@ -47,7 +47,7 @@
     private static final float FALL_DISTANCE_SOUND_TRIGGER_THRESHOLD = 3.0F;
     private static final int WHIRL_SOUND_FREQUENCY_MIN = 1;
     private static final int WHIRL_SOUND_FREQUENCY_MAX = 80;
-    private static final Brain.Provider<Breeze> BRAIN_PROVIDER = Brain.provider(
+    private static final Brain.Provider<Breeze> BRAIN_PROVIDER = Brain.<Breeze>provider(
         List.of(SensorType.NEAREST_LIVING_ENTITIES, SensorType.HURT_BY, SensorType.NEAREST_PLAYERS, SensorType.BREEZE_ATTACK_ENTITY_SENSOR),
         BreezeAi::getActivities
     );
@@ -81,7 +81,7 @@
 
     @Override
     public Brain<Breeze> getBrain() {
-        return super.getBrain();
+        return (Brain<Breeze>)super.getBrain();
     }
 
     @Override
