--- a/net/minecraft/client/renderer/blockentity/BeaconRenderer.java
+++ b/net/minecraft/client/renderer/blockentity/BeaconRenderer.java
@@ -110,14 +110,8 @@
             )
         );
         p_430215_.m_85849_();
-        f2 = -p_424527_;
-        float f3 = -p_424527_;
-        f4 = -p_424527_;
-        f5 = -p_424527_;
-        f9 = 0.0F;
-        f10 = 1.0F;
-        f11 = -1.0F + f1;
-        f12 = p_426161_ * p_430510_ + f11;
+        float f11_f = -1.0F + f1;
+        float f12_f = p_426161_ * p_430510_ + f11_f;
         p_423842_.m_415038_(
             p_430215_,
             RenderType.m_110460_(p_431585_, true),
@@ -127,18 +121,18 @@
                 ARGB.m_356955_(32, p_427198_),
                 p_427512_,
                 i,
-                f2,
-                f3,
+                -p_424527_,
+                -p_424527_,
                 p_424527_,
-                f4,
-                f5,
+                -p_424527_,
+                -p_424527_,
                 p_424527_,
                 p_424527_,
                 p_424527_,
                 0.0F,
                 1.0F,
-                f12,
-                f11
+                f12_f,
+                f11_f
             )
         );
         p_430215_.m_85849_();
