--- a/net/minecraft/network/chat/ComponentSerialization.java
+++ b/net/minecraft/network/chat/ComponentSerialization.java
@@ -86,7 +86,7 @@
          for(MapDecoder<? extends T> mapdecoder : this.f_303286_) {
             DataResult<? extends T> dataresult = mapdecoder.decode(p_311662_, p_310979_);
             if (dataresult.result().isPresent()) {
-               return dataresult;
+               return (DataResult<T>)dataresult;
             }
          }
 
@@ -96,7 +96,7 @@
       }
 
       public <S> RecordBuilder<S> encode(T p_310202_, DynamicOps<S> p_312954_, RecordBuilder<S> p_312771_) {
-         MapEncoder<T> mapencoder = this.f_303269_.apply(p_310202_);
+         MapEncoder<T> mapencoder = (MapEncoder<T>)this.f_303269_.apply(p_310202_);
          return mapencoder.encode(p_310202_, p_312954_, p_312771_);
       }
 
