--- a/net/minecraft/resources/RegistryResourceAccess.java
+++ b/net/minecraft/resources/RegistryResourceAccess.java
@@ -147,7 +147,7 @@
       }
    }
 
-   public static record ParsedEntry(E f_195950_, OptionalInt f_195951_) {
+   public static record ParsedEntry<E>(E f_195950_, OptionalInt f_195951_) {
       public static <E> RegistryResourceAccess.ParsedEntry<E> m_195956_(E p_195957_) {
          return new RegistryResourceAccess.ParsedEntry(p_195957_, OptionalInt.empty());
       }
