--- a/net/minecraft/resources/RegistryFixedCodec.java
+++ b/net/minecraft/resources/RegistryFixedCodec.java
@@ -50,7 +50,7 @@
                return optional.get().m_254902_(ResourceKey.m_135785_(this.f_206721_, resourcelocation)).map(DataResult::success).orElseGet(() -> {
                   return DataResult.error("Failed to get element " + resourcelocation);
                }).map((p_256041_) -> {
-                  return Pair.of(p_256041_, (T)p_255515_.getSecond());
+                  return Pair.of((Holder<E>) p_256041_, (T)p_255515_.getSecond());
                }).setLifecycle(Lifecycle.stable());
             });
          }
