--- a/net/minecraft/world/gen/feature/structure/MineshaftPieces.java
+++ b/net/minecraft/world/gen/feature/structure/MineshaftPieces.java
@@ -539,7 +539,7 @@
          }
 
          int k;
-         for(lvt_5_1_ = 0; k < this.field_74887_e.func_78883_b(); k = k + 4) {
+         for(k = 0; k < this.field_74887_e.func_78883_b(); k = k + 4) {
             k = k + p_74861_3_.nextInt(this.field_74887_e.func_78883_b());
             if (k + 3 > this.field_74887_e.func_78883_b()) {
                break;
