--- a/net/minecraft/core/registries/BuiltInRegistries.java
+++ b/net/minecraft/core/registries/BuiltInRegistries.java
@@ -258,7 +258,7 @@
         Bootstrap.m_179912_(() -> "registry " + p_259230_.m_135782_());
         ResourceLocation resourcelocation = p_259230_.m_135782_();
         f_256751_.put(resourcelocation, () -> p_259210_.m_257957_(p_260327_));
-        f_256935_.m_255290_((ResourceKey<WritableRegistry<?>>)p_259230_, p_260327_, RegistrationInfo.f_316022_);
+        f_256935_.m_255290_((ResourceKey)p_259230_, p_260327_, RegistrationInfo.f_316022_);
         return p_260327_;
     }
 
