--- a/net/minecraft/world/entity/monster/warden/WardenAi.java
+++ b/net/minecraft/world/entity/monster/warden/WardenAi.java
@@ -94,15 +94,15 @@
    }
 
    private static void m_225755_(Brain<Warden> p_225756_) {
-      p_225756_.m_21891_(Activity.f_37979_, 10, ImmutableList.of(new SetRoarTarget<>(Warden::m_225677_), new TryToSniff(), new RunOne(ImmutableMap.of(MemoryModuleType.f_225330_, MemoryStatus.VALUE_ABSENT), ImmutableList.of(Pair.of(new RandomStroll(0.5F), 2), Pair.of(new DoNothing(30, 60), 1)))));
+      p_225756_.m_21891_(Activity.f_37979_, 10, ImmutableList.of(new SetRoarTarget<>(Warden::m_225677_), new TryToSniff(), new RunOne<>(ImmutableMap.of(MemoryModuleType.f_225330_, MemoryStatus.VALUE_ABSENT), ImmutableList.of(Pair.of(new RandomStroll(0.5F), 2), Pair.of(new DoNothing(30, 60), 1)))));
    }
 
    private static void m_225760_(Brain<Warden> p_225761_) {
-      p_225761_.m_21895_(Activity.f_225870_, 5, ImmutableList.of(new SetRoarTarget<>(Warden::m_225677_), new GoToTargetLocation(MemoryModuleType.f_225328_, 2, 0.7F)), MemoryModuleType.f_225328_);
+      p_225761_.m_21895_(Activity.f_225870_, 5, ImmutableList.of(new SetRoarTarget<>(Warden::m_225677_), new GoToTargetLocation<>(MemoryModuleType.f_225328_, 2, 0.7F)), MemoryModuleType.f_225328_);
    }
 
    private static void m_225765_(Brain<Warden> p_225766_) {
-      p_225766_.m_21895_(Activity.f_225869_, 5, ImmutableList.of(new SetRoarTarget<>(Warden::m_225677_), new Sniffing(f_225711_)), MemoryModuleType.f_225330_);
+      p_225766_.m_21895_(Activity.f_225869_, 5, ImmutableList.of(new SetRoarTarget<>(Warden::m_225677_), new Sniffing<>(f_225711_)), MemoryModuleType.f_225330_);
    }
 
    private static void m_225767_(Brain<Warden> p_225768_) {
