--- a/net/minecraft/util/filefix/virtualfilesystem/CopyOnWriteFSProvider.java
+++ b/net/minecraft/util/filefix/virtualfilesystem/CopyOnWriteFSProvider.java
@@ -103,9 +103,9 @@
                     }
 
                     DirectoryNode directorynode = this.fs.fileTree().directoryByPath(Objects.requireNonNull(copyonwritefspath.getParent()));
-                    Path path = this.fs.createTemporaryFilePath();
-                    C c = channelFactory.newChannel(path, options, attrs);
-                    FileNode filenode1 = new FileNode(copyonwritefspath, path, true);
+                    Path pathx = this.fs.createTemporaryFilePath();
+                    C c = channelFactory.newChannel(pathx, options, attrs);
+                    FileNode filenode1 = new FileNode(copyonwritefspath, pathx, true);
                     directorynode.addChild(filenode1);
                     yield c;
                 }
