--- a/net/minecraft/world/level/NaturalSpawner.java
+++ b/net/minecraft/world/level/NaturalSpawner.java
@@ -83,14 +83,15 @@
 
          MobCategory mobcategory = entity.m_6095_().m_20674_();
          if (mobcategory != MobCategory.MISC) {
+            Entity finalEntity = entity;
             BlockPos blockpos = entity.m_142538_();
             p_186527_.m_47103_(ChunkPos.m_151388_(blockpos), (p_186541_) -> {
-               MobSpawnSettings.MobSpawnCost mobspawnsettings$mobspawncost = m_47095_(blockpos, p_186541_).m_47518_().m_48345_(entity.m_6095_());
+               MobSpawnSettings.MobSpawnCost mobspawnsettings$mobspawncost = m_47095_(blockpos, p_186541_).m_47518_().m_48345_(finalEntity.m_6095_());
                if (mobspawnsettings$mobspawncost != null) {
-                  potentialcalculator.m_47192_(entity.m_142538_(), mobspawnsettings$mobspawncost.m_48400_());
+                  potentialcalculator.m_47192_(finalEntity.m_142538_(), mobspawnsettings$mobspawncost.m_48400_());
                }
 
-               if (entity instanceof Mob) {
+               if (finalEntity instanceof Mob) {
                   p_186528_.m_186512_(p_186541_.m_7697_(), mobcategory);
                }
 
