--- a/net\minecraft\entity\Entity.java
+++ b/net\minecraft\entity\Entity.java
@@ -607,7 +607,7 @@
             }
          }
 
-         boolean flag = this.field_70122_E || p_70091_4_ != p_70091_4_ && p_70091_4_ < 0.0D;
+         boolean flag = this.field_70122_E || d3 != p_70091_4_ && d3 < 0.0D;
          if (this.field_70138_W > 0.0F && flag && (d2 != p_70091_2_ || d4 != p_70091_6_)) {
             double d6 = p_70091_2_;
             double d7 = p_70091_4_;
@@ -684,7 +684,7 @@
          this.field_70170_p.field_72984_F.func_76320_a("rest");
          this.func_174829_m();
          this.field_70123_F = d2 != p_70091_2_ || d4 != p_70091_6_;
-         this.field_70124_G = p_70091_4_ != p_70091_4_;
+         this.field_70124_G = d3 != p_70091_4_;
          this.field_70122_E = this.field_70124_G && d3 < 0.0D;
          this.field_70132_H = this.field_70123_F || this.field_70124_G;
          int j1 = MathHelper.func_76128_c(this.field_70165_t);
