--- a/net/minecraft/client/renderer/block/model/ModelBakery.java
+++ b/net/minecraft/client/renderer/block/model/ModelBakery.java
@@ -193,7 +193,7 @@
             label160: {
                try {
                   label161: {
-                     List lvt_9_4_;
+                     List<Pair<String, ModelBlockDefinition>> lvt_9_4_;
                      try {
                         flag = true;
                         lvt_9_4_ = (List)this.field_177598_f.func_199004_b(resourcelocation1).stream().map((p_209591_1_) -> {
