--- a/net/minecraft/util/IStringSerializable.java
+++ b/net/minecraft/util/IStringSerializable.java
@@ -18,7 +18,7 @@
    String func_176610_l();
 
    static <E extends Enum<E> & IStringSerializable> Codec<E> func_233023_a_(Supplier<E[]> p_233023_0_, Function<? super String, ? extends E> p_233023_1_) {
-      E[] ae = (Enum[])p_233023_0_.get();
+      E[] ae = p_233023_0_.get();
       return func_233024_a_(Enum::ordinal, (p_233026_1_) -> {
          return ae[p_233026_1_];
       }, p_233023_1_);
