--- a/net/minecraft/advancements/critereon/InventoryChangeTrigger.java
+++ b/net/minecraft/advancements/critereon/InventoryChangeTrigger.java
@@ -52,7 +52,7 @@
       });
 
       public static Criterion<InventoryChangeTrigger.TriggerInstance> m_43199_(ItemPredicate.Builder... p_297239_) {
-         return m_43197_(Stream.of(p_297239_).map(ItemPredicate.Builder::m_45077_).toArray((p_296132_) -> {
+         return m_43197_((ItemPredicate[])Stream.of(p_297239_).map(ItemPredicate.Builder::m_45077_).toArray((p_296132_) -> {
             return new ItemPredicate[p_296132_];
          }));
       }
