--- a/net/minecraft/util/valueproviders/TrapezoidFloat.java
+++ b/net/minecraft/util/valueproviders/TrapezoidFloat.java
@@ -7,7 +7,7 @@
 import net.minecraft.util.RandomSource;
 
 public class TrapezoidFloat extends FloatProvider {
-   public static final Codec<TrapezoidFloat> f_146561_ = RecordCodecBuilder.create((p_146578_) -> {
+   public static final Codec<TrapezoidFloat> f_146561_ = RecordCodecBuilder.<TrapezoidFloat>create((p_146578_) -> {
       return p_146578_.group(Codec.FLOAT.fieldOf("min").forGetter((p_146588_) -> {
          return p_146588_.f_146562_;
       }), Codec.FLOAT.fieldOf("max").forGetter((p_146586_) -> {
