--- a/net/minecraft/util/datafix/fixes/BedBlockEntityInjecter.java
+++ b/net/minecraft/util/datafix/fixes/BedBlockEntityInjecter.java
@@ -40,7 +40,7 @@
       int i = 416;
       return TypeRewriteRule.seq(this.fixTypeEverywhere("InjectBedBlockEntityType", this.getInputSchema().findChoiceType(References.f_16781_), this.getOutputSchema().findChoiceType(References.f_16781_), (p_14717_) -> {
          return (p_145085_) -> {
-            return p_145085_;
+            return (com.mojang.datafixers.util.Pair)p_145085_;
          };
       }), this.fixTypeEverywhereTyped("BedBlockEntityInjecter", this.getOutputSchema().getType(References.f_16773_), (p_14708_) -> {
          Typed<?> typed = p_14708_.getTyped(opticfinder);
@@ -58,7 +58,7 @@
             });
             int j1 = 0;
 
-            for(int k1 : stream::iterator) {
+            for(int k1 : (Iterable<Integer>) () -> stream.iterator()) {
                if (416 == (k1 & 255) << 4) {
                   int l1 = j1 & 15;
                   int i2 = j1 >> 8 & 15;
