--- a/net/minecraft/util/datafix/fixes/OptionsKeyTranslationFix.java
+++ b/net/minecraft/util/datafix/fixes/OptionsKeyTranslationFix.java
@@ -26,7 +26,7 @@
 
                   return Pair.of(p_145585_.getKey(), p_145585_.getValue());
                }).collect(Collectors.toMap(Pair::getFirst, Pair::getSecond)));
-            }).result().orElse(p_145582_);
+            }).result().orElse((com.mojang.serialization.Dynamic) p_145582_);
          });
       });
    }
