--- a/net/minecraft/client/util/SuffixArray.java
+++ b/net/minecraft/client/util/SuffixArray.java
@@ -58,7 +58,7 @@
          }
 
          public int compare(Integer p_compare_1_, Integer p_compare_2_) {
-            return this.compare(p_compare_1_, p_compare_2_);
+            return this.compare(p_compare_1_.intValue(), p_compare_2_.intValue());
          }
       };
       Swapper swapper = (p_194054_3_, p_194054_4_) -> {
