--- a/net/minecraft/network/play/server/SUpdateRecipesPacket.java
+++ b/net/minecraft/network/play/server/SUpdateRecipesPacket.java
@@ -55,6 +55,6 @@
    public static <T extends IRecipe<?>> void func_218771_a(T p_218771_0_, PacketBuffer p_218771_1_) {
       p_218771_1_.func_192572_a(Registry.field_218368_I.func_177774_c(p_218771_0_.func_199559_b()));
       p_218771_1_.func_192572_a(p_218771_0_.func_199560_c());
-      p_218771_0_.func_199559_b().func_199427_a_(p_218771_1_, p_218771_0_);
+      ((net.minecraft.item.crafting.IRecipeSerializer<T>)p_218771_0_.func_199559_b()).func_199427_a_(p_218771_1_, p_218771_0_);
    }
 }
