--- a/net/minecraft/util/datafix/fixes/OptionsKeyTranslationFix.java
+++ b/net/minecraft/util/datafix/fixes/OptionsKeyTranslationFix.java
@@ -15,7 +15,7 @@
    public TypeRewriteRule makeRule() {
       return this.fixTypeEverywhereTyped("OptionsKeyTranslationFix", this.getInputSchema().getType(References.f_16775_), (p_16648_) -> {
          return p_16648_.update(DSL.remainderFinder(), (p_16650_) -> {
-            return p_16650_.getMapValues().map((p_16656_) -> {
+            return p_16650_.getMapValues().<com.mojang.serialization.Dynamic<?>>map((p_16656_) -> {
                return p_16650_.createMap(p_16656_.entrySet().stream().map((p_16653_) -> {
                   if (p_16653_.getKey().asString("").startsWith("key_")) {
                      String s = p_16653_.getValue().asString("");
