--- a/net/minecraft/util/valueproviders/ClampedInt.java
+++ b/net/minecraft/util/valueproviders/ClampedInt.java
@@ -8,7 +8,7 @@
 import net.minecraft.util.RandomSource;
 
 public class ClampedInt extends IntProvider {
-   public static final Codec<ClampedInt> f_146383_ = RecordCodecBuilder.create((p_146400_) -> {
+   public static final Codec<ClampedInt> f_146383_ = RecordCodecBuilder.<ClampedInt>create((p_146400_) -> {
       return p_146400_.group(IntProvider.f_146531_.fieldOf("source").forGetter((p_146410_) -> {
          return p_146410_.f_146384_;
       }), Codec.INT.fieldOf("min_inclusive").forGetter((p_146408_) -> {
