--- a/net/minecraft/tags/TagContainer.java
+++ b/net/minecraft/tags/TagContainer.java
@@ -26,11 +26,11 @@
 
    @Nullable
    private <T> TagCollection<T> m_144471_(ResourceKey<? extends Registry<T>> p_144472_) {
-      return this.f_144433_.get(p_144472_);
+      return (TagCollection<T>)this.f_144433_.get(p_144472_);
    }
 
    public <T> TagCollection<T> m_144452_(ResourceKey<? extends Registry<T>> p_144453_) {
-      return this.f_144433_.getOrDefault(p_144453_, TagCollection.m_13410_());
+      return (TagCollection<T>)this.f_144433_.getOrDefault(p_144453_, TagCollection.<T>m_13410_());
    }
 
    public <T, E extends Exception> Tag<T> m_144458_(ResourceKey<? extends Registry<T>> p_144459_, ResourceLocation p_144460_, Function<ResourceLocation, E> p_144461_) throws E {
@@ -68,7 +68,7 @@
    }
 
    private static <T> void m_144438_(TagContainer.CollectionConsumer p_144439_, ResourceKey<? extends Registry<?>> p_144440_, TagCollection<?> p_144441_) {
-      p_144439_.m_142574_(p_144440_, p_144441_);
+      p_144439_.m_142574_((ResourceKey<? extends Registry<T>>)p_144440_, (TagCollection<T>)p_144441_);
    }
 
    public void m_13431_() {
