--- a/net/minecraft/util/valueproviders/BiasedToBottomInt.java
+++ b/net/minecraft/util/valueproviders/BiasedToBottomInt.java
@@ -7,7 +7,7 @@
 import java.util.function.Function;
 
 public class BiasedToBottomInt extends IntProvider {
-   public static final Codec<BiasedToBottomInt> f_146359_ = RecordCodecBuilder.create((p_146373_) -> {
+   public static final Codec<BiasedToBottomInt> f_146359_ = RecordCodecBuilder.<BiasedToBottomInt>create((p_146373_) -> {
       return p_146373_.group(Codec.INT.fieldOf("min_inclusive").forGetter((p_146381_) -> {
          return p_146381_.f_146360_;
       }), Codec.INT.fieldOf("max_inclusive").forGetter((p_146378_) -> {
