--- a/net/minecraft/entity/ai/brain/task/InteractWithEntityTask.java
+++ b/net/minecraft/entity/ai/brain/task/InteractWithEntityTask.java
@@ -56,7 +56,7 @@
          p_220437_3_.stream().filter((p_220440_1_) -> {
             return this.field_220448_c.equals(p_220440_1_.func_200600_R());
          }).map((p_220439_0_) -> {
-            return p_220439_0_;
+            return (T)p_220439_0_;
          }).filter((p_220443_2_) -> {
             return p_220443_2_.func_70068_e(p_212831_2_) <= (double)this.field_220449_d;
          }).filter(this.field_220450_e).findFirst().ifPresent((p_220438_2_) -> {
