--- a/net/minecraft/world/entity/monster/piglin/PiglinBrute.java
+++ b/net/minecraft/world/entity/monster/piglin/PiglinBrute.java
@@ -33,7 +33,7 @@
     private static final float MOVEMENT_SPEED_WHEN_FIGHTING = 0.35F;
     private static final int ATTACK_DAMAGE = 7;
     private static final double TARGETING_RANGE = 12.0;
-    public static final Brain.Provider<PiglinBrute> BRAIN_PROVIDER = Brain.provider(
+    public static final Brain.Provider<PiglinBrute> BRAIN_PROVIDER = Brain.<PiglinBrute>provider(
         List.of(MemoryModuleType.NEAREST_VISIBLE_ADULT_PIGLINS),
         List.of(
             SensorType.NEAREST_LIVING_ENTITIES,
@@ -79,7 +79,7 @@
 
     @Override
     public Brain<PiglinBrute> getBrain() {
-        return super.getBrain();
+        return (Brain<PiglinBrute>)super.getBrain();
     }
 
     @Override
