--- a/net/minecraft/item/crafting/RecipeSerializers.java
+++ b/net/minecraft/item/crafting/RecipeSerializers.java
@@ -61,7 +61,7 @@
    public static <T extends IRecipe> void func_199574_a(T p_199574_0_, PacketBuffer p_199574_1_) {
       p_199574_1_.func_192572_a(p_199574_0_.func_199560_c());
       p_199574_1_.func_180714_a(p_199574_0_.func_199559_b().func_199567_a());
-      IRecipeSerializer<T> irecipeserializer = p_199574_0_.func_199559_b();
+      IRecipeSerializer<T> irecipeserializer = (IRecipeSerializer<T>)p_199574_0_.func_199559_b();
       irecipeserializer.func_199427_a_(p_199574_1_, p_199574_0_);
    }
 
