--- a/net\minecraft\tags\NetworkTagCollection.java
+++ b/net\minecraft\tags\NetworkTagCollection.java
@@ -43,7 +43,7 @@
             list.add(this.field_200044_a.func_148754_a(p_200043_1_.func_150792_a()));
          }
 
-         this.func_200039_c().put(resourcelocation, Tag.Builder.func_200047_a().func_200046_a(list).func_200051_a(resourcelocation));
+         this.func_200039_c().put(resourcelocation, Tag.Builder.<T>func_200047_a().func_200046_a(list).func_200051_a(resourcelocation));
       }
 
    }
