--- a/net/minecraft/world/entity/monster/piglin/PiglinAi.java
+++ b/net/minecraft/world/entity/monster/piglin/PiglinAi.java
@@ -172,7 +172,7 @@
                 BehaviorBuilder.m_257845_(PiglinAi::m_34918_, BackUpIfTooClose.m_257698_(5, 0.75F)),
                 SetWalkTargetFromAttackTargetIfTargetOutOfReach.m_257469_(1.0F),
                 MeleeAttack.m_257733_(20),
-                new CrossbowAttack(),
+                new CrossbowAttack<>(),
                 RememberIfHoglinWasKilled.m_257745_(),
                 EraseMemoryIf.m_258093_(PiglinAi::m_34998_, MemoryModuleType.f_26372_)
             ),
@@ -193,7 +193,7 @@
                 new RunOne<LivingEntity>(
                     ImmutableList.of(
                         Pair.of(SetEntityLookTarget.m_258096_(EntityType.f_20511_, 8.0F), 1),
-                        Pair.of(RandomStroll.m_257960_(0.6F, 2, 1), 1),
+                        Pair.of((BehaviorControl<LivingEntity>)(BehaviorControl)RandomStroll.m_257960_(0.6F, 2, 1), 1),
                         Pair.of(new DoNothing(10, 20), 1)
                     )
                 )
@@ -223,7 +223,7 @@
                 SetWalkTargetAwayFrom.m_257370_(MemoryModuleType.f_26383_, 1.0F, 12, true),
                 m_34805_(),
                 m_34882_(),
-                EraseMemoryIf.<PathfinderMob>m_258093_(PiglinAi::m_35008_, MemoryModuleType.f_26383_)
+                EraseMemoryIf.m_258093_(PiglinAi::m_35008_, MemoryModuleType.f_26383_)
             ),
             MemoryModuleType.f_26383_
         );
@@ -245,7 +245,7 @@
                             .build()
                     )
                 ),
-                DismountOrSkipMounting.<LivingEntity>m_257459_(8, PiglinAi::m_34834_)
+                DismountOrSkipMounting.m_257459_(8, PiglinAi::m_34834_)
             ),
             MemoryModuleType.f_26376_
         );
