--- a/net/minecraft/client/renderer/entity/RenderZombie.java
+++ b/net/minecraft/client/renderer/entity/RenderZombie.java
@@ -12,8 +12,8 @@
       super(p_i46127_1_, new ModelZombie(), 0.5F);
       LayerBipedArmor layerbipedarmor = new LayerBipedArmor(this) {
          protected void func_177177_a() {
-            this.field_177189_c = (T)(new ModelZombie(0.5F, true));
-            this.field_177186_d = (T)(new ModelZombie(1.0F, true));
+            this.field_177189_c = new ModelZombie(0.5F, true);
+            this.field_177186_d = new ModelZombie(1.0F, true);
          }
       };
       this.func_177094_a(layerbipedarmor);
