--- a/net/minecraft/world/level/levelgen/WorldDimensions.java
+++ b/net/minecraft/world/level/levelgen/WorldDimensions.java
@@ -172,18 +172,17 @@
    }
 
    public WorldDimensions.Complete m_245300_(Registry<LevelStem> p_248787_) {
+      record Entry(ResourceKey<LevelStem> f_244236_, LevelStem f_243758_) {
+         Lifecycle m_246583_() {
+            return WorldDimensions.m_245357_(this.f_244236_, this.f_243758_);
+         }
+      }
       Stream<ResourceKey<LevelStem>> stream = Stream.concat(p_248787_.m_214010_().stream(), this.f_243948_.m_214010_().stream()).distinct();
       List<Entry> list = new ArrayList<>();
       m_247421_(stream).forEach((p_248571_) -> {
          p_248787_.m_123009_(p_248571_).or(() -> {
             return this.f_243948_.m_123009_(p_248571_);
          }).ifPresent((p_250263_) -> {
-            record Entry(ResourceKey<LevelStem> f_244236_, LevelStem f_243758_) {
-               Lifecycle m_246583_() {
-                  return WorldDimensions.m_245357_(this.f_244236_, this.f_243758_);
-               }
-            }
-
             list.add(new Entry(p_248571_, p_250263_));
          });
       });
