--- a/net/minecraft/client/renderer/item/ConditionalItemModel.java
+++ b/net/minecraft/client/renderer/item/ConditionalItemModel.java
@@ -91,7 +91,7 @@
         private static <T extends ConditionalItemModelProperty> T swapContext(
             final T originalProperty, final RegistryContextSwapper contextSwapper, final ClientLevel context
         ) {
-            return (T)contextSwapper.swapTo(originalProperty.type().codec(), originalProperty, context.registryAccess()).result().orElse(originalProperty);
+            return (T)contextSwapper.swapTo(((MapCodec<T>)originalProperty.type()).codec(), originalProperty, context.registryAccess()).result().orElse(originalProperty);
         }
 
         @Override
