--- a/net/minecraft/util/SortedArraySet.java
+++ b/net/minecraft/util/SortedArraySet.java
@@ -22,7 +22,7 @@
    }
 
    public static <T extends Comparable<T>> SortedArraySet<T> func_226172_a_(int p_226172_0_) {
-      return new SortedArraySet<>(p_226172_0_, Comparator.naturalOrder());
+      return new SortedArraySet<>(p_226172_0_, Comparator.<T>naturalOrder());
    }
 
    private static <T> T[] func_226177_a_(Object[] p_226177_0_) {
