--- a/net/minecraft/world/chunk/Chunk.java
+++ b/net/minecraft/world/chunk/Chunk.java
@@ -732,7 +732,7 @@
 
    public void func_222879_B() {
       if (this.field_201621_s instanceof ChunkPrimerTickList) {
-         ((ChunkPrimerTickList)this.field_201621_s).func_205381_a(this.field_76637_e.func_205220_G_(), (p_222881_1_) -> {
+         ((ChunkPrimerTickList<Block>)this.field_201621_s).func_205381_a(this.field_76637_e.func_205220_G_(), (p_222881_1_) -> {
             return this.func_180495_p(p_222881_1_).func_177230_c();
          });
          this.field_201621_s = EmptyTickList.func_205388_a();
@@ -742,7 +742,7 @@
       }
 
       if (this.field_205325_u instanceof ChunkPrimerTickList) {
-         ((ChunkPrimerTickList)this.field_205325_u).func_205381_a(this.field_76637_e.func_205219_F_(), (p_222878_1_) -> {
+         ((ChunkPrimerTickList<Fluid>)this.field_205325_u).func_205381_a(this.field_76637_e.func_205219_F_(), (p_222878_1_) -> {
             return this.func_204610_c(p_222878_1_).func_206886_c();
          });
          this.field_205325_u = EmptyTickList.func_205388_a();
@@ -754,12 +754,12 @@
    }
 
    public void func_222880_a(ServerWorld p_222880_1_) {
-      if (this.field_201621_s == EmptyTickList.func_205388_a()) {
+      if (this.field_201621_s == EmptyTickList.<Block>func_205388_a()) {
          this.field_201621_s = new SerializableTickList<>(Registry.field_212618_g::func_177774_c, p_222880_1_.func_205220_G_().func_223188_a(this.field_212816_F, true, false));
          this.func_177427_f(true);
       }
 
-      if (this.field_205325_u == EmptyTickList.func_205388_a()) {
+      if (this.field_205325_u == EmptyTickList.<Fluid>func_205388_a()) {
          this.field_205325_u = new SerializableTickList<>(Registry.field_212619_h::func_177774_c, p_222880_1_.func_205219_F_().func_223188_a(this.field_212816_F, true, false));
          this.func_177427_f(true);
       }
