--- a/net/minecraft/world/level/levelgen/feature/configurations/SpringConfiguration.java
+++ b/net/minecraft/world/level/levelgen/feature/configurations/SpringConfiguration.java
@@ -20,7 +20,7 @@
       }), Codec.INT.fieldOf("hole_count").orElse(1).forGetter((p_161199_) -> {
          return p_161199_.f_68127_;
       }), Registry.f_122824_.m_194605_().listOf().fieldOf("valid_blocks").xmap(ImmutableSet::copyOf, ImmutableList::copyOf).forGetter((p_161197_) -> {
-         return p_161197_.f_68128_;
+         return (ImmutableSet<Block>)p_161197_.f_68128_;
       })).apply(p_68139_, SpringConfiguration::new);
    });
    public final FluidState f_68124_;
