--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -563,7 +563,7 @@
          if (j == 0 || this.func_71255_r()) {
             WorldServer worldserver = this.field_71305_c[j];
             this.field_71304_b.func_194340_a(() -> {
-               return p_194304_0_.func_72912_H().func_76065_j();
+               return worldserver.func_72912_H().func_76065_j();
             });
             if (this.field_71315_w % 20 == 0) {
                this.field_71304_b.func_76320_a("timeSync");
