--- a/net/minecraft/util/datafix/fixes/ChunkProtoTickListFix.java
+++ b/net/minecraft/util/datafix/fixes/ChunkProtoTickListFix.java
@@ -40,7 +40,7 @@
       OpticFinder<?> opticfinder3 = opticfinder2.type().findField("block_states");
       OpticFinder<?> opticfinder4 = opticfinder2.type().findField("biomes");
       OpticFinder<?> opticfinder5 = opticfinder3.type().findField("palette");
-      OpticFinder<?> opticfinder6 = opticfinder.type().findField("TileTicks");
+      OpticFinder<Typed<?>> opticfinder6 = (OpticFinder<Typed<?>>) opticfinder.type().findField("TileTicks");
       return this.fixTypeEverywhereTyped("ChunkProtoTickListFix", type, (p_185002_) -> {
          return p_185002_.updateTyped(opticfinder, (p_185010_) -> {
             p_185010_ = p_185010_.update(DSL.remainderFinder(), (p_185078_) -> {
@@ -85,9 +85,9 @@
                int j = dynamic.get("zPos").asInt(0);
                Dynamic<?> dynamic1 = this.m_185036_(dynamic, int2objectmap, b0, i, j, "LiquidsToBeTicked", ChunkProtoTickListFix::m_185068_);
                Dynamic<?> dynamic2 = this.m_185036_(dynamic, int2objectmap, b0, i, j, "ToBeTicked", ChunkProtoTickListFix::m_185031_);
-               Optional<? extends Pair<? extends Typed<?>, ?>> optional = opticfinder6.type().readTyped(dynamic2).result();
+               Optional<? extends Pair<? extends Typed<Typed<?>>, ?>> optional = opticfinder6.type().readTyped(dynamic2).result();
                if (optional.isPresent()) {
-                  p_185010_ = p_185010_.set(opticfinder6, optional.get().getFirst());
+                  p_185010_ = p_185010_.<Typed<?>, Typed<?>>set(opticfinder6, optional.get().getFirst());
                }
 
                return p_185010_.update(DSL.remainderFinder(), (p_185035_) -> {
@@ -144,7 +144,7 @@
       int j = p_185050_ >>> 4 & 15;
       int k = p_185050_ >>> 8 & 15;
       String s = p_185051_.apply(p_185046_ != null ? p_185046_.get().m_185090_(i, j, k) : null);
-      return p_185045_.createMap(ImmutableMap.builder().put(p_185045_.createString("i"), p_185045_.createString(s)).put(p_185045_.createString("x"), p_185045_.createInt(p_185047_ * 16 + i)).put(p_185045_.createString("y"), p_185045_.createInt(p_185048_ * 16 + j)).put(p_185045_.createString("z"), p_185045_.createInt(p_185049_ * 16 + k)).put(p_185045_.createString("t"), p_185045_.createInt(0)).put(p_185045_.createString("p"), p_185045_.createInt(0)).build());
+      return p_185045_.createMap(ImmutableMap.<Dynamic<?>, Dynamic<?>>builder().put(p_185045_.createString("i"), p_185045_.createString(s)).put(p_185045_.createString("x"), p_185045_.createInt(p_185047_ * 16 + i)).put(p_185045_.createString("y"), p_185045_.createInt(p_185048_ * 16 + j)).put(p_185045_.createString("z"), p_185045_.createInt(p_185049_ * 16 + k)).put(p_185045_.createString("t"), p_185045_.createInt(0)).put(p_185045_.createString("p"), p_185045_.createInt(0)).build());
    }
 
    public static final class PoorMansPalettedContainer {
