--- a/net/minecraft/util/datafix/fixes/OptionsKeyLwjgl3Fix.java
+++ b/net/minecraft/util/datafix/fixes/OptionsKeyLwjgl3Fix.java
@@ -133,7 +133,7 @@
    public TypeRewriteRule makeRule() {
       return this.fixTypeEverywhereTyped("OptionsKeyLwjgl3Fix", this.getInputSchema().getType(References.f_16775_), (p_16633_) -> {
          return p_16633_.update(DSL.remainderFinder(), (p_16635_) -> {
-            return p_16635_.getMapValues().map((p_16638_) -> {
+            return p_16635_.getMapValues().<com.mojang.serialization.Dynamic<?>>map((p_16638_) -> {
                return p_16635_.createMap(p_16638_.entrySet().stream().map((p_16642_) -> {
                   if (p_16642_.getKey().asString("").startsWith("key_")) {
                      int i = Integer.parseInt(p_16642_.getValue().asString(""));
