diff -r -U 3 minecraft_server\net\minecraft\world\chunk\storage\AnvilChunkLoader.java minecraft_server_patched\net\minecraft\world\chunk\storage\AnvilChunkLoader.java
--- minecraft_server\net\minecraft\world\chunk\storage\AnvilChunkLoader.java
+++ minecraft_server_patched\net\minecraft\world\chunk\storage\AnvilChunkLoader.java
@@ -165,11 +165,7 @@
       try {
          this.field_183014_e = true;
 
-         while(true) {
-            if (this.func_75814_c()) {
-               continue;
-            }
-         }
+         while(this.func_75814_c());
       } finally {
          this.field_183014_e = false;
       }
