diff -r -U 3 minecraft_server\net\minecraft\util\IObjectIntIterable.java minecraft_server_patched\net\minecraft\util\IObjectIntIterable.java
--- minecraft_server\net\minecraft\util\IObjectIntIterable.java
+++ minecraft_server_patched\net\minecraft\util\IObjectIntIterable.java
@@ -1,4 +1,4 @@
 package net.minecraft.util;
 
-public interface IObjectIntIterable extends Iterable {
+public interface IObjectIntIterable<V> extends Iterable<V> {
 }
