diff -r -U 3 minecraft_server\net\minecraft\server\management\UserList.java minecraft_server_patched\net\minecraft\server\management\UserList.java
--- minecraft_server\net\minecraft\server\management\UserList.java
+++ minecraft_server_patched\net\minecraft\server\management\UserList.java
@@ -128,7 +128,7 @@
    }
 
    protected UserListEntry<K> func_152682_a(JsonObject p_152682_1_) {
-      return new UserListEntry<K>((Object)null, p_152682_1_);
+      return new UserListEntry<K>(null, p_152682_1_);
    }
 
    protected Map<String, V> func_152688_e() {
@@ -166,7 +166,7 @@
 
             for(UserListEntry<K> userlistentry : collection) {
                if (userlistentry.func_152640_f() != null) {
-                  this.field_152696_d.put(this.func_152681_a(userlistentry.func_152640_f()), userlistentry);
+                  this.field_152696_d.put(this.func_152681_a(userlistentry.func_152640_f()), (V)userlistentry);
                }
             }
          }
