diff -r -U 3 minecraft_server\net\minecraft\network\datasync\EntityDataManager.java minecraft_server_patched\net\minecraft\network\datasync\EntityDataManager.java
--- minecraft_server\net\minecraft\network\datasync\EntityDataManager.java
+++ minecraft_server_patched\net\minecraft\network\datasync\EntityDataManager.java
@@ -96,7 +96,7 @@
 
       EntityDataManager.DataEntry<T> dataentry;
       try {
-         dataentry = this.field_187234_c.get(p_187219_1_.func_187155_a());
+         dataentry = (EntityDataManager.DataEntry<T>)this.field_187234_c.get(p_187219_1_.func_187155_a());
       } catch (Throwable throwable) {
          CrashReport crashreport = CrashReport.func_85055_a(throwable, "Getting synched entity data");
          CrashReportCategory crashreportcategory = crashreport.func_85058_a("Synched entity data");
