diff -r -U 3 minecraft_server\net\minecraft\util\text\TextComponentKeybind.java minecraft_server_patched\net\minecraft\util\text\TextComponentKeybind.java
--- minecraft_server\net\minecraft\util\text\TextComponentKeybind.java
+++ minecraft_server_patched\net\minecraft\util\text\TextComponentKeybind.java
@@ -6,7 +6,7 @@
 public class TextComponentKeybind extends TextComponentBase {
    public static Function<String, Supplier<String>> field_193637_b = (p_193635_0_) -> {
       return () -> {
-         return p_193634_0_;
+         return p_193635_0_;
       };
    };
    private final String field_193638_c;
