diff -r -U 3 minecraft_server\net\minecraft\block\state\pattern\FactoryBlockPattern.java minecraft_server_patched\net\minecraft\block\state\pattern\FactoryBlockPattern.java
--- minecraft_server\net\minecraft\block\state\pattern\FactoryBlockPattern.java
+++ minecraft_server_patched\net\minecraft\block\state\pattern\FactoryBlockPattern.java
@@ -41,7 +41,7 @@
 
                for(char c0 : s.toCharArray()) {
                   if (!this.field_177666_c.containsKey(Character.valueOf(c0))) {
-                     this.field_177666_c.put(Character.valueOf(c0), (Object)null);
+                     this.field_177666_c.put(Character.valueOf(c0), null);
                   }
                }
             }
