--- a/net/minecraft/client/gui/narration/ScreenNarrationCollector.java
+++ b/net/minecraft/client/gui/narration/ScreenNarrationCollector.java
@@ -10,7 +10,7 @@
 @OnlyIn(Dist.CLIENT)
 public class ScreenNarrationCollector {
    int f_169181_;
-   final Map<ScreenNarrationCollector.EntryKey, ScreenNarrationCollector.NarrationEntry> f_169182_ = Maps.newTreeMap(Comparator.comparing((p_169196_) -> {
+   final Map<ScreenNarrationCollector.EntryKey, ScreenNarrationCollector.NarrationEntry> f_169182_ = Maps.newTreeMap(Comparator.<ScreenNarrationCollector.EntryKey, NarratedElementType>comparing((p_169196_) -> {
       return p_169196_.f_169207_;
    }).thenComparing((p_169185_) -> {
       return p_169185_.f_169208_;
