--- a/net/minecraft/network/NettyEncryptingEncoder.java
+++ b/net/minecraft/network/NettyEncryptingEncoder.java
@@ -13,7 +13,7 @@
       this.field_150750_a = new NettyEncryptionTranslator(p_i45142_1_);
    }
 
-   protected void encode(ChannelHandlerContext p_encode_1_, ByteBuf p_encode_2_, ByteBuf p_encode_3_) throws Exception, ShortBufferException {
+   protected void encode(ChannelHandlerContext p_encode_1_, ByteBuf p_encode_2_, ByteBuf p_encode_3_) throws ShortBufferException, Exception {
       this.field_150750_a.func_150504_a(p_encode_2_, p_encode_3_);
    }
 }
