--- a/net/minecraft/world/entity/SpawnPlacements.java
+++ b/net/minecraft/world/entity/SpawnPlacements.java
@@ -60,7 +60,7 @@
 
    public static <T extends Entity> boolean m_21759_(EntityType<T> p_21760_, ServerLevelAccessor p_21761_, MobSpawnType p_21762_, BlockPos p_21763_, Random p_21764_) {
       SpawnPlacements.Data spawnplacements$data = f_21750_.get(p_21760_);
-      return spawnplacements$data == null || spawnplacements$data.f_21769_.m_21780_(p_21760_, p_21761_, p_21762_, p_21763_, p_21764_);
+      return spawnplacements$data == null || spawnplacements$data.f_21769_.m_21780_((EntityType)p_21760_, p_21761_, p_21762_, p_21763_, p_21764_);
    }
 
    static {
