--- a/net/minecraft/network/protocol/ProtocolInfoBuilder.java
+++ b/net/minecraft/network/protocol/ProtocolInfoBuilder.java
@@ -42,7 +42,7 @@
         final PacketType<P> bundlerPacket, final Function<Iterable<Packet<? super T>>, P> constructor, final D delimiterPacket
     ) {
         StreamCodec<ByteBuf, D> streamcodec = StreamCodec.unit(delimiterPacket);
-        PacketType<D> packettype = (PacketType<D>)delimiterPacket.type();
+        PacketType<D> packettype = (PacketType)delimiterPacket.type();
         this.codecs.add(new ProtocolInfoBuilder.CodecEntry<>(packettype, streamcodec, null));
         this.bundlerInfo = BundlerInfo.createForPacket(bundlerPacket, constructor, delimiterPacket);
         return this;
