--- a/net/minecraft/util/Util.java
+++ b/net/minecraft/util/Util.java
@@ -203,7 +203,7 @@
       @OnlyIn(Dist.CLIENT)
       public void func_195639_a(URL p_195639_1_) {
          try {
-            Process process = (Process)AccessController.doPrivileged(() -> {
+            Process process = (Process)AccessController.doPrivileged((java.security.PrivilegedExceptionAction<Process>) () -> {
                return Runtime.getRuntime().exec(this.func_195643_b(p_195639_1_));
             });
 
