--- a/net/minecraft/util/datafix/fixes/HorseSplit.java
+++ b/net/minecraft/util/datafix/fixes/HorseSplit.java
@@ -39,7 +39,7 @@
 
          dynamic.remove("Type");
          Type<?> type = this.getOutputSchema().findChoiceType(TypeReferences.field_211299_o).types().get(s);
-         return Pair.of(s, (Typed<?>)((Pair)p_209149_2_.write().flatMap(type::readTyped).result().orElseThrow(() -> {
+         return Pair.of(s, (Typed<?>)((Pair)((com.mojang.serialization.DataResult<Dynamic<?>>)p_209149_2_.write()).flatMap(type::readTyped).result().orElseThrow(() -> {
             return new IllegalStateException("Could not parse the new horse");
          })).getFirst());
       } else {
