--- a/net/minecraft/client/renderer/texture/TextureUtil.java
+++ b/net/minecraft/client/renderer/texture/TextureUtil.java
@@ -119,11 +119,7 @@
          FileChannel filechannel = fileinputstream.getChannel();
          bytebuffer = MemoryUtil.memAlloc((int)filechannel.size() + 1);
 
-         while(true) {
-            if (filechannel.read(bytebuffer) != -1) {
-               continue;
-            }
-         }
+         while(filechannel.read(bytebuffer) != -1);
       } else {
          bytebuffer = MemoryUtil.memAlloc(8192);
          ReadableByteChannel readablebytechannel = Channels.newChannel(p_195724_0_);
