--- a/net/minecraft/util/registry/WorldGenRegistries.java
+++ b/net/minecraft/util/registry/WorldGenRegistries.java
@@ -61,8 +61,8 @@
    private static <T, R extends MutableRegistry<T>> R func_243666_a(RegistryKey<? extends Registry<T>> p_243666_0_, R p_243666_1_, Supplier<T> p_243666_2_, Lifecycle p_243666_3_) {
       ResourceLocation resourcelocation = p_243666_0_.func_240901_a_();
       field_243659_k.put(resourcelocation, p_243666_2_);
-      MutableRegistry<R> mutableregistry = field_243660_l;
-      return mutableregistry.func_218381_a(p_243666_0_, p_243666_1_, p_243666_3_);
+      MutableRegistry<R> mutableregistry = (MutableRegistry<R>)field_243660_l;
+      return (R)mutableregistry.func_218381_a((RegistryKey)p_243666_0_, p_243666_1_, p_243666_3_);
    }
 
    public static <T> T func_243663_a(Registry<? super T> p_243663_0_, String p_243663_1_, T p_243663_2_) {
@@ -70,11 +70,11 @@
    }
 
    public static <V, T extends V> T func_243664_a(Registry<V> p_243664_0_, ResourceLocation p_243664_1_, T p_243664_2_) {
-      return ((MutableRegistry)p_243664_0_).func_218381_a(RegistryKey.func_240903_a_(p_243664_0_.func_243578_f(), p_243664_1_), p_243664_2_, Lifecycle.stable());
+      return ((MutableRegistry<V>)p_243664_0_).func_218381_a(RegistryKey.func_240903_a_(p_243664_0_.func_243578_f(), p_243664_1_), p_243664_2_, Lifecycle.stable());
    }
 
    public static <V, T extends V> T func_243662_a(Registry<V> p_243662_0_, int p_243662_1_, RegistryKey<V> p_243662_2_, T p_243662_3_) {
-      return ((MutableRegistry)p_243662_0_).func_218382_a(p_243662_1_, p_243662_2_, p_243662_3_, Lifecycle.stable());
+      return ((MutableRegistry<V>)p_243662_0_).func_218382_a(p_243662_1_, p_243662_2_, p_243662_3_, Lifecycle.stable());
    }
 
    public static void func_243661_a() {
