--- a/net/minecraft/item/ItemElytra.java
+++ b/net/minecraft/item/ItemElytra.java
@@ -14,9 +14,7 @@
 public class ItemElytra extends Item {
    public ItemElytra(Item.Builder p_i48507_1_) {
       super(p_i48507_1_);
-      this.func_185043_a(new ResourceLocation("broken"), (p_210312_0_, p_210312_1_, p_210312_2_) -> {
-         return func_185069_d(p_210312_0_) ? 0.0F : 1.0F;
-      });
+      this.func_185043_a(new ResourceLocation("broken"), new IItemPropertyGetter() {});
       BlockDispenser.func_199774_a(this, ItemArmor.field_96605_cw);
    }
 
