--- a/net/minecraft/item/ItemBow.java
+++ b/net/minecraft/item/ItemBow.java
@@ -18,17 +18,9 @@
 public class ItemBow extends Item {
    public ItemBow(Item.Builder p_i48522_1_) {
       super(p_i48522_1_);
-      this.func_185043_a(new ResourceLocation("pull"), (p_210310_0_, p_210310_1_, p_210310_2_) -> {
-         if (p_210310_2_ == null) {
-            return 0.0F;
-         } else {
-            return p_210310_2_.func_184607_cu().func_77973_b() != Items.field_151031_f ? 0.0F : (float)(p_210310_0_.func_77988_m() - p_210310_2_.func_184605_cv()) / 20.0F;
+      this.func_185043_a(new ResourceLocation("pull"), new IItemPropertyGetter() {});
+      this.func_185043_a(new ResourceLocation("pulling"), new IItemPropertyGetter() {});
          }
-      });
-      this.func_185043_a(new ResourceLocation("pulling"), (p_210309_0_, p_210309_1_, p_210309_2_) -> {
-         return p_210309_2_ != null && p_210309_2_.func_184587_cr() && p_210309_2_.func_184607_cu() == p_210309_0_ ? 1.0F : 0.0F;
-      });
-   }
 
    private ItemStack func_185060_a(EntityPlayer p_185060_1_) {
       if (this.func_185058_h_(p_185060_1_.func_184586_b(EnumHand.OFF_HAND))) {
