--- a/net/minecraft/item/ItemFishingRod.java
+++ b/net/minecraft/item/ItemFishingRod.java
@@ -15,21 +15,9 @@
 public class ItemFishingRod extends Item {
    public ItemFishingRod(Item.Builder p_i48494_1_) {
       super(p_i48494_1_);
-      this.func_185043_a(new ResourceLocation("cast"), (p_210313_0_, p_210313_1_, p_210313_2_) -> {
-         if (p_210313_2_ == null) {
-            return 0.0F;
-         } else {
-            boolean flag = p_210313_2_.func_184614_ca() == p_210313_0_;
-            boolean flag1 = p_210313_2_.func_184592_cb() == p_210313_0_;
-            if (p_210313_2_.func_184614_ca().func_77973_b() instanceof ItemFishingRod) {
-               flag1 = false;
+      this.func_185043_a(new ResourceLocation("cast"), new IItemPropertyGetter() {});
             }
 
-            return (flag || flag1) && p_210313_2_ instanceof EntityPlayer && ((EntityPlayer)p_210313_2_).field_71104_cf != null ? 1.0F : 0.0F;
-         }
-      });
-   }
-
    public ActionResult<ItemStack> func_77659_a(World p_77659_1_, EntityPlayer p_77659_2_, EnumHand p_77659_3_) {
       ItemStack itemstack = p_77659_2_.func_184586_b(p_77659_3_);
       if (p_77659_2_.field_71104_cf != null) {
