--- a/net/minecraft/item/ItemTrident.java
+++ b/net/minecraft/item/ItemTrident.java
@@ -26,9 +26,7 @@
 public class ItemTrident extends Item {
    public ItemTrident(Item.Builder p_i48788_1_) {
       super(p_i48788_1_);
-      this.func_185043_a(new ResourceLocation("throwing"), (p_210315_0_, p_210315_1_, p_210315_2_) -> {
-         return p_210315_2_ != null && p_210315_2_.func_184587_cr() && p_210315_2_.func_184607_cu() == p_210315_0_ ? 1.0F : 0.0F;
-      });
+      this.func_185043_a(new ResourceLocation("throwing"), new IItemPropertyGetter() {});
    }
 
    public boolean func_195938_a(IBlockState p_195938_1_, World p_195938_2_, BlockPos p_195938_3_, EntityPlayer p_195938_4_) {
