--- a/net/minecraft/nbt/visitors/CollectFields.java
+++ b/net/minecraft/nbt/visitors/CollectFields.java
@@ -80,9 +80,6 @@
    }
 
    static record StackFrame(int f_197616_, Map<String, TagType<?>> f_197617_, Map<String, CollectFields.StackFrame> f_197618_) {
-      final Map<String, TagType<?>> f_197617_;
-      final Map<String, CollectFields.StackFrame> f_197618_;
-
       public StackFrame(int p_197620_) {
          this(p_197620_, new HashMap<>(), new HashMap<>());
       }
@@ -100,10 +97,6 @@
    }
 
    public static record WantedField(List<String> f_197636_, TagType<?> f_197637_, String f_197638_) {
-      final List<String> f_197636_;
-      final TagType<?> f_197637_;
-      final String f_197638_;
-
       public WantedField(TagType<?> p_197653_, String p_197654_) {
          this(List.of(), p_197653_, p_197654_);
       }
