--- a/net/minecraft/world/level/ChunkTickList.java
+++ b/net/minecraft/world/level/ChunkTickList.java
@@ -15,7 +15,7 @@
 
    public ChunkTickList(Function<T, ResourceLocation> p_45637_, List<TickNextTickData<T>> p_45638_, long p_45639_) {
       this(p_45637_, p_45638_.stream().map((p_45642_) -> {
-         return new ChunkTickList.ScheduledTick(p_45642_.m_47340_(), p_45642_.f_47323_, (int)(p_45642_.f_47324_ - p_45639_), p_45642_.f_47325_);
+         return new ChunkTickList.ScheduledTick<>(p_45642_.m_47340_(), p_45642_.f_47323_, (int)(p_45642_.f_47324_ - p_45639_), p_45642_.f_47325_);
       }).collect(Collectors.toList()));
    }
 
