--- a/com/mojang/blaze3d/platform/MacosUtil.java
+++ b/com/mojang/blaze3d/platform/MacosUtil.java
@@ -25,7 +25,7 @@
    }
 
    private static boolean m_182519_(NSObject p_182520_) {
-      return (p_182520_.sendRaw("styleMask", new Object[0]) & 16384L) == 16384L;
+      return (((Number)p_182520_.sendRaw("styleMask", new Object[0])).longValue() & 16384L) == 16384L;
    }
 
    private static void m_182523_(NSObject p_182524_) {
