--- a/net/minecraft/client/PeriodicNotificationManager.java
+++ b/net/minecraft/client/PeriodicNotificationManager.java
@@ -31,7 +31,7 @@
 import org.slf4j.Logger;
 
 public class PeriodicNotificationManager extends SimplePreparableReloadListener<Map<String, List<PeriodicNotificationManager.Notification>>> implements AutoCloseable {
-   private static final Codec<Map<String, List<PeriodicNotificationManager.Notification>>> f_205285_ = Codec.unboundedMap(Codec.STRING, RecordCodecBuilder.create((p_205303_) -> {
+   private static final Codec<Map<String, List<PeriodicNotificationManager.Notification>>> f_205285_ = Codec.unboundedMap(Codec.STRING, RecordCodecBuilder.<PeriodicNotificationManager.Notification>create((p_205303_) -> {
       return p_205303_.group(Codec.LONG.optionalFieldOf("delay", Long.valueOf(0L)).forGetter(PeriodicNotificationManager.Notification::f_205328_), Codec.LONG.fieldOf("period").forGetter(PeriodicNotificationManager.Notification::f_205329_), Codec.STRING.fieldOf("title").forGetter(PeriodicNotificationManager.Notification::f_205330_), Codec.STRING.fieldOf("message").forGetter(PeriodicNotificationManager.Notification::f_205331_)).apply(p_205303_, PeriodicNotificationManager.Notification::new);
    }).listOf());
    private static final Logger f_205286_ = LogUtils.getLogger();
