--- a/net/minecraft/network/rcon/ClientThread.java
+++ b/net/minecraft/network/rcon/ClientThread.java
@@ -32,8 +32,8 @@
    }
 
    public void run() {
-      while(true) {
          try {
+         while(true) {
             if (!this.field_72619_a) {
                return;
             }
@@ -84,6 +84,8 @@
                   continue;
                }
             }
+            return;
+         }
          } catch (IOException ioexception) {
             return;
          } catch (Exception exception1) {
@@ -94,10 +96,7 @@
             field_164005_h.info("Thread {} shutting down", (Object)this.field_164003_c);
             this.field_72619_a = false;
          }
-
-         return;
       }
-   }
 
    private void func_72654_a(int p_72654_1_, int p_72654_2_, String p_72654_3_) throws IOException {
       ByteArrayOutputStream bytearrayoutputstream = new ByteArrayOutputStream(1248);
