diff -r -U 3 minecraft\net\minecraft\client\Minecraft.java minecraft_patched\net\minecraft\client\Minecraft.java
--- minecraft\net\minecraft\client\Minecraft.java
+++ minecraft_patched\net\minecraft\client\Minecraft.java
@@ -231,7 +231,7 @@
    public GuiScreen field_71462_r;
    public EntityRenderer field_71460_t;
    public DebugRenderer field_184132_p;
-   int field_71429_W;
+   public int field_71429_W;
    @Nullable
    private IntegratedServer field_71437_Z;
    public GuiIngame field_71456_v;
@@ -284,7 +284,7 @@
    private long field_71419_L = func_71386_F();
    private int field_71420_M;
    private final Tutorial field_193035_aW;
-   boolean field_195555_I;
+   public boolean field_195555_I;
    private ResourceLocation field_195552_aP;
    private DynamicTexture field_195553_aQ;
    long field_71421_N = -1L;
@@ -338,9 +338,7 @@
 
       while(true) {
          try {
-            if (!this.field_71425_J) {
-               break;
-            }
+            while(this.field_71425_J) {
 
             if (!this.field_71434_R || this.field_71433_S == null) {
                try {
@@ -350,10 +348,12 @@
                   this.func_147108_a(new GuiMemoryErrorScreen());
                   System.gc();
                }
-               continue;
-            }
-
+            }
+            else
+            {
             this.func_71377_b(this.field_71433_S);
+            }
+            }
          } catch (MinecraftError var12) {
             break;
          } catch (ReportedException reportedexception) {
@@ -1004,6 +1004,7 @@
       }
    }
 
+   @SuppressWarnings("incomplete-switch")
    private void func_147121_ag() {
       if (!this.field_71442_b.func_181040_m()) {
          this.field_71467_ac = 4;
