diff -r -U 3 minecraft\net\minecraft\entity\player\EntityPlayer.java minecraft_patched\net\minecraft\entity\player\EntityPlayer.java
--- minecraft\net\minecraft\entity\player\EntityPlayer.java
+++ minecraft_patched\net\minecraft\entity\player\EntityPlayer.java
@@ -91,6 +91,7 @@
 import net.minecraft.world.World;
 import net.minecraft.world.WorldServer;
 
+@SuppressWarnings("incomplete-switch")
 public abstract class EntityPlayer extends EntityLivingBase {
    private static final DataParameter<Float> field_184829_a = EntityDataManager.<Float>func_187226_a(EntityPlayer.class, DataSerializers.field_187193_c);
    private static final DataParameter<Integer> field_184830_b = EntityDataManager.<Integer>func_187226_a(EntityPlayer.class, DataSerializers.field_187192_b);
