diff -r -U 3 minecraft\net\minecraft\block\state\BlockStateBase.java minecraft_patched\net\minecraft\block\state\BlockStateBase.java
--- minecraft\net\minecraft\block\state\BlockStateBase.java
+++ minecraft_patched\net\minecraft\block\state\BlockStateBase.java
@@ -24,12 +24,12 @@
       }
 
       private <T extends Comparable<T>> String func_185886_a(IProperty<T> p_185886_1_, Comparable<?> p_185886_2_) {
-         return p_185886_1_.func_177702_a(p_185886_2_);
+         return p_185886_1_.func_177702_a((T)p_185886_2_);
       }
    };
 
    public <T extends Comparable<T>> IBlockState func_177231_a(IProperty<T> p_177231_1_) {
-      return this.func_177226_a(p_177231_1_, (Comparable)func_177232_a(p_177231_1_.func_177700_c(), this.func_177229_b(p_177231_1_)));
+      return this.func_177226_a(p_177231_1_, func_177232_a(p_177231_1_.func_177700_c(), this.func_177229_b(p_177231_1_)));
    }
 
    protected static <T> T func_177232_a(Collection<T> p_177232_0_, T p_177232_1_) {
