diff -r -U 3 minecraft\net\minecraft\block\BlockFurnace.java minecraft_patched\net\minecraft\block\BlockFurnace.java
--- minecraft\net\minecraft\block\BlockFurnace.java
+++ minecraft_patched\net\minecraft\block\BlockFurnace.java
@@ -67,6 +67,7 @@
       }
    }
 
+   @SuppressWarnings("incomplete-switch")
    public void func_180655_c(IBlockState p_180655_1_, World p_180655_2_, BlockPos p_180655_3_, Random p_180655_4_) {
       if (this.field_149932_b) {
          EnumFacing enumfacing = (EnumFacing)p_180655_1_.func_177229_b(field_176447_a);
