diff -r -U 3 minecraft\net\minecraft\block\BlockRail.java minecraft_patched\net\minecraft\block\BlockRail.java
--- minecraft\net\minecraft\block\BlockRail.java
+++ minecraft_patched\net\minecraft\block\BlockRail.java
@@ -31,6 +31,7 @@
       return field_176565_b;
    }
 
+   @SuppressWarnings("incomplete-switch")
    public IBlockState func_185499_a(IBlockState p_185499_1_, Rotation p_185499_2_) {
       switch(p_185499_2_) {
       case CLOCKWISE_180:
@@ -103,6 +104,7 @@
       }
    }
 
+   @SuppressWarnings("incomplete-switch")
    public IBlockState func_185471_a(IBlockState p_185471_1_, Mirror p_185471_2_) {
       BlockRailBase.EnumRailDirection blockrailbase$enumraildirection = (BlockRailBase.EnumRailDirection)p_185471_1_.func_177229_b(field_176565_b);
       switch(p_185471_2_) {
