--- a/net/minecraft/world/chunk/ChunkTaskPriorityQueueSorter.java
+++ b/net/minecraft/world/chunk/ChunkTaskPriorityQueueSorter.java
@@ -54,8 +54,8 @@
    }
 
    public <T> ITaskExecutor<ChunkTaskPriorityQueueSorter.FunctionEntry<T>> func_219087_a(ITaskExecutor<T> p_219087_1_, boolean p_219087_2_) {
-      return (ITaskExecutor)this.field_219096_d.<ChunkTaskPriorityQueueSorter.FunctionEntry<T>>func_213141_a((p_219086_3_) -> {
-         return (T)(new ITaskQueue.RunnableWithPriority(0, () -> {
+      return this.field_219096_d.<ITaskExecutor<ChunkTaskPriorityQueueSorter.FunctionEntry<T>>>func_213141_a((p_219086_3_) -> {
+         return (new ITaskQueue.RunnableWithPriority(0, () -> {
             this.func_219068_b(p_219087_1_);
             p_219086_3_.func_212871_a_(ITaskExecutor.func_213140_a("chunk priority sorter around " + p_219087_1_.func_213142_bd(), (p_219071_3_) -> {
                this.func_219067_a(p_219087_1_, p_219071_3_.field_219428_a, p_219071_3_.field_219429_b, p_219071_3_.field_219430_c, p_219087_2_);
@@ -65,7 +65,7 @@
    }
 
    public ITaskExecutor<ChunkTaskPriorityQueueSorter.RunnableEntry> func_219091_a(ITaskExecutor<Runnable> p_219091_1_) {
-      return (ITaskExecutor)this.field_219096_d.<ChunkTaskPriorityQueueSorter.RunnableEntry>func_213141_a((p_219080_2_) -> {
+      return this.field_219096_d.<ITaskExecutor<ChunkTaskPriorityQueueSorter.RunnableEntry>>func_213141_a((p_219080_2_) -> {
          return new ITaskQueue.RunnableWithPriority(0, () -> {
             p_219080_2_.func_212871_a_(ITaskExecutor.func_213140_a("chunk priority sorter around " + p_219091_1_.func_213142_bd(), (p_219075_2_) -> {
                this.func_219074_a(p_219091_1_, p_219075_2_.field_219435_b, p_219075_2_.field_219434_a, p_219075_2_.field_219436_c);
@@ -136,7 +136,7 @@
       if (chunktaskpriorityqueue == null) {
          throw new IllegalArgumentException("No queue for: " + p_219068_1_);
       } else {
-         return chunktaskpriorityqueue;
+         return (ChunkTaskPriorityQueue<Function<ITaskExecutor<Unit>, T>>)chunktaskpriorityqueue;
       }
    }
 
