--- a/net/minecraft/nbt/SnbtGrammar.java
+++ b/net/minecraft/nbt/SnbtGrammar.java
@@ -229,12 +229,8 @@
             String s = stringbuilder.toString();
 
             return (T)(switch (p_392491_) {
-                case Object object1 -> {
-                }
-                case Object object -> {
-                }
-                case Object object2 -> {
-                }
+                case FLOAT -> m_386150_(p_393782_, p_395596_, s);
+                case null, DOUBLE -> m_388734_(p_393782_, p_395596_, s);
                 default -> {
                     p_395596_.m_323339_().m_323756_(p_395596_.m_320129_(), f_379767_);
                     yield null;
