--- a/net/minecraft/world/level/levelgen/blending/BlendingData.java
+++ b/net/minecraft/world/level/levelgen/blending/BlendingData.java
@@ -54,7 +54,7 @@
    private final List<List<Holder<Biome>>> f_202276_;
    private final transient double[][] f_190270_;
    private static final Codec<double[]> f_190272_ = Codec.DOUBLE.listOf().xmap(Doubles::toArray, Doubles::asList);
-   public static final Codec<BlendingData> f_190254_ = RecordCodecBuilder.create((p_219131_) -> {
+   public static final Codec<BlendingData> f_190254_ = RecordCodecBuilder.<BlendingData>create((p_219131_) -> {
       return p_219131_.group(Codec.INT.fieldOf("min_section").forGetter((p_219129_) -> {
          return p_219129_.f_215026_.m_151560_();
       }), Codec.INT.fieldOf("max_section").forGetter((p_219127_) -> {
