--- a/net/minecraft/server/level/ChunkMap.java
+++ b/net/minecraft/server/level/ChunkMap.java
@@ -568,7 +568,7 @@
 
             return flag;
          });
-      }).thenApplyAsync((p_282692_) -> {
+      }).<Either<ChunkAccess, ChunkHolder.ChunkLoadingFailure>>thenApplyAsync((p_282692_) -> {
          this.f_140133_.m_46473_().m_6174_("chunkLoad");
          if (p_282692_.isPresent()) {
             ChunkAccess chunkaccess = ChunkSerializer.m_188230_(this.f_140133_, this.f_140138_, p_140418_, p_282692_.get());
