--- a/net/minecraft/client/gui/ScreenManager.java
+++ b/net/minecraft/client/gui/ScreenManager.java
@@ -50,7 +50,7 @@
 
    @Nullable
    private static <T extends Container> ScreenManager.IScreenFactory<T, ?> func_216912_a(ContainerType<T> p_216912_0_) {
-      return field_216914_b.get(p_216912_0_);
+      return (ScreenManager.IScreenFactory<T, ?>)field_216914_b.get(p_216912_0_);
    }
 
    private static <M extends Container, U extends Screen & IHasContainer<M>> void func_216911_a(ContainerType<? extends M> p_216911_0_, ScreenManager.IScreenFactory<M, U> p_216911_1_) {
