--- a/net/minecraft/util/CubicSpline.java
+++ b/net/minecraft/util/CubicSpline.java
@@ -19,11 +19,11 @@
    String m_183628_();
 
    static <C> Codec<CubicSpline<C>> m_184262_(Codec<ToFloatFunction<C>> p_184263_) {
+      record Point<C>(float f_184273_, CubicSpline<C> f_184274_, float f_184275_) {
+      }
       MutableObject<Codec<CubicSpline<C>>> mutableobject = new MutableObject<>();
       Codec<Point<C>> codec = RecordCodecBuilder.create((p_184270_) -> {
          return p_184270_.group(Codec.FLOAT.fieldOf("location").forGetter(Point::f_184273_), ExtraCodecs.m_184415_(mutableobject::getValue).fieldOf("value").forGetter(Point::f_184274_), Codec.FLOAT.fieldOf("derivative").forGetter(Point::f_184275_)).apply(p_184270_, (p_184242_, p_184243_, p_184244_) -> {
-            record Point(float f_184273_, CubicSpline<C> f_184274_, float f_184275_) {
-            }
 
             return new Point((float)p_184242_, p_184243_, (float)p_184244_);
          });
@@ -31,7 +31,7 @@
       Codec<CubicSpline.Multipoint<C>> codec1 = RecordCodecBuilder.create((p_184267_) -> {
          return p_184267_.group(p_184263_.fieldOf("coordinate").forGetter(CubicSpline.Multipoint::f_184319_), codec.listOf().fieldOf("points").forGetter((p_184272_) -> {
             return IntStream.range(0, p_184272_.f_184320_.length).mapToObj((p_184249_) -> {
-               return new Point(p_184272_.f_184320_()[p_184249_], (CubicSpline)p_184272_.f_184321_().get(p_184249_), p_184272_.f_184322_()[p_184249_]);
+               return new Point<C>(p_184272_.f_184320_()[p_184249_], (CubicSpline)p_184272_.f_184321_().get(p_184249_), p_184272_.f_184322_()[p_184249_]);
             }).toList();
          })).apply(p_184267_, (p_184258_, p_184259_) -> {
             float[] afloat = new float[p_184259_.size()];
@@ -55,7 +55,7 @@
       }, (p_184251_) -> {
          Either either;
          if (p_184251_ instanceof CubicSpline.Constant) {
-            CubicSpline.Constant<C> constant = p_184251_;
+            CubicSpline.Constant<C> constant = (Constant<C>) p_184251_;
             either = Either.left(constant.f_184308_());
          } else {
             either = Either.right(p_184251_);
@@ -132,7 +132,7 @@
    }
 
    @VisibleForDebug
-   public static record Multipoint(ToFloatFunction<C> f_184319_, float[] f_184320_, List<CubicSpline<C>> f_184321_, float[] f_184322_) implements CubicSpline {
+   public static record Multipoint<C>(ToFloatFunction<C> f_184319_, float[] f_184320_, List<CubicSpline<C>> f_184321_, float[] f_184322_) implements CubicSpline<C> {
       public Multipoint(ToFloatFunction<C> f_184319_, float[] f_184320_, List<CubicSpline<C>> f_184321_, float[] f_184322_) {
          if (f_184320_.length == f_184321_.size() && f_184320_.length == f_184322_.length) {
             this.f_184319_ = f_184319_;
