--- a/net/minecraft/world/level/levelgen/feature/configurations/StructureFeatureConfiguration.java
+++ b/net/minecraft/world/level/levelgen/feature/configurations/StructureFeatureConfiguration.java
@@ -6,7 +6,7 @@
 import java.util.function.Function;
 
 public class StructureFeatureConfiguration {
-   public static final Codec<StructureFeatureConfiguration> f_68162_ = RecordCodecBuilder.create((p_68175_) -> {
+   public static final Codec<StructureFeatureConfiguration> f_68162_ = RecordCodecBuilder.<StructureFeatureConfiguration>create((p_68175_) -> {
       return p_68175_.group(Codec.intRange(0, 4096).fieldOf("spacing").forGetter((p_68183_) -> {
          return p_68183_.f_68163_;
       }), Codec.intRange(0, 4096).fieldOf("separation").forGetter((p_68181_) -> {
