--- a/net/minecraft/world/item/component/PiercingWeapon.java
+++ b/net/minecraft/world/item/component/PiercingWeapon.java
@@ -83,7 +83,7 @@
         for (EntityHitResult entityhitresult : ProjectileUtil.getHitEntitiesAlong(
                 attacker, attackrange, e1 -> canHitEntity(attacker, e1), ClipContext.Block.COLLIDER
             )
-            .map(a -> List.of(), e -> (Collection)e)) {
+            .map(a -> List.<EntityHitResult>of(), e -> (Collection<EntityHitResult>)e)) {
             flag |= attacker.stabAttack(hand, entityhitresult.getEntity(), f, true, this.dealsKnockback, this.dismounts);
         }
 
