--- a/net/minecraft/advancements/criterion/MinMaxBounds.java
+++ b/net/minecraft/advancements/criterion/MinMaxBounds.java
@@ -75,7 +75,7 @@
         }
 
         public static <T extends Number & Comparable<T>> MinMaxBounds.Bounds<T> any() {
-            return new MinMaxBounds.Bounds<>(Optional.empty(), Optional.empty());
+            return new MinMaxBounds.Bounds<T>(Optional.empty(), Optional.empty());
         }
 
         public static <T extends Number & Comparable<T>> MinMaxBounds.Bounds<T> exactly(final T value) {
