--- a/net/minecraft/commands/synchronization/SuggestionProviders.java
+++ b/net/minecraft/commands/synchronization/SuggestionProviders.java
@@ -42,7 +42,7 @@
          throw new IllegalArgumentException("A command suggestion provider is already registered with the name " + p_121659_);
       } else {
          f_121646_.put(p_121659_, p_121660_);
-         return new SuggestionProviders.Wrapper(p_121659_, p_121660_);
+         return (SuggestionProvider<S>)new SuggestionProviders.Wrapper(p_121659_, p_121660_);
       }
    }
 
