--- a/net/minecraft/client/renderer/entity/RenderLivingBase.java
+++ b/net/minecraft/client/renderer/entity/RenderLivingBase.java
@@ -49,7 +49,7 @@
    }
 
    protected <V extends EntityLivingBase, U extends LayerRenderer<V>> boolean func_177094_a(U p_177094_1_) {
-      return this.field_177097_h.add(p_177094_1_);
+      return this.field_177097_h.add((LayerRenderer<T>)p_177094_1_);
    }
 
    public ModelBase func_177087_b() {
