--- a/net/minecraft/core/component/TypedDataComponent.java
+++ b/net/minecraft/core/component/TypedDataComponent.java
@@ -11,7 +11,7 @@
     public static final StreamCodec<RegistryFriendlyByteBuf, TypedDataComponent<?>> STREAM_CODEC = new StreamCodec<RegistryFriendlyByteBuf, TypedDataComponent<?>>() {
         public TypedDataComponent<?> decode(final RegistryFriendlyByteBuf input) {
             DataComponentType<?> type = DataComponentType.STREAM_CODEC.decode(input);
-            return decodeTyped(input, (DataComponentType<T>)type);
+            return decodeTyped(input, (DataComponentType)type);
         }
 
         private static <T> TypedDataComponent<T> decodeTyped(final RegistryFriendlyByteBuf input, final DataComponentType<T> type) {
@@ -19,7 +19,7 @@
         }
 
         public void encode(final RegistryFriendlyByteBuf output, final TypedDataComponent<?> value) {
-            encodeCap(output, (TypedDataComponent<T>)value);
+            encodeCap(output, (TypedDataComponent)value);
         }
 
         private static <T> void encodeCap(final RegistryFriendlyByteBuf output, final TypedDataComponent<T> component) {
