--- a/net/minecraft/client/gui/screens/PresetFlatWorldScreen.java
+++ b/net/minecraft/client/gui/screens/PresetFlatWorldScreen.java
@@ -266,7 +266,7 @@
 
          public Entry(Holder<FlatLevelGeneratorPreset> p_222403_) {
             this.f_169357_ = p_222403_.m_203334_();
-            this.f_222400_ = p_222403_.m_203543_().map((p_222405_) -> {
+            this.f_222400_ = p_222403_.m_203543_().<Component>map((p_222405_) -> {
                return Component.m_239796_(p_222405_.m_135782_().m_224270_("flat_world_preset"));
             }).orElse(PresetFlatWorldScreen.f_222396_);
          }
