--- a/net/minecraft/world/entity/monster/Zoglin.java
+++ b/net/minecraft/world/entity/monster/Zoglin.java
@@ -87,11 +87,11 @@
    }
 
    private static void m_34228_(Brain<Zoglin> p_34229_) {
-      p_34229_.m_21891_(Activity.f_37979_, 10, ImmutableList.of(new StartAttacking<>(Zoglin::m_34251_), new RunSometimes(new SetEntityLookTarget(8.0F), UniformInt.m_146622_(30, 60)), new RunOne(ImmutableList.of(Pair.of(new RandomStroll(0.4F), 2), Pair.of(new SetWalkTargetFromLookTarget(0.4F, 3), 2), Pair.of(new DoNothing(30, 60), 1)))));
+      p_34229_.m_21891_(Activity.f_37979_, 10, ImmutableList.of(new StartAttacking<Zoglin>(Zoglin::m_34251_), new RunSometimes<Zoglin>(new SetEntityLookTarget(8.0F), UniformInt.m_146622_(30, 60)), new RunOne<Zoglin>(ImmutableList.of(Pair.of(new RandomStroll(0.4F), 2), Pair.of(new SetWalkTargetFromLookTarget(0.4F, 3), 2), Pair.of(new DoNothing(30, 60), 1)))));
    }
 
    private static void m_34236_(Brain<Zoglin> p_34237_) {
-      p_34237_.m_21895_(Activity.f_37988_, 10, ImmutableList.of(new SetWalkTargetFromAttackTargetIfTargetOutOfReach(1.0F), new RunIf<>(Zoglin::m_34247_, new MeleeAttack(40)), new RunIf<>(Zoglin::m_6162_, new MeleeAttack(15)), new StopAttackingIfTargetInvalid()), MemoryModuleType.f_26372_);
+      p_34237_.m_21895_(Activity.f_37988_, 10, ImmutableList.<net.minecraft.world.entity.ai.behavior.Behavior<net.minecraft.world.entity.Mob>>of(new SetWalkTargetFromAttackTargetIfTargetOutOfReach(1.0F), (net.minecraft.world.entity.ai.behavior.Behavior<net.minecraft.world.entity.Mob>)(net.minecraft.world.entity.ai.behavior.Behavior)new RunIf<Zoglin>(Zoglin::m_34247_, new MeleeAttack(40)), (net.minecraft.world.entity.ai.behavior.Behavior<net.minecraft.world.entity.Mob>)(net.minecraft.world.entity.ai.behavior.Behavior)new RunIf<Zoglin>(Zoglin::m_6162_, new MeleeAttack(15)), new StopAttackingIfTargetInvalid()), MemoryModuleType.f_26372_);
    }
 
    private Optional<? extends LivingEntity> m_34251_() {
@@ -172,7 +172,7 @@
    }
 
    public Brain<Zoglin> m_6274_() {
-      return super.m_6274_();
+      return (Brain<Zoglin>)super.m_6274_();
    }
 
    protected void m_34248_() {
