--- a/net/minecraft/resources/RegistryReadOps.java
+++ b/net/minecraft/resources/RegistryReadOps.java
@@ -60,7 +60,7 @@
       this.f_135635_ = p_135642_;
       this.f_135636_ = p_135643_;
       this.f_135637_ = p_135644_;
-      this.f_135638_ = p_135641_ == JsonOps.INSTANCE ? this : new RegistryReadOps<>(JsonOps.INSTANCE, p_135642_, p_135643_, p_135644_);
+      this.f_135638_ = p_135641_ == JsonOps.INSTANCE ? (RegistryReadOps<JsonElement>)this : new RegistryReadOps<>(JsonOps.INSTANCE, p_135642_, p_135643_, p_135644_);
    }
 
    protected <E> DataResult<Pair<Supplier<E>, T>> m_135677_(T p_135678_, ResourceKey<? extends Registry<E>> p_135679_, Codec<E> p_135680_, boolean p_135681_) {
@@ -155,13 +155,13 @@
    }
 
    private <E> RegistryReadOps.ReadCache<E> m_135699_(ResourceKey<? extends Registry<E>> p_135700_) {
-      return this.f_135637_.computeIfAbsent(p_135700_, (p_135707_) -> {
+      return (RegistryReadOps.ReadCache<E>)this.f_135637_.computeIfAbsent(p_135700_, (p_135707_) -> {
          return new RegistryReadOps.ReadCache();
       });
    }
 
    protected <E> DataResult<Registry<E>> m_135682_(ResourceKey<? extends Registry<E>> p_135683_) {
-      return this.f_135636_.m_6632_(p_135683_).map((p_135667_) -> {
+      return (DataResult)this.f_135636_.m_6632_(p_135683_).map((p_135667_) -> {
          return DataResult.success(p_135667_, p_135667_.m_7837_());
       }).orElseGet(() -> {
          return DataResult.error("Unknown registry: " + p_135683_);
