--- a/net/minecraft/util/StringRepresentable.java
+++ b/net/minecraft/util/StringRepresentable.java
@@ -19,7 +19,7 @@
    String m_7912_();
 
    static <E extends Enum<E> & StringRepresentable> StringRepresentable.EnumCodec<E> m_236045_(Supplier<E[]> p_236046_) {
-      E[] ae = (Enum[])p_236046_.get();
+      E[] ae = p_236046_.get();
       if (ae.length > 16) {
          Map<String, E> map = Arrays.stream(ae).collect(Collectors.toMap((p_184753_) -> {
             return p_184753_.m_7912_();
