--- a/net/minecraft/advancements/criterion/EntityPredicate.java
+++ b/net/minecraft/advancements/criterion/EntityPredicate.java
@@ -164,7 +164,7 @@
    }
 
    static {
-      field_192483_a = new EntityPredicate(EntityTypePredicate.field_209371_a, DistancePredicate.field_193423_a, LocationPredicate.field_193455_a, MobEffectsPredicate.field_193473_a, NBTPredicate.field_193479_a, EntityFlagsPredicate.field_217979_a, EntityEquipmentPredicate.field_217958_a, PlayerPredicate.field_226989_a_, FishingPredicate.field_234635_a_, field_192483_a, field_192483_a, (String)null, (ResourceLocation)null);
+      field_192483_a = new EntityPredicate(EntityTypePredicate.field_209371_a, DistancePredicate.field_193423_a, LocationPredicate.field_193455_a, MobEffectsPredicate.field_193473_a, NBTPredicate.field_193479_a, EntityFlagsPredicate.field_217979_a, EntityEquipmentPredicate.field_217958_a, PlayerPredicate.field_226989_a_, FishingPredicate.field_234635_a_, EntityPredicate.field_192483_a, EntityPredicate.field_192483_a, (String)null, (ResourceLocation)null);
    }
 
    public static class AndPredicate {
