--- a/net/minecraft/util/datafix/fixes/AddBedTileEntity.java
+++ b/net/minecraft/util/datafix/fixes/AddBedTileEntity.java
@@ -41,7 +41,7 @@
       int i = 416;
       return TypeRewriteRule.seq(this.fixTypeEverywhere("InjectBedBlockEntityType", this.getInputSchema().findChoiceType(TypeReferences.field_211294_j), this.getOutputSchema().findChoiceType(TypeReferences.field_211294_j), (p_212021_0_) -> {
          return (p_209696_0_) -> {
-            return p_209696_0_;
+            return (com.mojang.datafixers.util.Pair)p_209696_0_;
          };
       }), this.fixTypeEverywhereTyped("BedBlockEntityInjecter", this.getOutputSchema().getType(TypeReferences.field_211287_c), (p_207434_3_) -> {
          Typed<?> typed = p_207434_3_.getTyped(opticfinder);
@@ -59,7 +59,7 @@
             });
             int j1 = 0;
 
-            for(int k1 : stream::iterator) {
+            for(int k1 : (Iterable<Integer>)stream::iterator) {
                if (416 == (k1 & 255) << 4) {
                   int l1 = j1 & 15;
                   int i2 = j1 >> 8 & 15;
