--- a/net/minecraft/util/datafix/fixes/BlockStateFlatternEntities.java
+++ b/net/minecraft/util/datafix/fixes/BlockStateFlatternEntities.java
@@ -329,7 +329,7 @@
       Dynamic<?> dynamic = (Dynamic)p_211442_1_.get(DSL.remainderFinder());
       return p_211442_1_.update(type.finder(), type1, (p_211437_1_) -> {
          int i = p_211437_1_.map((p_211440_0_) -> {
-            return (Integer)((Either)p_211440_0_.getSecond()).map((p_211436_0_) -> {
+            return (Integer)(p_211440_0_.getSecond()).map((p_211436_0_) -> {
                return p_211436_0_;
             }, BlockStateFlatternEntities::func_199171_a);
          }, (p_211441_1_) -> {
@@ -348,7 +348,7 @@
       Type<Pair<String, Dynamic<?>>> type1 = DSL.field(p_211434_4_, DSL.named(TypeReferences.field_211296_l.typeName(), DSL.remainderType()));
       Dynamic<?> dynamic = (Dynamic)p_211434_1_.getOrCreate(DSL.remainderFinder());
       return p_211434_1_.update(type.finder(), type1, (p_211432_2_) -> {
-         int i = ((Either)p_211432_2_.getSecond()).map((p_211435_0_) -> {
+         int i = (p_211432_2_.getSecond()).map((p_211435_0_) -> {
             return p_211435_0_;
          }, BlockStateFlatternEntities::func_199171_a);
          int j = dynamic.getInt(p_211434_3_) & 15;
