--- a/net/minecraft/world/level/chunk/LevelChunk.java
+++ b/net/minecraft/world/level/chunk/LevelChunk.java
@@ -730,34 +730,34 @@
 
    public void m_62813_() {
       if (this.f_62784_ instanceof ProtoTickList) {
-         ((ProtoTickList)this.f_62784_).m_63305_(this.f_62776_.m_6219_(), (p_62941_) -> {
+         ((ProtoTickList<Block>)this.f_62784_).m_63305_(this.f_62776_.m_6219_(), (p_62941_) -> {
             return this.m_8055_(p_62941_).m_60734_();
          });
          this.f_62784_ = EmptyTickList.m_45888_();
       } else if (this.f_62784_ instanceof ChunkTickList) {
-         ((ChunkTickList)this.f_62784_).m_45643_(this.f_62776_.m_6219_());
+         ((ChunkTickList<Block>)this.f_62784_).m_45643_(this.f_62776_.m_6219_());
          this.f_62784_ = EmptyTickList.m_45888_();
       }
 
       if (this.f_62785_ instanceof ProtoTickList) {
-         ((ProtoTickList)this.f_62785_).m_63305_(this.f_62776_.m_6217_(), (p_62938_) -> {
+         ((ProtoTickList<Fluid>)this.f_62785_).m_63305_(this.f_62776_.m_6217_(), (p_62938_) -> {
             return this.m_6425_(p_62938_).m_76152_();
          });
          this.f_62785_ = EmptyTickList.m_45888_();
       } else if (this.f_62785_ instanceof ChunkTickList) {
-         ((ChunkTickList)this.f_62785_).m_45643_(this.f_62776_.m_6217_());
+         ((ChunkTickList<Fluid>)this.f_62785_).m_45643_(this.f_62776_.m_6217_());
          this.f_62785_ = EmptyTickList.m_45888_();
       }
 
    }
 
    public void m_62823_(ServerLevel p_62824_) {
-      if (this.f_62784_ == EmptyTickList.m_45888_()) {
+      if (this.f_62784_ == EmptyTickList.<Block>m_45888_()) {
          this.f_62784_ = new ChunkTickList<>(Registry.f_122824_::m_7981_, p_62824_.m_6219_().m_47223_(this.f_62792_, true, false), p_62824_.m_46467_());
          this.m_8092_(true);
       }
 
-      if (this.f_62785_ == EmptyTickList.m_45888_()) {
+      if (this.f_62785_ == EmptyTickList.<Fluid>m_45888_()) {
          this.f_62785_ = new ChunkTickList<>(Registry.f_122822_::m_7981_, p_62824_.m_6217_().m_47223_(this.f_62792_, true, false), p_62824_.m_46467_());
          this.m_8092_(true);
       }
