--- a/net\minecraft\client\MouseHelper.java
+++ b/net\minecraft\client\MouseHelper.java
@@ -73,13 +73,14 @@
          } else {
             double d0 = this.field_198040_e * (double)this.field_198036_a.field_195558_d.func_198107_o() / (double)this.field_198036_a.field_195558_d.func_198105_m();
             double d1 = this.field_198041_f * (double)this.field_198036_a.field_195558_d.func_198087_p() / (double)this.field_198036_a.field_195558_d.func_198083_n();
+            final int pFinal =  p_198023_3_;
             if (flag) {
                GuiScreen.func_195121_a(() -> {
-                  aboolean[0] = this.field_198036_a.field_71462_r.mouseClicked(d0, d1, p_198023_3_);
+                  aboolean[0] = this.field_198036_a.field_71462_r.mouseClicked(d0, d1, pFinal);
                }, "mouseClicked event handler", this.field_198036_a.field_71462_r.getClass().getCanonicalName());
             } else {
                GuiScreen.func_195121_a(() -> {
-                  aboolean[0] = this.field_198036_a.field_71462_r.mouseReleased(d0, d1, p_198023_3_);
+                  aboolean[0] = this.field_198036_a.field_71462_r.mouseReleased(d0, d1, pFinal);
                }, "mouseReleased event handler", this.field_198036_a.field_71462_r.getClass().getCanonicalName());
             }
          }
