--- a/com/mojang/realmsclient/gui/screens/RealmsSelectWorldTemplateScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsSelectWorldTemplateScreen.java
@@ -76,10 +76,10 @@
         this.f_167479_ = p_167486_;
         this.f_89598_ = p_167487_;
         if (p_167488_ == null) {
-            this.f_302619_ = new RealmsSelectWorldTemplateScreen.WorldTemplateList(this);
+            this.f_302619_ = new RealmsSelectWorldTemplateScreen.WorldTemplateList();
             this.m_89653_(new WorldTemplatePaginatedList(10));
         } else {
-            this.f_302619_ = new RealmsSelectWorldTemplateScreen.WorldTemplateList(this, Lists.newArrayList(p_167488_.f_87753_));
+            this.f_302619_ = new RealmsSelectWorldTemplateScreen.WorldTemplateList(Lists.newArrayList(p_167488_.f_87753_));
             this.m_89653_(p_167488_);
         }
     }
@@ -91,7 +91,7 @@
     @Override
     public void m_7856_() {
         this.f_291318_.m_324480_(this.f_96539_, this.f_96547_);
-        this.f_302619_ = this.f_291318_.m_268999_(new RealmsSelectWorldTemplateScreen.WorldTemplateList(this, this.f_302619_.m_89818_()));
+        this.f_302619_ = this.f_291318_.m_268999_(new RealmsSelectWorldTemplateScreen.WorldTemplateList(this.f_302619_.m_89818_()));
         LinearLayout linearlayout = this.f_291318_.m_269281_(LinearLayout.m_295847_().m_294554_(10));
         linearlayout.m_294823_().m_264356_();
         this.f_89616_ = linearlayout.m_264406_(Button.m_253074_(f_291283_, p_89701_ -> this.m_89738_()).m_252780_(100).m_253136_());
@@ -368,24 +368,23 @@
 
     @OnlyIn(Dist.CLIENT)
     class WorldTemplateList extends ObjectSelectionList<RealmsSelectWorldTemplateScreen.Entry> {
-        public WorldTemplateList(final RealmsSelectWorldTemplateScreen param1) {
-            this(var1, Collections.emptyList());
+        public WorldTemplateList() {
+            this(Collections.emptyList());
         }
 
-        public WorldTemplateList(final Iterable<WorldTemplate> param1, final Iterable p_89795_) {
-            super(Minecraft.m_91087_(), var1.f_96543_, var1.f_96544_ - 33 - var1.m_294786_(), var1.m_294786_(), 46);
-            this.f_303630_ = var1;
+        public WorldTemplateList(final Iterable<WorldTemplate> p_89795_) {
+            super(Minecraft.m_91087_(), RealmsSelectWorldTemplateScreen.this.f_96543_, RealmsSelectWorldTemplateScreen.this.f_96544_ - 33 - RealmsSelectWorldTemplateScreen.this.m_294786_(), RealmsSelectWorldTemplateScreen.this.m_294786_(), 46);
             p_89795_.forEach(this::m_89804_);
         }
 
         public void m_89804_(WorldTemplate p_89805_) {
-            this.m_7085_(this.f_303630_.new Entry(p_89805_));
+            this.m_7085_(RealmsSelectWorldTemplateScreen.this.new Entry(p_89805_));
         }
 
         @Override
         public boolean m_6375_(double p_89797_, double p_89798_, int p_89799_) {
-            if (this.f_303630_.f_89597_ != null) {
-                ConfirmLinkScreen.m_274480_(this.f_303630_, this.f_303630_.f_89597_);
+            if (RealmsSelectWorldTemplateScreen.this.f_89597_ != null) {
+                ConfirmLinkScreen.m_274480_(RealmsSelectWorldTemplateScreen.this, RealmsSelectWorldTemplateScreen.this.f_89597_);
                 return true;
             } else {
                 return super.m_6375_(p_89797_, p_89798_, p_89799_);
@@ -394,8 +393,8 @@
 
         public void m_6987_(@Nullable RealmsSelectWorldTemplateScreen.Entry p_89807_) {
             super.m_6987_(p_89807_);
-            this.f_303630_.f_89613_ = p_89807_ == null ? null : p_89807_.f_89750_;
-            this.f_303630_.m_89718_();
+            RealmsSelectWorldTemplateScreen.this.f_89613_ = p_89807_ == null ? null : p_89807_.f_89750_;
+            RealmsSelectWorldTemplateScreen.this.m_89718_();
         }
 
         @Override
