--- a/net/minecraft/world/ticks/LevelTicks.java
+++ b/net/minecraft/world/ticks/LevelTicks.java
@@ -276,7 +276,7 @@
       long i = longsummarystatistics.getMin();
       long j = longsummarystatistics.getMax();
       list.forEach((p_200885_) -> {
-         this.m_183393_(new ScheduledTick<>(p_200885_.f_193376_(), p_200885_.f_193377_().m_141952_(p_193244_), p_200885_.f_193378_(), p_200885_.f_193379_(), p_200885_.f_193380_() - i + j + 1L));
+         this.m_183393_(new ScheduledTick<T>(p_200885_.f_193376_(), p_200885_.f_193377_().m_141952_(p_193244_), p_200885_.f_193378_(), p_200885_.f_193379_(), p_200885_.f_193380_() - i + j + 1L));
       });
    }
 
