--- a/net/minecraft/nbt/NBTUtil.java
+++ b/net/minecraft/nbt/NBTUtil.java
@@ -239,6 +239,6 @@
    }
 
    public static CompoundNBT func_210821_a(DataFixer p_210821_0_, DefaultTypeReferences p_210821_1_, CompoundNBT p_210821_2_, int p_210821_3_, int p_210821_4_) {
-      return p_210821_0_.update(p_210821_1_.func_219816_a(), new Dynamic<>(NBTDynamicOps.field_210820_a, p_210821_2_), p_210821_3_, p_210821_4_).getValue();
+      return (CompoundNBT)p_210821_0_.update(p_210821_1_.func_219816_a(), new Dynamic<>(NBTDynamicOps.field_210820_a, p_210821_2_), p_210821_3_, p_210821_4_).getValue();
    }
 }
