--- a/net/minecraft/advancements/PlayerAdvancements.java
+++ b/net/minecraft/advancements/PlayerAdvancements.java
@@ -136,7 +136,7 @@
 
                Stream<Entry<ResourceLocation, AdvancementProgress>> stream = map.entrySet().stream().sorted(Comparator.comparing(Entry::getValue));
 
-               for(Entry<ResourceLocation, AdvancementProgress> entry : (List)stream.collect(Collectors.toList())) {
+               for(Entry<ResourceLocation, AdvancementProgress> entry : stream.collect(Collectors.toList())) {
                   Advancement advancement = this.field_192756_d.func_191949_aK().func_192778_a(entry.getKey());
                   if (advancement == null) {
                      field_192753_a.warn("Ignored advancement '{}' in progress file {} - it doesn't exist anymore?", entry.getKey(), this.field_192757_e);
