--- a/net/minecraft/world/ticks/BlackholeTickAccess.java
+++ b/net/minecraft/world/ticks/BlackholeTickAccess.java
@@ -33,10 +33,10 @@
    };
 
    public static <T> TickContainerAccess<T> m_193144_() {
-      return f_193140_;
+      return (TickContainerAccess<T>)f_193140_;
    }
 
    public static <T> LevelTickAccess<T> m_193145_() {
-      return f_193141_;
+      return (LevelTickAccess<T>)f_193141_;
    }
 }
