--- a/net\minecraft\command\arguments\BlockStateParser.java
+++ b/net\minecraft\command\arguments\BlockStateParser.java
@@ -197,7 +197,7 @@
    private static <T extends Comparable<T>> SuggestionsBuilder func_201037_a(SuggestionsBuilder p_201037_0_, IProperty<T> p_201037_1_) {
       for(T t : p_201037_1_.func_177700_c()) {
          if (t instanceof Integer) {
-            p_201037_0_.suggest(t);
+            p_201037_0_.suggest((Integer)t);
          } else {
             p_201037_0_.suggest(p_201037_1_.func_177702_a(t));
          }
@@ -424,7 +424,7 @@
    private <T extends Comparable<T>> void func_197253_a(IProperty<T> p_197253_1_, String p_197253_2_, int p_197253_3_) throws CommandSyntaxException {
       Optional<T> optional = p_197253_1_.func_185929_b(p_197253_2_);
       if (optional.isPresent()) {
-         this.field_197270_l = (IBlockState)this.field_197270_l.func_206870_a(p_197253_1_, (Comparable)optional.get());
+         this.field_197270_l = (IBlockState)this.field_197270_l.func_206870_a(p_197253_1_, optional.get());
          this.field_197267_i.put(p_197253_1_, optional.get());
       } else {
          this.field_197266_h.setCursor(p_197253_3_);
