--- a/net/minecraft/network/chat/contents/NbtContents.java
+++ b/net/minecraft/network/chat/contents/NbtContents.java
@@ -151,15 +151,7 @@
     }
 
     private static String m_385183_(Tag p_396919_) {
-        if (p_396919_ instanceof StringTag stringtag) {
-            StringTag stringtag1 = stringtag;
-
-            try {
-                s = stringtag1.f_382711_();
-            } catch (Throwable throwable) {
-                throw new MatchException(throwable.toString(), throwable);
-            }
-
+        if (p_396919_ instanceof StringTag(String s)) {
             return s;
         } else {
             return p_396919_.toString();
