--- a/net/minecraft/world/entity/ai/Brain.java
+++ b/net/minecraft/world/entity/ai/Brain.java
@@ -238,7 +238,7 @@
     }
 
     private static <U> void callVisitor(final Brain.Visitor visitor, final MemoryModuleType<U> memoryModuleType, final MemorySlot<?> slot) {
-        slot.visit(memoryModuleType, visitor);
+        ((MemorySlot<U>)slot).visit(memoryModuleType, visitor);
     }
 
     public <U> boolean isMemoryValue(final MemoryModuleType<U> memoryType, final U value) {
