--- a/net/minecraft/util/ClassInstanceMultiMap.java
+++ b/net/minecraft/util/ClassInstanceMultiMap.java
@@ -58,7 +58,7 @@
          List<? extends T> list = this.f_13527_.computeIfAbsent(p_13534_, (p_13538_) -> {
             return this.f_13529_.stream().filter(p_13538_::isInstance).collect(Collectors.toList());
          });
-         return Collections.unmodifiableCollection(list);
+         return (Collection<S>)Collections.unmodifiableCollection(list);
       }
    }
 
