--- a/net/minecraft/client/multiplayer/ClientLevel.java
+++ b/net/minecraft/client/multiplayer/ClientLevel.java
@@ -1111,6 +1111,7 @@
                 case EnderDragon enderdragon:
                     ClientLevel.this.f_371413_.addAll(Arrays.asList(enderdragon.m_31156_()));
                     break;
+                default:
             }
         }
 
@@ -1124,6 +1125,7 @@
                 case EnderDragon enderdragon:
                     ClientLevel.this.f_371413_.removeAll(Arrays.asList(enderdragon.m_31156_()));
                     break;
+                default:
             }
         }
 
