--- a/com/mojang/realmsclient/RealmsMainScreen.java
+++ b/com/mojang/realmsclient/RealmsMainScreen.java
@@ -1134,9 +1134,12 @@
    @OnlyIn(Dist.CLIENT)
    class CloseButton extends Button {
       public CloseButton() {
-         super(RealmsMainScreen.this.m_86363_() + 4, RealmsMainScreen.this.m_86366_() + 4, 12, 12, new TranslatableComponent("mco.selectServer.close"), (p_86775_) -> {
+         super(RealmsMainScreen.this.m_86363_() + 4, RealmsMainScreen.this.m_86366_() + 4, 12, 12, new TranslatableComponent("mco.selectServer.close"), null);
+      }
+
+      @Override
+      public void m_5691_() {
             RealmsMainScreen.this.m_86360_();
-         });
       }
 
       public void m_6303_(PoseStack p_86777_, int p_86778_, int p_86779_, float p_86780_) {
@@ -1166,7 +1169,11 @@
    @OnlyIn(Dist.CLIENT)
    class NewsButton extends Button {
       public NewsButton() {
-         super(RealmsMainScreen.this.f_96543_ - 62, 6, 20, 20, new TranslatableComponent("mco.news"), (p_86804_) -> {
+         super(RealmsMainScreen.this.f_96543_ - 62, 6, 20, 20, new TranslatableComponent("mco.news"), null);
+      }
+
+      @Override
+      public void m_5691_() {
             if (RealmsMainScreen.this.f_86259_ != null) {
                Util.m_137581_().m_137646_(RealmsMainScreen.this.f_86259_);
                if (RealmsMainScreen.this.f_86258_) {
@@ -1177,7 +1184,6 @@
                }
 
             }
-         });
       }
 
       public void m_6303_(PoseStack p_86806_, int p_86807_, int p_86808_, float p_86809_) {
@@ -1191,6 +1197,11 @@
          super(RealmsMainScreen.this.f_96543_ / 2 + 47, 6, 22, 22, TextComponent.f_131282_, RealmsMainScreen.this::m_86518_);
       }
 
+      @Override
+      public void m_5691_() {
+         RealmsMainScreen.this.m_86518_(this);
+      }
+
       public void m_86821_() {
          this.m_93666_(RealmsMainScreen.this.f_86292_ == 0 ? RealmsMainScreen.f_86235_ : RealmsMainScreen.f_86236_);
       }
@@ -1472,9 +1483,12 @@
    @OnlyIn(Dist.CLIENT)
    class ShowPopupButton extends Button {
       public ShowPopupButton() {
-         super(RealmsMainScreen.this.f_96543_ - 37, 6, 20, 20, new TranslatableComponent("mco.selectServer.info"), (p_86897_) -> {
+         super(RealmsMainScreen.this.f_96543_ - 37, 6, 20, 20, new TranslatableComponent("mco.selectServer.info"), null);
+      }
+
+      @Override
+      public void m_5691_() {
             RealmsMainScreen.this.f_86295_ = !RealmsMainScreen.this.f_86295_;
-         });
       }
 
       public void m_6303_(PoseStack p_86899_, int p_86900_, int p_86901_, float p_86902_) {
