--- a/net/minecraft/client/resources/model/ModelBakery.java
+++ b/net/minecraft/client/resources/model/ModelBakery.java
@@ -355,7 +355,7 @@
                      try {
                         InputStream inputstream = p_119258_.m_6679_();
 
-                        Pair pair2;
+                        Pair<String, BlockModelDefinition> pair2;
                         try {
                            pair2 = Pair.of(p_119258_.m_7816_(), BlockModelDefinition.m_111540_(this.f_119211_, new InputStreamReader(inputstream, StandardCharsets.UTF_8)));
                         } catch (Throwable throwable1) {
@@ -425,7 +425,7 @@
             } catch (Exception exception) {
                throw new ModelBakery.BlockStateDefinitionException(String.format("Exception loading blockstate definition: '%s': %s", resourcelocation1, exception));
             } finally {
-               HashMap hashmap = Maps.newHashMap();
+               HashMap<ModelBakery.ModelGroupKey, Set<BlockState>> hashmap = Maps.newHashMap();
                map.forEach((p_119336_, p_119337_) -> {
                   Pair<UnbakedModel, Supplier<ModelBakery.ModelGroupKey>> pair2 = map1.get(p_119337_);
                   if (pair2 == null) {
