--- a/net/minecraft/client/renderer/texture/TextureAtlasSprite.java
+++ b/net/minecraft/client/renderer/texture/TextureAtlasSprite.java
@@ -237,7 +237,7 @@
       } else if (this.f_118348_ != null) {
          if (!RenderSystem.m_69586_()) {
             RenderSystem.m_69879_(() -> {
-               p_118384_.m_118453_();
+               TextureAtlasSprite.this.f_118348_.m_118453_();
             });
          } else {
             this.f_118348_.m_118453_();
