--- a/net/minecraft/util/valueproviders/UniformInt.java
+++ b/net/minecraft/util/valueproviders/UniformInt.java
@@ -8,7 +8,7 @@
 import net.minecraft.util.Mth;
 
 public class UniformInt extends IntProvider {
-   public static final Codec<UniformInt> f_146614_ = RecordCodecBuilder.create((p_146628_) -> {
+   public static final Codec<UniformInt> f_146614_ = RecordCodecBuilder.<UniformInt>create((p_146628_) -> {
       return p_146628_.group(Codec.INT.fieldOf("min_inclusive").forGetter((p_146636_) -> {
          return p_146636_.f_146615_;
       }), Codec.INT.fieldOf("max_inclusive").forGetter((p_146633_) -> {
