--- a/net/minecraft/world/level/levelgen/feature/configurations/SpringConfiguration.java
+++ b/net/minecraft/world/level/levelgen/feature/configurations/SpringConfiguration.java
@@ -19,7 +19,7 @@
          return p_68143_.f_68126_;
       }), Codec.INT.fieldOf("hole_count").orElse(1).forGetter((p_68141_) -> {
          return p_68141_.f_68127_;
-      }), Registry.f_122824_.listOf().fieldOf("valid_blocks").xmap(ImmutableSet::copyOf, ImmutableList::copyOf).forGetter((p_68137_) -> {
+      }), Registry.f_122824_.listOf().fieldOf("valid_blocks").<Set<Block>>xmap(ImmutableSet::copyOf, ImmutableList::copyOf).forGetter((p_68137_) -> {
          return p_68137_.f_68128_;
       })).apply(p_68139_, SpringConfiguration::new);
    });
