--- a/net/minecraft/world/spawner/WorldEntitySpawner.java
+++ b/net/minecraft/world/spawner/WorldEntitySpawner.java
@@ -79,14 +79,15 @@
             mobentity = (MobEntity)entity;
          } while(mobentity.func_104002_bU() || mobentity.func_213392_I());
 
+         final Entity entity_f = entity;
          EntityClassification entityclassification = entity.func_200600_R().func_220339_d();
          if (entityclassification != EntityClassification.MISC) {
             BlockPos blockpos = entity.func_233580_cy_();
             long i = ChunkPos.func_77272_a(blockpos.func_177958_n() >> 4, blockpos.func_177952_p() >> 4);
             p_234964_2_.query(i, (p_234971_5_) -> {
-               MobSpawnInfo.SpawnCosts mobspawninfo$spawncosts = func_234980_b_(blockpos, p_234971_5_).func_242433_b().func_242558_a(entity.func_200600_R());
+               MobSpawnInfo.SpawnCosts mobspawninfo$spawncosts = func_234980_b_(blockpos, p_234971_5_).func_242433_b().func_242558_a(entity_f.func_200600_R());
                if (mobspawninfo$spawncosts != null) {
-                  mobdensitytracker.func_234998_a_(entity.func_233580_cy_(), mobspawninfo$spawncosts.func_242585_b());
+                  mobdensitytracker.func_234998_a_(entity_f.func_233580_cy_(), mobspawninfo$spawncosts.func_242585_b());
                }
 
                object2intopenhashmap.addTo(entityclassification, 1);
