--- a/net/minecraft/util/LowerCaseEnumTypeAdapterFactory.java
+++ b/net/minecraft/util/LowerCaseEnumTypeAdapterFactory.java
@@ -16,7 +16,7 @@
 public class LowerCaseEnumTypeAdapterFactory implements TypeAdapterFactory {
    @Nullable
    public <T> TypeAdapter<T> create(Gson p_13982_, TypeToken<T> p_13983_) {
-      Class<T> oclass = p_13983_.getRawType();
+      Class<T> oclass = (Class<T>)p_13983_.getRawType();
       if (!oclass.isEnum()) {
          return null;
       } else {
