--- a/net/minecraft/util/UniformInt.java
+++ b/net/minecraft/util/UniformInt.java
@@ -9,7 +9,7 @@
 import java.util.function.Function;
 
 public class UniformInt {
-   public static final Codec<UniformInt> f_14420_ = Codec.either(Codec.INT, RecordCodecBuilder.create((p_14446_) -> {
+   public static final Codec<UniformInt> f_14420_ = Codec.either(Codec.INT, RecordCodecBuilder.<UniformInt>create((p_14446_) -> {
       return p_14446_.group(Codec.INT.fieldOf("base").forGetter((p_14456_) -> {
          return p_14456_.f_14421_;
       }), Codec.INT.fieldOf("spread").forGetter((p_14454_) -> {
