--- a/net/minecraft/command/impl/ForceLoadCommand.java
+++ b/net/minecraft/command/impl/ForceLoadCommand.java
@@ -81,7 +81,7 @@
       DimensionType dimensiontype = p_212722_0_.func_197023_e().func_201675_m().func_186058_p();
       WorldServer worldserver = p_212722_0_.func_197028_i().func_71218_a(dimensiontype);
       LongSet longset = worldserver.func_212412_ag();
-      longset.forEach((p_212720_1_) -> {
+      longset.forEach((long p_212720_1_) -> {
          worldserver.func_212414_b(ChunkPos.func_212578_a(p_212720_1_), ChunkPos.func_212579_b(p_212720_1_), false);
       });
       p_212722_0_.func_197030_a(new TextComponentTranslation("commands.forceload.removed.all", new Object[]{dimensiontype}), true);
