--- a/net/minecraft/world/entity/monster/Phantom.java
+++ b/net/minecraft/world/entity/monster/Phantom.java
@@ -226,7 +226,7 @@
             this.f_33193_ = m_186073_(60);
             List<Player> list = Phantom.this.m_9236_().m_45955_(this.f_33192_, Phantom.this, Phantom.this.m_20191_().m_82377_(16.0D, 64.0D, 16.0D));
             if (!list.isEmpty()) {
-               list.sort(Comparator.comparing(Entity::m_20186_).reversed());
+               list.sort(Comparator.<Entity, Double>comparing(Entity::m_20186_).reversed());
 
                for(Player player : list) {
                   if (Phantom.this.m_21040_(player, TargetingConditions.f_26872_)) {
