--- a/net/minecraft/client/gui/screens/inventory/CommandBlockEditScreen.java
+++ b/net/minecraft/client/gui/screens/inventory/CommandBlockEditScreen.java
@@ -34,7 +34,7 @@
 
    protected void m_7856_() {
       super.m_7856_();
-      this.f_98375_ = this.m_142416_(CycleButton.m_168894_((p_169719_) -> {
+      this.f_98375_ = this.m_142416_(CycleButton.<CommandBlockEntity.Mode>m_168894_((p_169719_) -> {
          switch(p_169719_) {
          case SEQUENCE:
             return Component.m_239796_("advMode.mode.sequence");
