--- a/net/minecraft/client/gui/screens/inventory/BookViewScreen.java
+++ b/net/minecraft/client/gui/screens/inventory/BookViewScreen.java
@@ -279,7 +279,7 @@
                 return new BookViewScreen.BookAccess(writtenbookcontent.getPages(flag));
             } else {
                 WritableBookContent writablebookcontent = itemStack.get(DataComponents.WRITABLE_BOOK_CONTENT);
-                return writablebookcontent != null ? new BookViewScreen.BookAccess(writablebookcontent.getPages(flag).map(Component::literal).toList()) : null;
+                return writablebookcontent != null ? new BookViewScreen.BookAccess(writablebookcontent.getPages(flag).<Component>map(Component::literal).toList()) : null;
             }
         }
     }
