--- a/net/minecraft/world/chunk/ChunkTaskPriorityQueue.java
+++ b/net/minecraft/world/chunk/ChunkTaskPriorityQueue.java
@@ -18,7 +18,7 @@
 public class ChunkTaskPriorityQueue<T> {
    public static final int field_219419_a = ChunkManager.field_219249_a + 2;
    private final List<Long2ObjectLinkedOpenHashMap<List<Optional<T>>>> field_219420_b = IntStream.range(0, field_219419_a).mapToObj((p_219415_0_) -> {
-      return new Long2ObjectLinkedOpenHashMap<>();
+      return new Long2ObjectLinkedOpenHashMap<List<Optional<T>>>();
    }).collect(Collectors.toList());
    private volatile int field_219421_c = field_219419_a;
    private final String field_219422_d;
