--- a/net/minecraft/data/registries/RegistriesDatapackGenerator.java
+++ b/net/minecraft/data/registries/RegistriesDatapackGenerator.java
@@ -52,7 +52,7 @@
                     PackOutput.PathProvider packoutput$pathprovider = this.f_254743_.m_339483_(resourcekey);
                     return CompletableFuture.allOf(
                         p_358460_.m_205916_()
-                            .map(
+                            .<CompletableFuture<?>>map(
                                 p_256105_ -> m_255283_(
                                     packoutput$pathprovider.m_245731_(p_256105_.m_205785_().m_135782_()),
                                     p_256502_,
