--- a/net/minecraft/test/TestTickResult.java
+++ b/net/minecraft/test/TestTickResult.java
@@ -6,4 +6,10 @@
    @Nullable
    public final Long field_229485_a_;
    public final Runnable field_229486_b_;
+
+   private TestTickResult() {
+      this.field_229485_a_ = null;
+      this.field_229486_b_ = null;
+      throw new IllegalStateException("This is a synthetic added by MCP do not call");
 }
+}
