--- a/net/minecraft/world/ticks/ScheduledTick.java
+++ b/net/minecraft/world/ticks/ScheduledTick.java
@@ -5,7 +5,7 @@
 import javax.annotation.Nullable;
 import net.minecraft.core.BlockPos;
 
-public record ScheduledTick(T f_193376_, BlockPos f_193377_, long f_193378_, TickPriority f_193379_, long f_193380_) {
+public record ScheduledTick<T>(T f_193376_, BlockPos f_193377_, long f_193378_, TickPriority f_193379_, long f_193380_) {
    public static final Comparator<ScheduledTick<?>> f_193373_ = (p_193406_, p_193407_) -> {
       int i = Long.compare(p_193406_.f_193378_, p_193407_.f_193378_);
       if (i != 0) {
