--- a/net/minecraft/client/resources/ResourceIndex.java
+++ b/net/minecraft/client/resources/ResourceIndex.java
@@ -73,7 +73,7 @@
    public Collection<String> func_211685_a(String p_211685_1_, int p_211685_2_, Predicate<String> p_211685_3_) {
       return (Collection)this.field_152784_b.keySet().stream().filter((p_211684_0_) -> {
          return !p_211684_0_.endsWith(".mcmeta");
-      }).<String>map(ResourceLocation::new).<String>map(ResourceLocation::func_110623_a).filter((p_211683_1_) -> {
+      }).map(ResourceLocation::new).<String>map(ResourceLocation::func_110623_a).filter((p_211683_1_) -> {
          return p_211683_1_.startsWith(p_211685_1_ + "/");
       }).filter(p_211685_3_).collect(Collectors.toList());
    }
