--- a/net/minecraft/world/level/lighting/DynamicGraphMinFixedPoint.java
+++ b/net/minecraft/world/level/lighting/DynamicGraphMinFixedPoint.java
@@ -84,13 +84,13 @@
 
    public void m_75581_(LongPredicate p_75582_) {
       LongList longlist = new LongArrayList();
-      this.f_75539_.keySet().forEach((p_75586_) -> {
+      this.f_75539_.keySet().forEach((long p_75586_) -> {
          if (p_75582_.test(p_75586_)) {
             longlist.add(p_75586_);
          }
 
       });
-      longlist.forEach(this::m_75600_);
+      longlist.forEach((java.util.function.LongConsumer)this::m_75600_);
    }
 
    private void m_75558_(long p_75559_, int p_75560_, int p_75561_, boolean p_75562_) {
