--- a/net/minecraft/world/level/block/entity/AbstractFurnaceBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/AbstractFurnaceBlockEntity.java
@@ -106,7 +106,7 @@
 
    protected AbstractFurnaceBlockEntity(BlockEntityType<?> p_154991_, BlockPos p_154992_, BlockState p_154993_, RecipeType<? extends AbstractCookingRecipe> p_154994_) {
       super(p_154991_, p_154992_, p_154993_);
-      this.f_222691_ = RecipeManager.m_220267_(p_154994_);
+      this.f_222691_ = RecipeManager.m_220267_((RecipeType)p_154994_);
    }
 
    public static Map<Item, Integer> m_58423_() {
@@ -243,7 +243,7 @@
       if (p_155017_.m_58425_() || flag3 && flag2) {
          Recipe<?> recipe;
          if (flag2) {
-            recipe = p_155017_.f_222691_.m_213657_(p_155017_, p_155014_).orElse((AbstractCookingRecipe)null);
+            recipe = p_155017_.f_222691_.m_213657_(p_155017_, p_155014_).orElse(null);
          } else {
             recipe = null;
          }
