diff -r -U 3 minecraft\net\minecraft\block\state\BlockStateContainer.java minecraft_patched\net\minecraft\block\state\BlockStateContainer.java
--- minecraft\net\minecraft\block\state\BlockStateContainer.java
+++ minecraft_patched\net\minecraft\block\state\BlockStateContainer.java
@@ -116,7 +116,7 @@
 
       while(unmodifiableiterator.hasNext()) {
          IProperty<?> iproperty = (IProperty)unmodifiableiterator.next();
-         list.add(iproperty.func_177700_c());
+         list.add(((IProperty)iproperty).func_177700_c());
       }
 
       return list;
