diff -r -U 3 minecraft\net\minecraft\world\chunk\storage\AnvilChunkLoader.java minecraft_patched\net\minecraft\world\chunk\storage\AnvilChunkLoader.java
--- minecraft\net\minecraft\world\chunk\storage\AnvilChunkLoader.java
+++ minecraft_patched\net\minecraft\world\chunk\storage\AnvilChunkLoader.java
@@ -164,11 +164,7 @@
       try {
          this.field_183014_e = true;
 
-         while(true) {
-            if (this.func_75814_c()) {
-               continue;
-            }
-         }
+         while(this.func_75814_c());
       } finally {
          this.field_183014_e = false;
       }
