diff -r -U 3 minecraft\net\minecraft\client\renderer\texture\TextureManager.java minecraft_patched\net\minecraft\client\renderer\texture\TextureManager.java
--- minecraft\net\minecraft\client\renderer\texture\TextureManager.java
+++ minecraft_patched\net\minecraft\client\renderer\texture\TextureManager.java
@@ -51,7 +51,7 @@
       }
    }
 
-   public boolean func_110579_a(ResourceLocation p_110579_1_, final ITextureObject p_110579_2_) {
+   public boolean func_110579_a(ResourceLocation p_110579_1_, ITextureObject p_110579_2_) {
       boolean flag = true;
 
       try {
@@ -68,9 +68,10 @@
          CrashReport crashreport = CrashReport.func_85055_a(throwable, "Registering texture");
          CrashReportCategory crashreportcategory = crashreport.func_85058_a("Resource location being registered");
          crashreportcategory.func_71507_a("Resource location", p_110579_1_);
+         final ITextureObject texture = p_110579_2_;
          crashreportcategory.func_189529_a("Texture object class", new ICrashReportDetail<String>() {
             public String call() throws Exception {
-               return p_110579_2_.getClass().getName();
+               return texture.getClass().getName();
             }
          });
          throw new ReportedException(crashreport);
