--- a/net/minecraft/state/StateContainer.java
+++ b/net/minecraft/state/StateContainer.java
@@ -55,10 +55,10 @@
             });
          });
       }
-
+      final MapCodec<S> mapcodec_f = mapcodec;
       stream.forEach((p_201000_5_) -> {
          ImmutableMap<Property<?>, Comparable<?>> immutablemap = p_201000_5_.stream().collect(ImmutableMap.toImmutableMap(Pair::getFirst, Pair::getSecond));
-         S s1 = p_i231877_3_.create(p_i231877_2_, immutablemap, mapcodec);
+         S s1 = p_i231877_3_.create(p_i231877_2_, immutablemap, mapcodec_f);
          map.put(immutablemap, s1);
          list.add(s1);
       });
