--- a/net/minecraft/client/resources/model/ModelBakery.java
+++ b/net/minecraft/client/resources/model/ModelBakery.java
@@ -396,7 +396,7 @@
             } catch (Exception exception) {
                throw new ModelBakery.BlockStateDefinitionException(String.format("Exception loading blockstate definition: '%s': %s", resourcelocation1, exception));
             } finally {
-               HashMap lvt_20_1_ = Maps.newHashMap();
+               HashMap<ModelBakery.ModelGroupKey, Set<BlockState>> lvt_20_1_ = Maps.newHashMap();
                map.forEach((p_119336_, p_119337_) -> {
                   Pair<UnbakedModel, Supplier<ModelBakery.ModelGroupKey>> pair2 = map1.get(p_119337_);
                   if (pair2 == null) {
