--- a/net/minecraft/client/renderer/entity/RenderSkeleton.java
+++ b/net/minecraft/client/renderer/entity/RenderSkeleton.java
@@ -18,8 +18,8 @@
       this.func_177094_a(new LayerHeldItem(this));
       this.func_177094_a(new LayerBipedArmor(this) {
          protected void func_177177_a() {
-            this.field_177189_c = (T)(new ModelSkeleton(0.5F, true));
-            this.field_177186_d = (T)(new ModelSkeleton(1.0F, true));
+            this.field_177189_c = new ModelSkeleton(0.5F, true);
+            this.field_177186_d = new ModelSkeleton(1.0F, true);
          }
       });
    }
