--- a/net/minecraft/world/entity/animal/allay/AllayAi.java
+++ b/net/minecraft/world/entity/animal/allay/AllayAi.java
@@ -63,7 +63,7 @@
             0,
             ImmutableList.of(
                 new Swim<>(0.8F),
-                new AnimalPanic(2.5F),
+                new AnimalPanic<Allay>(2.5F),
                 new LookAtTargetSink(45, 90),
                 new MoveToTargetSink(),
                 new CountDownCooldownTicks(MemoryModuleType.f_217780_),
