--- a/net/minecraft/util/datafix/fixes/LWJGL3KeyOptions.java
+++ b/net/minecraft/util/datafix/fixes/LWJGL3KeyOptions.java
@@ -134,7 +134,7 @@
    public TypeRewriteRule makeRule() {
       return this.fixTypeEverywhereTyped("OptionsKeyLwjgl3Fix", this.getInputSchema().getType(TypeReferences.field_211289_e), (p_207423_0_) -> {
          return p_207423_0_.update(DSL.remainderFinder(), (p_207424_0_) -> {
-            return p_207424_0_.getMapValues().map((p_209663_1_) -> {
+            return p_207424_0_.getMapValues().<com.mojang.serialization.Dynamic<?>>map((p_209663_1_) -> {
                return p_207424_0_.createMap(p_209663_1_.entrySet().stream().map((p_209661_0_) -> {
                   if (p_209661_0_.getKey().asString("").startsWith("key_")) {
                      int i = Integer.parseInt(p_209661_0_.getValue().asString(""));
