--- a/net/minecraft/client/renderer/PostChain.java
+++ b/net/minecraft/client/renderer/PostChain.java
@@ -96,125 +96,17 @@
         RenderPipeline renderpipeline = renderpipeline$builder.build();
         List<PostPass.Input> list = new ArrayList<>();
 
-        label110:
         for (PostChainConfig.Input postchainconfig$input1 : config.inputs()) {
-            Objects.requireNonNull(postchainconfig$input1);
-            PostChainConfig.Input postchainconfig$input2 = postchainconfig$input1;
-            byte b0 = 0;
-
-            while (true) {
-                switch (SwitchBootstraps.typeSwitch<"typeSwitch",PostChainConfig.TextureInput,PostChainConfig.TargetInput>(postchainconfig$input2, b0)) {
-                    case 0:
-                        PostChainConfig.TextureInput postchainconfig$textureinput = (PostChainConfig.TextureInput)postchainconfig$input2;
-                        PostChainConfig.TextureInput postchainconfig$textureinput1 = postchainconfig$textureinput;
-
-                        try {
-                            s4 = postchainconfig$textureinput1.samplerName();
-                        } catch (Throwable throwable8) {
-                            throw new MatchException(throwable8.toString(), throwable8);
-                        }
-
-                        String s2 = s4;
-                        postchainconfig$textureinput1 = postchainconfig$textureinput;
-
-                        try {
-                            identifier3 = postchainconfig$textureinput1.location();
-                        } catch (Throwable throwable7) {
-                            throw new MatchException(throwable7.toString(), throwable7);
-                        }
-
-                        Identifier identifier = identifier3;
-                        postchainconfig$textureinput1 = postchainconfig$textureinput;
-
-                        try {
-                            k = postchainconfig$textureinput1.width();
-                        } catch (Throwable throwable6) {
-                            throw new MatchException(throwable6.toString(), throwable6);
-                        }
-
-                        int i = k;
-                        if (true) {
-                            postchainconfig$textureinput1 = postchainconfig$textureinput;
-
-                            try {
-                                l = postchainconfig$textureinput1.height();
-                            } catch (Throwable throwable5) {
-                                throw new MatchException(throwable5.toString(), throwable5);
-                            }
-
-                            int j = l;
-                            if (true) {
-                                postchainconfig$textureinput1 = postchainconfig$textureinput;
-
-                                try {
-                                    flag5 = postchainconfig$textureinput1.bilinear();
-                                } catch (Throwable throwable4) {
-                                    throw new MatchException(throwable4.toString(), throwable4);
-                                }
-
-                                boolean flag = flag5;
-                                if (true) {
+            switch (postchainconfig$input1) {
+                case PostChainConfig.TextureInput(String s2, Identifier identifier, int i, int j, boolean flag):
                                     AbstractTexture abstracttexture = textureManager.getTexture(identifier.withPath(path -> "textures/effect/" + path + ".png"));
                                     list.add(new PostPass.TextureInput(s2, abstracttexture, i, j, flag));
-                                    continue label110;
-                                }
-                            }
-                        }
-
-                        b0 = 1;
                         break;
-                    case 1:
-                        PostChainConfig.TargetInput texture = (PostChainConfig.TargetInput)postchainconfig$input2;
-                        PostChainConfig.TargetInput postchainconfig$targetinput1 = texture;
-
-                        try {
-                            s3 = postchainconfig$targetinput1.samplerName();
-                        } catch (Throwable throwable3) {
-                            throw new MatchException(throwable3.toString(), throwable3);
-                        }
-
-                        String s = s3;
-                        postchainconfig$targetinput1 = texture;
-
-                        try {
-                            identifier2 = postchainconfig$targetinput1.targetId();
-                        } catch (Throwable throwable2) {
-                            throw new MatchException(throwable2.toString(), throwable2);
-                        }
-
-                        Identifier identifier1 = identifier2;
-                        postchainconfig$targetinput1 = texture;
-
-                        try {
-                            flag3 = postchainconfig$targetinput1.useDepthBuffer();
-                        } catch (Throwable throwable1) {
-                            throw new MatchException(throwable1.toString(), throwable1);
-                        }
-
-                        boolean flag1 = flag3;
-                        if (true) {
-                            postchainconfig$targetinput1 = texture;
-
-                            try {
-                                flag4 = postchainconfig$targetinput1.bilinear();
-                            } catch (Throwable throwable) {
-                                throw new MatchException(throwable.toString(), throwable);
-                            }
-
-                            boolean flag2 = flag4;
-                            if (true) {
+                case PostChainConfig.TargetInput(String s, Identifier identifier1, boolean flag1, boolean flag2):
                                 list.add(new PostPass.TargetInput(s, identifier1, flag1, flag2));
-                                continue label110;
-                            }
-                        }
-
-                        b0 = 2;
                         break;
-                    default:
-                        throw new MatchException(null, null);
                 }
             }
-        }
 
         return new PostPass(renderpipeline, config.outputTarget(), config.uniforms(), list);
     }
