--- a/net/minecraft/client/particle/ParticleManager.java
+++ b/net/minecraft/client/particle/ParticleManager.java
@@ -140,7 +140,7 @@
 
    @Nullable
    private <T extends IParticleData> Particle func_199927_b(T p_199927_1_, double p_199927_2_, double p_199927_4_, double p_199927_6_, double p_199927_8_, double p_199927_10_, double p_199927_12_) {
-      IParticleFactory<T> iparticlefactory = this.field_178932_g.get(ParticleType.field_197577_a.func_148757_b(p_199927_1_.func_197554_b()));
+      IParticleFactory<T> iparticlefactory = (IParticleFactory<T>)this.field_178932_g.get(ParticleType.field_197577_a.func_148757_b(p_199927_1_.func_197554_b()));
       return iparticlefactory == null ? null : iparticlefactory.func_199234_a(p_199927_1_, this.field_78878_a, p_199927_2_, p_199927_4_, p_199927_6_, p_199927_8_, p_199927_10_, p_199927_12_);
    }
 
@@ -242,7 +242,8 @@
       GlStateManager.func_187401_a(GlStateManager.SourceFactor.SRC_ALPHA, GlStateManager.DestFactor.ONE_MINUS_SRC_ALPHA);
       GlStateManager.func_179092_a(516, 0.003921569F);
 
-      for(int i = 0; i < 3; ++i) {
+      for(int i_nf = 0; i_nf < 3; ++i_nf) {
+         final int i = i_nf;
          for(int j = 0; j < 2; ++j) {
             if (!this.field_78876_b[i][j].isEmpty()) {
                switch(j) {
