--- a/net/minecraft/client/renderer/texture/Stitcher.java
+++ b/net/minecraft/client/renderer/texture/Stitcher.java
@@ -13,7 +13,7 @@
 
 @OnlyIn(Dist.CLIENT)
 public class Stitcher {
-   private static final Comparator<Stitcher.Holder> f_118161_ = Comparator.comparing((p_118201_) -> {
+   private static final Comparator<Stitcher.Holder> f_118161_ = Comparator.<Stitcher.Holder, Integer>comparing((p_118201_) -> {
       return -p_118201_.f_118204_;
    }).thenComparing((p_118199_) -> {
       return -p_118199_.f_118203_;
