--- a/net/minecraft/world/level/block/entity/SignBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/SignBlockEntity.java
@@ -194,17 +194,7 @@
         for (Component component : this.m_277157_(p_278254_).m_276945_(p_279304_.m_143387_())) {
             Style style = component.m_7383_();
             ClickEvent clickevent = style.m_131182_();
-            if (clickevent instanceof ClickEvent.RunCommand) {
-                ClickEvent.RunCommand clickevent$runcommand = (ClickEvent.RunCommand)clickevent;
-                ClickEvent.RunCommand clickevent$runcommand1 = clickevent$runcommand;
-
-                try {
-                    s1 = clickevent$runcommand1.f_380064_();
-                } catch (Throwable throwable) {
-                    throw new MatchException(throwable.toString(), throwable);
-                }
-
-                String s = s1;
+            if (clickevent instanceof ClickEvent.RunCommand(String s)) {
                 p_279304_.m_20194_().m_129892_().m_230957_(m_278808_(p_279304_, p_279201_, p_278282_), s);
                 flag = true;
             }
