--- a/com/mojang/blaze3d/font/TrueTypeGlyphProvider.java
+++ b/com/mojang/blaze3d/font/TrueTypeGlyphProvider.java
@@ -65,7 +65,7 @@
                 int k = ft_glyphslot.bitmap_top();
                 int l = ft_bitmap.width();
                 int i1 = ft_bitmap.rows();
-                return (GlyphInfo)(l > 0 && i1 > 0 ? new TrueTypeGlyphProvider.Glyph((float)j, (float)k, l, i1, f, i) : () -> f / this.f_83839_);
+                return (GlyphInfo)(l > 0 && i1 > 0 ? new TrueTypeGlyphProvider.Glyph((float)j, (float)k, l, i1, f, i) : (GlyphInfo.SpaceGlyphInfo)() -> f / this.f_83839_);
             }
         }
     }
