--- a/net/minecraft/resources/RegistryLoader.java
+++ b/net/minecraft/resources/RegistryLoader.java
@@ -76,7 +76,7 @@
    }
 
    private <E> RegistryLoader.ReadCache<E> m_206773_(ResourceKey<? extends Registry<E>> p_206774_) {
-      return this.f_206751_.computeIfAbsent(p_206774_, (p_206782_) -> {
+      return (RegistryLoader.ReadCache<E>)this.f_206751_.computeIfAbsent(p_206774_, (p_206782_) -> {
          return new RegistryLoader.ReadCache();
       });
    }
