--- a/net/minecraft/resources/RegistryResourceAccess.java
+++ b/net/minecraft/resources/RegistryResourceAccess.java
@@ -151,7 +151,7 @@
          return this.f_195913_.entrySet().stream().filter((p_217504_) -> {
             return p_217504_.getKey().m_135783_(p_224966_);
          }).collect(Collectors.toMap((p_217510_) -> {
-            return p_217510_.getKey();
+            return (ResourceKey<E>)p_217510_.getKey();
          }, (p_217512_) -> {
             return p_217512_.getValue()::m_223944_;
          }));
