--- a/net/minecraft/core/Registry.java
+++ b/net/minecraft/core/Registry.java
@@ -349,9 +349,9 @@
    public static final Registry<Codec<? extends ChunkGenerator>> f_122890_ = m_205999_(f_122853_, Lifecycle.stable(), (p_206063_) -> {
       return ChunkGenerator.f_62136_;
    });
-   public static final Registry<Codec<? extends SurfaceRules.ConditionSource>> f_194573_ = m_206008_(f_194571_, SurfaceRules.ConditionSource::m_204624_);
-   public static final Registry<Codec<? extends SurfaceRules.RuleSource>> f_194574_ = m_206008_(f_194572_, SurfaceRules.RuleSource::m_204630_);
-   public static final Registry<Codec<? extends DensityFunction>> f_211076_ = m_206008_(f_211075_, DensityFunctions::m_208342_);
+   public static final Registry<Codec<? extends SurfaceRules.ConditionSource>> f_194573_ = m_206008_(f_194571_, (RegistryBootstrap)SurfaceRules.ConditionSource::m_204624_);
+   public static final Registry<Codec<? extends SurfaceRules.RuleSource>> f_194574_ = m_206008_(f_194572_, (RegistryBootstrap)SurfaceRules.RuleSource::m_204630_);
+   public static final Registry<Codec<? extends DensityFunction>> f_211076_ = m_206008_(f_211075_, (RegistryBootstrap)DensityFunctions::m_208342_);
    public static final Registry<StructureProcessorType<?>> f_122891_ = m_206008_(f_122854_, (p_206056_) -> {
       return StructureProcessorType.f_74456_;
    });
@@ -412,7 +412,7 @@
       f_122834_.put(resourcelocation, () -> {
          return p_206014_.m_206149_(p_206013_);
       });
-      f_122896_.m_203505_(p_206012_, p_206013_, p_206015_);
+      f_122896_.m_203505_((ResourceKey)p_206012_, p_206013_, p_206015_);
       return p_206013_;
    }
 
@@ -571,7 +571,7 @@
    public abstract Optional<HolderSet.Named<T>> m_203431_(TagKey<T> p_206052_);
 
    public Iterable<Holder<T>> m_206058_(TagKey<T> p_206059_) {
-      return DataFixUtils.orElse(this.m_203431_(p_206059_), List.<T>of());
+      return DataFixUtils.orElse(this.m_203431_(p_206059_), List.of());
    }
 
    public abstract HolderSet.Named<T> m_203561_(TagKey<T> p_206045_);
@@ -594,7 +594,7 @@
 
          @Nullable
          public Holder<T> m_7942_(int p_206147_) {
-            return (Holder)Registry.this.m_203300_(p_206147_).orElse((T)null);
+            return (Holder)Registry.this.m_203300_(p_206147_).orElse(null);
          }
 
          public int m_183450_() {
@@ -603,7 +603,7 @@
 
          public Iterator<Holder<T>> iterator() {
             return Registry.this.m_203611_().map((p_206140_) -> {
-               return p_206140_;
+               return (Holder<T>)p_206140_;
             }).iterator();
          }
       };
