--- a/net/minecraft/util/datafix/fixes/BlockStateFlattenVillageCrops.java
+++ b/net/minecraft/util/datafix/fixes/BlockStateFlattenVillageCrops.java
@@ -25,7 +25,7 @@
    }
 
    private static Stream<? extends Dynamic<?>> func_210586_a(Stream<? extends Dynamic<?>> p_210586_0_) {
-      return p_210586_0_.map((p_210587_0_) -> {
+      return p_210586_0_.<Dynamic<?>>map((p_210587_0_) -> {
          String s = p_210587_0_.getString("id");
          if ("ViF".equals(s)) {
             return func_210588_c(p_210587_0_);
