--- a/net/minecraft/network/protocol/BundlerInfo.java
+++ b/net/minecraft/network/protocol/BundlerInfo.java
@@ -48,7 +48,7 @@
                   } else if (this.f_263710_.size() >= 4096) {
                      throw new IllegalStateException("Too many packets in a bundle");
                   } else {
-                     this.f_263710_.add(p_265205_);
+                     this.f_263710_.add((Packet<T>)p_265205_);
                      return null;
                   }
                }
