--- a/net/minecraft/server/level/ChunkTaskPriorityQueueSorter.java
+++ b/net/minecraft/server/level/ChunkTaskPriorityQueueSorter.java
@@ -63,21 +63,21 @@
    }
 
    public <T> ProcessorHandle<ChunkTaskPriorityQueueSorter.Message<T>> m_140604_(ProcessorHandle<T> p_140605_, boolean p_140606_) {
-      return this.f_140552_.m_18720_((p_140610_) -> {
+      return this.f_140552_.<ProcessorHandle<ChunkTaskPriorityQueueSorter.Message<T>>>m_18720_((p_140610_) -> {
          return new StrictQueue.IntRunnable(0, () -> {
             this.m_140652_(p_140605_);
             p_140610_.m_6937_(ProcessorHandle.m_18714_("chunk priority sorter around " + p_140605_.m_7326_(), (p_143176_) -> {
-               this.m_140589_(p_140605_, p_143176_.f_140664_, p_143176_.f_140665_, p_143176_.f_140666_, p_140606_);
+               this.m_140589_(p_140605_, ((Message<T>)p_143176_).f_140664_, ((Message<T>)p_143176_).f_140665_, ((Message<T>)p_143176_).f_140666_, p_140606_);
             }));
          });
       }).join();
    }
 
    public ProcessorHandle<ChunkTaskPriorityQueueSorter.Release> m_140567_(ProcessorHandle<Runnable> p_140568_) {
-      return this.f_140552_.m_18720_((p_140581_) -> {
+      return this.f_140552_.<ProcessorHandle<ChunkTaskPriorityQueueSorter.Release>>m_18720_((p_140581_) -> {
          return new StrictQueue.IntRunnable(0, () -> {
             p_140581_.m_6937_(ProcessorHandle.m_18714_("chunk priority sorter around " + p_140568_.m_7326_(), (p_143165_) -> {
-               this.m_140569_(p_140568_, p_143165_.f_140683_, p_143165_.f_140682_, p_143165_.f_140684_);
+               this.m_140569_(p_140568_, ((Release)p_143165_).f_140683_, ((Release)p_143165_).f_140682_, ((Release)p_143165_).f_140684_);
             }));
          });
       }).join();
@@ -145,7 +145,7 @@
       if (chunktaskpriorityqueue == null) {
          throw (IllegalArgumentException)Util.m_137570_(new IllegalArgumentException("No queue for: " + p_140653_));
       } else {
-         return chunktaskpriorityqueue;
+         return (ChunkTaskPriorityQueue<Function<ProcessorHandle<Unit>, T>>)chunktaskpriorityqueue;
       }
    }
 
