--- a/net/minecraft/server/level/ChunkTaskPriorityQueueSorter.java
+++ b/net/minecraft/server/level/ChunkTaskPriorityQueueSorter.java
@@ -55,7 +55,7 @@
    }
 
    public <T> ProcessorHandle<ChunkTaskPriorityQueueSorter.Message<T>> m_140604_(ProcessorHandle<T> p_140605_, boolean p_140606_) {
-      return this.f_140552_.m_18720_((p_140610_) -> {
+      return this.f_140552_.<ProcessorHandle<ChunkTaskPriorityQueueSorter.Message<T>>>m_18720_((p_140610_) -> {
          return new StrictQueue.IntRunnable(0, () -> {
             this.m_140652_(p_140605_);
             p_140610_.m_6937_(ProcessorHandle.m_18714_("chunk priority sorter around " + p_140605_.m_7326_(), (p_140614_) -> {
@@ -66,7 +66,7 @@
    }
 
    public ProcessorHandle<ChunkTaskPriorityQueueSorter.Release> m_140567_(ProcessorHandle<Runnable> p_140568_) {
-      return this.f_140552_.m_18720_((p_140581_) -> {
+      return this.f_140552_.<ProcessorHandle<ChunkTaskPriorityQueueSorter.Release>>m_18720_((p_140581_) -> {
          return new StrictQueue.IntRunnable(0, () -> {
             p_140581_.m_6937_(ProcessorHandle.m_18714_("chunk priority sorter around " + p_140568_.m_7326_(), (p_140603_) -> {
                this.m_140569_(p_140568_, p_140603_.f_140683_, p_140603_.f_140682_, p_140603_.f_140684_);
@@ -137,7 +137,7 @@
       if (chunktaskpriorityqueue == null) {
          throw (IllegalArgumentException)Util.m_137570_(new IllegalArgumentException("No queue for: " + p_140653_));
       } else {
-         return chunktaskpriorityqueue;
+         return (ChunkTaskPriorityQueue<Function<ProcessorHandle<Unit>, T>>)chunktaskpriorityqueue;
       }
    }
 
