--- a/net/minecraft/world/Stopwatches.java
+++ b/net/minecraft/world/Stopwatches.java
@@ -29,7 +29,7 @@
     private static Stopwatches unpack(final Map<Identifier, Long> stopwatches) {
         Stopwatches stopwatchesx = new Stopwatches();
         long i = currentTime();
-        stopwatches.forEach((id, accumulatedElapsedTime) -> stopwatches.stopwatches.put(id, new Stopwatch(i, accumulatedElapsedTime)));
+        stopwatches.forEach((id, accumulatedElapsedTime) -> stopwatchesx.stopwatches.put(id, new Stopwatch(i, accumulatedElapsedTime)));
         return stopwatchesx;
     }
 
