--- a/net/minecraft/data/registries/RegistriesDatapackGenerator.java
+++ b/net/minecraft/data/registries/RegistriesDatapackGenerator.java
@@ -52,7 +52,7 @@
                     PackOutput.PathProvider packoutput$pathprovider = this.output.createRegistryElementsPathProvider(resourcekey);
                     return CompletableFuture.allOf(
                         registry.listElements()
-                            .map(e -> dumpValue(packoutput$pathprovider.json(e.key().identifier()), cache, writeOps, v.elementCodec(), e.value()))
+                            .<CompletableFuture<?>>map(e -> dumpValue(packoutput$pathprovider.json(e.key().identifier()), cache, writeOps, v.elementCodec(), e.value()))
                             .toArray(CompletableFuture[]::new)
                     );
                 }
