--- a/net/minecraft/resources/ResourceKey.java
+++ b/net/minecraft/resources/ResourceKey.java
@@ -27,7 +27,7 @@
    }
 
    private static <T> ResourceKey<T> m_135790_(ResourceLocation p_135791_, ResourceLocation p_135792_) {
-      return f_135775_.computeIfAbsent(new ResourceKey.InternKey(p_135791_, p_135792_), (p_258225_) -> {
+      return (ResourceKey<T>)f_135775_.computeIfAbsent(new ResourceKey.InternKey(p_135791_, p_135792_), (p_258225_) -> {
          return new ResourceKey(p_258225_.f_256880_, p_258225_.f_257046_);
       });
    }
@@ -46,7 +46,7 @@
    }
 
    public <E> Optional<ResourceKey<E>> m_195975_(ResourceKey<? extends Registry<E>> p_195976_) {
-      return this.m_135783_(p_195976_) ? Optional.of(this) : Optional.empty();
+      return this.m_135783_(p_195976_) ? Optional.of((ResourceKey<E>)this) : Optional.empty();
    }
 
    public ResourceLocation m_135782_() {
