--- a/net/minecraft/world/ticks/SavedTick.java
+++ b/net/minecraft/world/ticks/SavedTick.java
@@ -10,7 +10,7 @@
 import net.minecraft.nbt.ListTag;
 import net.minecraft.world.level.ChunkPos;
 
-record SavedTick(T f_193311_, BlockPos f_193312_, int f_193313_, TickPriority f_193314_) {
+record SavedTick<T>(T f_193311_, BlockPos f_193312_, int f_193313_, TickPriority f_193314_) {
    private static final String f_193315_ = "i";
    private static final String f_193316_ = "x";
    private static final String f_193317_ = "y";
