--- a/net/minecraft/client/gui/screens/inventory/BookViewScreen.java
+++ b/net/minecraft/client/gui/screens/inventory/BookViewScreen.java
@@ -267,7 +267,7 @@
             } else {
                 WritableBookContent writablebookcontent = p_98309_.m_318834_(DataComponents.f_314472_);
                 return writablebookcontent != null
-                    ? new BookViewScreen.BookAccess(writablebookcontent.m_320046_(flag).map(Component::m_237113_).toList())
+                    ? new BookViewScreen.BookAccess(writablebookcontent.m_320046_(flag).<Component>map(Component::m_237113_).toList())
                     : null;
             }
         }
