--- a/net/minecraft/client/particle/DripParticle.java
+++ b/net/minecraft/client/particle/DripParticle.java
@@ -69,6 +69,7 @@
       private final IParticleData field_217579_C;
 
       private Dripping(World p_i50509_1_, double p_i50509_2_, double p_i50509_4_, double p_i50509_6_, Fluid p_i50509_8_, IParticleData p_i50509_9_) {
+         super(p_i50509_1_, p_i50509_2_, p_i50509_4_, p_i50509_6_, p_i50509_8_);
          this.field_217579_C = p_i50509_9_;
          this.field_70545_g *= 0.02F;
          this.field_70547_e = 40;
@@ -92,6 +93,7 @@
    @OnlyIn(Dist.CLIENT)
    static class DrippingLava extends DripParticle.Dripping {
       private DrippingLava(World p_i50513_1_, double p_i50513_2_, double p_i50513_4_, double p_i50513_6_, Fluid p_i50513_8_, IParticleData p_i50513_9_) {
+         super(p_i50513_1_, p_i50513_2_, p_i50513_4_, p_i50513_6_, p_i50513_8_, p_i50513_9_);
       }
 
       protected void func_217576_g() {
@@ -138,6 +140,7 @@
       private final IParticleData field_217578_C;
 
       private Falling(World p_i50511_1_, double p_i50511_2_, double p_i50511_4_, double p_i50511_6_, Fluid p_i50511_8_, IParticleData p_i50511_9_) {
+         super(p_i50511_1_, p_i50511_2_, p_i50511_4_, p_i50511_6_, p_i50511_8_);
          this.field_217578_C = p_i50511_9_;
          this.field_70547_e = (int)(64.0D / (Math.random() * 0.8D + 0.2D));
       }
@@ -186,6 +189,7 @@
    @OnlyIn(Dist.CLIENT)
    static class Landing extends DripParticle {
       private Landing(World p_i50507_1_, double p_i50507_2_, double p_i50507_4_, double p_i50507_6_, Fluid p_i50507_8_) {
+         super(p_i50507_1_, p_i50507_2_, p_i50507_4_, p_i50507_6_, p_i50507_8_);
          this.field_70547_e = (int)(16.0D / (Math.random() * 0.8D + 0.2D));
       }
    }
