--- a/net/minecraft/server/packs/linkfs/LinkFSProvider.java
+++ b/net/minecraft/server/packs/linkfs/LinkFSProvider.java
@@ -75,7 +75,7 @@
                      throw new DirectoryIteratorException(ioexception);
                   }
                }).map((p_249891_) -> {
-                  return p_249891_;
+                  return (Path) p_249891_;
                }).iterator();
             }
 
