--- a/net\minecraft\nbt\JsonToNBT.java
+++ b/net\minecraft\nbt\JsonToNBT.java
@@ -237,11 +237,11 @@
             }
 
             if (p_193603_2_ == 1) {
-               list.add((T)((NBTPrimitive)inbtbase).func_150290_f());
+               list.add((T)Byte.valueOf(((NBTPrimitive)inbtbase).func_150290_f()));
             } else if (p_193603_2_ == 4) {
-               list.add((T)((NBTPrimitive)inbtbase).func_150291_c());
+               list.add((T)Long.valueOf(((NBTPrimitive)inbtbase).func_150291_c()));
             } else {
-               list.add((T)((NBTPrimitive)inbtbase).func_150287_d());
+               list.add((T)Integer.valueOf(((NBTPrimitive)inbtbase).func_150287_d()));
             }
 
             if (this.func_193613_m()) {
