--- a/net/minecraft/world/entity/animal/armadillo/Armadillo.java
+++ b/net/minecraft/world/entity/animal/armadillo/Armadillo.java
@@ -58,7 +58,7 @@
     public static final int SCARE_CHECK_INTERVAL = 80;
     private static final double SCARE_DISTANCE_HORIZONTAL = 7.0;
     private static final double SCARE_DISTANCE_VERTICAL = 2.0;
-    private static final Brain.Provider<Armadillo> BRAIN_PROVIDER = Brain.provider(
+    private static final Brain.Provider<Armadillo> BRAIN_PROVIDER = Brain.<Armadillo>provider(
         List.of(
             SensorType.NEAREST_LIVING_ENTITIES, SensorType.HURT_BY, SensorType.FOOD_TEMPTATIONS, SensorType.NEAREST_ADULT, SensorType.ARMADILLO_SCARE_DETECTED
         ),
@@ -131,7 +131,7 @@
 
     @Override
     public Brain<Armadillo> getBrain() {
-        return super.getBrain();
+        return (Brain<Armadillo>)super.getBrain();
     }
 
     @Override
