--- a/net/minecraft/world/server/ChunkManager.java
+++ b/net/minecraft/world/server/ChunkManager.java
@@ -564,7 +564,7 @@
                chunk.func_177417_c(true);
                this.field_219255_i.func_147448_a(chunk.func_177434_r().values());
                List<Entity> list = null;
-               ClassInheritanceMultiMap[] aclassinheritancemultimap = chunk.func_177429_s();
+               ClassInheritanceMultiMap<Entity>[] aclassinheritancemultimap = chunk.func_177429_s();
                int i = aclassinheritancemultimap.length;
 
                for(int j = 0; j < i; ++j) {
