--- a/net/minecraft/world/entity/Leashable.java
+++ b/net/minecraft/world/entity/Leashable.java
@@ -43,7 +43,7 @@
 
     default void m_339199_(int p_345000_) {
         this.m_338401_(new Leashable.LeashData(p_345000_));
-        m_339838_((Entity)this, false, false);
+        m_339838_((Entity & Leashable)this, false, false);
     }
 
     default void m_372826_(CompoundTag p_376404_) {
@@ -82,11 +82,11 @@
     }
 
     default void m_339020_() {
-        m_339838_((Entity)this, true, true);
+        m_339838_((Entity & Leashable)this, true, true);
     }
 
     default void m_372320_() {
-        m_339838_((Entity)this, true, false);
+        m_339838_((Entity & Leashable)this, true, false);
     }
 
     default void m_372585_() {
@@ -155,7 +155,7 @@
     }
 
     default void m_339176_(Entity p_342226_, float p_342283_) {
-        m_338797_((Entity)this, p_342226_, p_342283_);
+        m_338797_((Entity & Leashable)this, p_342226_, p_342283_);
     }
 
     private static <E extends Entity & Leashable> void m_338797_(E p_342325_, Entity p_343749_, float p_343654_) {
@@ -166,7 +166,7 @@
     }
 
     default void m_340273_(Entity p_342408_, boolean p_342255_) {
-        m_339256_((Entity)this, p_342408_, p_342255_);
+        m_339256_((Entity & Leashable)this, p_342408_, p_342255_);
     }
 
     private static <E extends Entity & Leashable> void m_339256_(E p_342775_, Entity p_342643_, boolean p_343557_) {
@@ -189,7 +189,7 @@
 
     @Nullable
     default Entity m_340614_() {
-        return m_339450_((Entity)this);
+        return m_339450_((Entity & Leashable)this);
     }
 
     @Nullable
