--- a/net/minecraft/core/BlockPos.java
+++ b/net/minecraft/core/BlockPos.java
@@ -557,9 +557,9 @@
             private int secondIndex;
             private int thirdIndex;
             private boolean end;
-            private final int firstDirX = direction.getStepX();
-            private final int firstDirY = direction.getStepY();
-            private final int firstDirZ = direction.getStepZ();
+            private final int firstDirX = directionx.getStepX();
+            private final int firstDirY = directionx.getStepY();
+            private final int firstDirZ = directionx.getStepZ();
             private final int secondDirX = direction1.getStepX();
             private final int secondDirY = direction1.getStepY();
             private final int secondDirZ = direction1.getStepZ();
