--- a/net/minecraft/server/level/ServerLevel.java
+++ b/net/minecraft/server/level/ServerLevel.java
@@ -1368,7 +1368,7 @@
             object2intopenhashmap.addTo(s, 1);
          }
 
-         return object2intopenhashmap.object2IntEntrySet().stream().sorted(Comparator.comparing(Entry::getIntValue).reversed()).limit(5L).map((p_184087_) -> {
+         return object2intopenhashmap.object2IntEntrySet().stream().sorted(Comparator.<Entry<String>,Integer>comparing(Entry::getIntValue).reversed()).limit(5L).map((p_184087_) -> {
             return (String)p_184087_.getKey() + ":" + p_184087_.getIntValue();
          }).collect(Collectors.joining(","));
       } catch (Exception exception) {
