--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -1004,7 +1004,7 @@
         if (carried.has(DataComponents.WRITABLE_BOOK_CONTENT)) {
             ItemStack writtenBook = carried.transmuteCopy(Items.WRITTEN_BOOK);
             writtenBook.remove(DataComponents.WRITABLE_BOOK_CONTENT);
-            List<Filterable<Component>> pages = contents.stream().map(page -> this.filterableFromOutgoing(page).map(Component::literal)).toList();
+            List<Filterable<Component>> pages = contents.stream().map(page -> this.filterableFromOutgoing(page).<Component>map(Component::literal)).toList();
             writtenBook.set(
                 DataComponents.WRITTEN_BOOK_CONTENT, new WrittenBookContent(this.filterableFromOutgoing(title), this.player.getPlainTextName(), 0, pages, true)
             );
