--- a/net/minecraft/world/gen/feature/template/Template.java
+++ b/net/minecraft/world/gen/feature/template/Template.java
@@ -119,7 +119,7 @@
    }
 
    private static List<Template.BlockInfo> func_237151_a_(List<Template.BlockInfo> p_237151_0_, List<Template.BlockInfo> p_237151_1_, List<Template.BlockInfo> p_237151_2_) {
-      Comparator<Template.BlockInfo> comparator = Comparator.comparingInt((p_237154_0_) -> {
+      Comparator<Template.BlockInfo> comparator = Comparator.<Template.BlockInfo>comparingInt((p_237154_0_) -> {
          return p_237154_0_.field_186242_a.func_177956_o();
       }).thenComparingInt((p_237153_0_) -> {
          return p_237153_0_.field_186242_a.func_177958_n();
