--- a/net/minecraft/world/inventory/AbstractContainerMenu.java
+++ b/net/minecraft/world/inventory/AbstractContainerMenu.java
@@ -447,7 +447,7 @@
                             Optional<ItemStack> newCarried = slotx.tryRemove(amount, Integer.MAX_VALUE, player);
                             newCarried.ifPresent(itemsTaken -> {
                                 this.setCarried(itemsTaken);
-                                slot.onTake(player, itemsTaken);
+                                slotx.onTake(player, itemsTaken);
                             });
                         } else if (slotx.mayPlace(carried)) {
                             if (ItemStack.isSameItemSameComponents(clicked, carried)) {
@@ -461,7 +461,7 @@
                             Optional<ItemStack> newCarried = slotx.tryRemove(clicked.getCount(), carried.getMaxStackSize() - carried.getCount(), player);
                             newCarried.ifPresent(itemsTaken -> {
                                 carried.grow(itemsTaken.getCount());
-                                slot.onTake(player, itemsTaken);
+                                slotx.onTake(player, itemsTaken);
                             });
                         }
                     }
