--- a/net/minecraft/world/level/entity/EntitySection.java
+++ b/net/minecraft/world/level/entity/EntitySection.java
@@ -44,7 +44,7 @@
       } else {
          for(T t : collection) {
             U u = (U)((EntityAccess)p_188349_.m_141992_(t));
-            if (u != null && t.m_20191_().m_82381_(p_188350_) && p_261535_.m_260972_((T)u).m_261146_()) {
+            if (u != null && t.m_20191_().m_82381_(p_188350_) && p_261535_.m_260972_(u).m_261146_()) {
                return AbortableIterationConsumer.Continuation.ABORT;
             }
          }
