--- a/net/minecraft/world/level/entity/EntitySectionStorage.java
+++ b/net/minecraft/world/level/entity/EntitySectionStorage.java
@@ -107,7 +107,7 @@
 
     public LongSet m_156857_() {
         LongSet longset = new LongOpenHashSet();
-        this.f_156852_.keySet().forEach(p_156886_ -> longset.add(m_156899_(p_156886_)));
+        this.f_156852_.keySet().forEach((long p_156886_) -> longset.add(m_156899_(p_156886_)));
         return longset;
     }
 
