--- a/net/minecraft/world/level/levelgen/structure/StructureStart.java
+++ b/net/minecraft/world/level/levelgen/structure/StructureStart.java
@@ -17,7 +17,7 @@
 
 public final class StructureStart<C extends FeatureConfiguration> {
    public static final String f_163590_ = "INVALID";
-   public static final StructureStart<?> f_73561_ = new StructureStart((StructureFeature<C>)null, new ChunkPos(0, 0), 0, new PiecesContainer(List.of()));
+   public static final StructureStart<?> f_73561_ = new StructureStart(null, new ChunkPos(0, 0), 0, new PiecesContainer(List.of()));
    private final StructureFeature<C> f_73565_;
    private final PiecesContainer f_192654_;
    private final ChunkPos f_163592_;
