--- a/net\minecraft\client\gui\recipebook\GuiRecipeBook.java
+++ b/net\minecraft\client\gui\recipebook\GuiRecipeBook.java
@@ -371,7 +371,7 @@
 
    public boolean keyReleased(int p_keyReleased_1_, int p_keyReleased_2_, int p_keyReleased_3_) {
       this.field_199738_u = false;
-      return super.keyReleased(p_keyReleased_1_, p_keyReleased_2_, p_keyReleased_3_);
+      return IGuiEventListener.super.keyReleased(p_keyReleased_1_, p_keyReleased_2_, p_keyReleased_3_);
    }
 
    public boolean charTyped(char p_charTyped_1_, int p_charTyped_2_) {
@@ -382,7 +382,7 @@
             this.func_195603_h();
             return true;
          } else {
-            return super.charTyped(p_charTyped_1_, p_charTyped_2_);
+            return IGuiEventListener.super.charTyped(p_charTyped_1_, p_charTyped_2_);
          }
       } else {
          return false;
