--- a/net/minecraft/server/packs/resources/FallbackResourceManager.java
+++ b/net/minecraft/server/packs/resources/FallbackResourceManager.java
@@ -147,6 +147,8 @@
    }
 
    public Map<ResourceLocation, Resource> m_214159_(String p_215413_, Predicate<ResourceLocation> p_215414_) {
+      record ResourceWithSourceAndIndex(PackResources f_243853_, IoSupplier<InputStream> f_244005_, int f_244110_) {
+      }
       Map<ResourceLocation, ResourceWithSourceAndIndex> map = new HashMap<>();
       Map<ResourceLocation, ResourceWithSourceAndIndex> map1 = new HashMap<>();
       int i = this.f_10599_.size();
@@ -159,9 +161,6 @@
          if (packresources != null) {
             int k = j;
             packresources.m_8031_(this.f_10601_, this.f_10602_, p_215413_, (p_248254_, p_248255_) -> {
-               record ResourceWithSourceAndIndex(PackResources f_243853_, IoSupplier<InputStream> f_244005_, int f_244110_) {
-               }
-
                if (m_245103_(p_248254_)) {
                   if (p_215414_.test(m_246569_(p_248254_))) {
                      map1.put(p_248254_, new ResourceWithSourceAndIndex(packresources, p_248255_, k));
