--- a/net/minecraft/item/crafting/RecipeManager.java
+++ b/net/minecraft/item/crafting/RecipeManager.java
@@ -76,7 +76,7 @@
 
    public <C extends IInventory, T extends IRecipe<C>> List<T> func_241447_a_(IRecipeType<T> p_241447_1_) {
       return this.func_215366_a(p_241447_1_).values().stream().map((p_241453_0_) -> {
-         return p_241453_0_;
+         return (T) p_241453_0_;
       }).collect(Collectors.toList());
    }
 
@@ -89,7 +89,7 @@
    }
 
    private <C extends IInventory, T extends IRecipe<C>> Map<ResourceLocation, IRecipe<C>> func_215366_a(IRecipeType<T> p_215366_1_) {
-      return this.field_199522_d.getOrDefault(p_215366_1_, Collections.emptyMap());
+      return (Map)this.field_199522_d.getOrDefault(p_215366_1_, Collections.emptyMap());
    }
 
    public <C extends IInventory, T extends IRecipe<C>> NonNullList<ItemStack> func_215369_c(IRecipeType<T> p_215369_1_, C p_215369_2_, World p_215369_3_) {
