--- a/com/mojang/realmsclient/RealmsMainScreen.java
+++ b/com/mojang/realmsclient/RealmsMainScreen.java
@@ -1114,9 +1114,12 @@
 
    class CloseButton extends Button {
       public CloseButton() {
-         super(RealmsMainScreen.this.m_86363_() + 4, RealmsMainScreen.this.m_86366_() + 4, 12, 12, new TranslatableComponent("mco.selectServer.close"), (p_86775_) -> {
+         super(RealmsMainScreen.this.m_86363_() + 4, RealmsMainScreen.this.m_86366_() + 4, 12, 12, new TranslatableComponent("mco.selectServer.close"), null);
+      }
+
+      @Override
+      public void m_5691_() {
             RealmsMainScreen.this.m_86360_();
-         });
       }
 
       public void m_6303_(PoseStack p_86777_, int p_86778_, int p_86779_, float p_86780_) {
@@ -1145,7 +1148,11 @@
 
    class NewsButton extends Button {
       public NewsButton() {
-         super(RealmsMainScreen.this.f_96543_ - 62, 6, 20, 20, new TranslatableComponent("mco.news"), (p_86804_) -> {
+         super(RealmsMainScreen.this.f_96543_ - 62, 6, 20, 20, new TranslatableComponent("mco.news"), null);
+      }
+
+      @Override
+      public void m_5691_() {
             if (RealmsMainScreen.this.f_86259_ != null) {
                Util.m_137581_().m_137646_(RealmsMainScreen.this.f_86259_);
                if (RealmsMainScreen.this.f_86258_) {
@@ -1156,7 +1163,6 @@
                }
 
             }
-         });
       }
 
       public void m_6303_(PoseStack p_86806_, int p_86807_, int p_86808_, float p_86809_) {
@@ -1391,9 +1397,12 @@
 
    class ShowPopupButton extends Button {
       public ShowPopupButton() {
-         super(RealmsMainScreen.this.f_96543_ - 37, 6, 20, 20, new TranslatableComponent("mco.selectServer.info"), (p_86897_) -> {
+         super(RealmsMainScreen.this.f_96543_ - 37, 6, 20, 20, new TranslatableComponent("mco.selectServer.info"), null);
+      }
+
+      @Override
+      public void m_5691_() {
             RealmsMainScreen.this.f_86295_ = !RealmsMainScreen.this.f_86295_;
-         });
       }
 
       public void m_6303_(PoseStack p_86899_, int p_86900_, int p_86901_, float p_86902_) {
