--- a/net/minecraft/network/chat/LastSeenMessages.java
+++ b/net/minecraft/network/chat/LastSeenMessages.java
@@ -42,7 +42,7 @@
         public static final LastSeenMessages.Packed EMPTY = new LastSeenMessages.Packed(List.of());
 
         public Packed(final FriendlyByteBuf input) {
-            this(input.readCollection(FriendlyByteBuf.limitValue(ArrayList::new, 20), MessageSignature.Packed::read));
+            this(input.<MessageSignature.Packed, List<MessageSignature.Packed>>readCollection(FriendlyByteBuf.limitValue(ArrayList::new, 20), MessageSignature.Packed::read));
         }
 
         public void write(final FriendlyByteBuf output) {
