--- a/net/minecraft/network/syncher/SynchedEntityData.java
+++ b/net/minecraft/network/syncher/SynchedEntityData.java
@@ -95,7 +95,7 @@
 
       SynchedEntityData.DataItem<T> dataitem;
       try {
-         dataitem = this.f_135345_.get(p_135380_.m_135015_());
+         dataitem = (SynchedEntityData.DataItem<T>)this.f_135345_.get(p_135380_.m_135015_());
       } catch (Throwable throwable) {
          CrashReport crashreport = CrashReport.m_127521_(throwable, "Getting synched entity data");
          CrashReportCategory crashreportcategory = crashreport.m_127514_("Synched entity data");
