--- a/net/minecraft/world/level/storage/loot/GsonAdapterFactory.java
+++ b/net/minecraft/world/level/storage/loot/GsonAdapterFactory.java
@@ -105,13 +105,13 @@
       public JsonElement serialize(E p_78852_, Type p_78853_, JsonSerializationContext p_78854_) {
          T t = this.f_78832_.apply(p_78852_);
          if (this.f_164993_ != null && this.f_164993_.getFirst() == t) {
-            return this.f_164993_.getSecond().m_142413_(p_78852_, p_78854_);
+            return ((InlineSerializer<E>)this.f_164993_.getSecond()).m_142413_(p_78852_, p_78854_);
          } else if (t == null) {
             throw new JsonSyntaxException("Unknown type: " + p_78852_);
          } else {
             JsonObject jsonobject = new JsonObject();
             jsonobject.addProperty(this.f_78831_, this.f_78829_.m_7981_(t).toString());
-            t.m_79331_().m_6170_(jsonobject, p_78852_, p_78854_);
+            ((Serializer<E>)t.m_79331_()).m_6170_(jsonobject, p_78852_, p_78854_);
             return jsonobject;
          }
       }
