--- a/net/minecraft/world/level/storage/loot/predicates/LootItemBlockStatePropertyCondition.java
+++ b/net/minecraft/world/level/storage/loot/predicates/LootItemBlockStatePropertyCondition.java
@@ -27,7 +27,7 @@
     private static DataResult<LootItemBlockStatePropertyCondition> validate(final LootItemBlockStatePropertyCondition condition) {
         return condition.properties()
             .flatMap(properties -> properties.checkState(condition.block().value().getStateDefinition()))
-            .map(name -> DataResult.error(() -> "Block " + condition.block() + " has no property" + name))
+            .map(name -> DataResult.<LootItemBlockStatePropertyCondition>error(() -> "Block " + condition.block() + " has no property" + name))
             .orElse(DataResult.success(condition));
     }
 
