--- a/net/minecraft/util/RegistryKey.java
+++ b/net/minecraft/util/RegistryKey.java
@@ -21,7 +21,7 @@
 
    private static <T> RegistryKey<T> func_240905_a_(ResourceLocation p_240905_0_, ResourceLocation p_240905_1_) {
       String s = (p_240905_0_ + ":" + p_240905_1_).intern();
-      return field_240898_a_.computeIfAbsent(s, (p_240906_2_) -> {
+      return (RegistryKey<T>)field_240898_a_.computeIfAbsent(s, (p_240906_2_) -> {
          return new RegistryKey(p_240905_0_, p_240905_1_);
       });
    }
