--- a/net/minecraft/resources/ResourceKey.java
+++ b/net/minecraft/resources/ResourceKey.java
@@ -21,7 +21,7 @@
 
    private static <T> ResourceKey<T> m_135790_(ResourceLocation p_135791_, ResourceLocation p_135792_) {
       String s = (p_135791_ + ":" + p_135792_).intern();
-      return f_135775_.computeIfAbsent(s, (p_135796_) -> {
+      return (ResourceKey<T>)f_135775_.computeIfAbsent(s, (p_135796_) -> {
          return new ResourceKey(p_135791_, p_135792_);
       });
    }
