--- a/net/minecraft/world/level/entity/EntitySection.java
+++ b/net/minecraft/world/level/entity/EntitySection.java
@@ -42,7 +42,7 @@
          for(T t : collection) {
             U u = (U)((EntityAccess)p_188349_.m_141992_(t));
             if (u != null && t.m_142469_().m_82381_(p_188350_)) {
-               p_188351_.accept((T)u);
+               p_188351_.accept(u);
             }
          }
 
