--- a/net/minecraft/network/FriendlyByteBuf.java
+++ b/net/minecraft/network/FriendlyByteBuf.java
@@ -137,7 +137,7 @@
 
    public void m_178345_(IntList p_178346_) {
       this.m_130130_(p_178346_.size());
-      p_178346_.forEach(this::m_130130_);
+      p_178346_.forEach((java.util.function.IntConsumer)this::m_130130_);
    }
 
    public <K, V, M extends Map<K, V>> M m_178374_(IntFunction<M> p_178375_, Function<FriendlyByteBuf, K> p_178376_, Function<FriendlyByteBuf, V> p_178377_) {
