--- a/net/minecraft/world/level/levelgen/feature/MultifaceGrowthFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/MultifaceGrowthFeature.java
@@ -65,7 +65,7 @@
         } else if (!(this.placeBlock instanceof MultifaceSpreadeableBlock placerBlock)) {
             return false;
         } else {
-            List var13 = this.getShuffledDirections(random);
+            List<Direction> var13 = this.getShuffledDirections(random);
             if (this.placeGrowthIfPossible(placerBlock, level, origin, level.getBlockState(origin), random, var13)) {
                 return true;
             }
