--- a/net/minecraft/client/gui/GuiGraphics.java
+++ b/net/minecraft/client/gui/GuiGraphics.java
@@ -1018,65 +1018,24 @@
         }
     }
 
-    // $VF: Inserted dummy exception handlers to handle obfuscated exceptions
     public void m_280304_(Font p_282584_, @Nullable Style p_282156_, int p_283623_, int p_282114_) {
         if (p_282156_ != null && p_282156_.m_131186_() != null) {
-            HoverEvent hoverevent = p_282156_.m_131186_();
-            Objects.requireNonNull(hoverevent);
-            Object object = hoverevent;
-            Throwable throwable4;
-            switch (object) {
-                case HoverEvent.ShowItem hoverevent$showitem:
-                    HoverEvent.ShowItem hoverevent$showitem1 = hoverevent$showitem;
-
-                    try {
-                        itemstack1 = hoverevent$showitem1.f_381055_();
-                    } catch (Throwable throwable2) {
-                        throwable4 = throwable2;
-                        boolean flag1 = false;
-                        break;
-                    }
-
-                    ItemStack itemstack = itemstack1;
+            switch (p_282156_.m_131186_()) {
+                case HoverEvent.ShowItem(ItemStack itemstack):
                     this.m_280153_(p_282584_, itemstack, p_283623_, p_282114_);
                     return;
-                case HoverEvent.ShowEntity hoverevent$showentity:
-                    HoverEvent.ShowEntity hoverevent$showentity1 = hoverevent$showentity;
-
-                    try {
-                        hoverevent$entitytooltipinfo1 = hoverevent$showentity1.f_379711_();
-                    } catch (Throwable throwable1) {
-                        throwable4 = throwable1;
-                        boolean flag = false;
-                        break;
-                    }
-
-                    HoverEvent.EntityTooltipInfo hoverevent$entitytooltipinfo = hoverevent$entitytooltipinfo1;
+                case HoverEvent.ShowEntity(HoverEvent.EntityTooltipInfo hoverevent$entitytooltipinfo):
                     if (this.f_279544_.f_91066_.f_92125_) {
                         this.m_280666_(p_282584_, hoverevent$entitytooltipinfo.m_130884_(), p_283623_, p_282114_);
                     }
 
                     return;
-                case HoverEvent.ShowText hoverevent$showtext:
-                    HoverEvent.ShowText hoverevent$showtext1 = hoverevent$showtext;
-
-                    try {
-                        component1 = hoverevent$showtext1.f_382704_();
-                    } catch (Throwable throwable) {
-                        throwable4 = throwable;
-                        boolean flag2 = false;
-                        break;
-                    }
-
-                    Component component = component1;
+                case HoverEvent.ShowText(Component component):
                     this.m_280245_(p_282584_, p_282584_.m_92923_(component, Math.max(this.m_280182_() / 2, 200)), p_283623_, p_282114_);
                     return;
                 default:
                     return;
             }
-
-            Throwable throwable3 = throwable4;
-            throw new MatchException(throwable3.toString(), throwable3);
         }
     }
 
