--- a/net/minecraft/inventory/container/RecipeBookContainer.java
+++ b/net/minecraft/inventory/container/RecipeBookContainer.java
@@ -12,7 +12,7 @@
    }
 
    public void func_217056_a(boolean p_217056_1_, IRecipe<?> p_217056_2_, ServerPlayerEntity p_217056_3_) {
-      (new ServerRecipePlacer<>(this)).func_194327_a(p_217056_3_, p_217056_2_, p_217056_1_);
+      (new ServerRecipePlacer<>(this)).func_194327_a(p_217056_3_, (IRecipe<C>)p_217056_2_, p_217056_1_);
    }
 
    public abstract void func_201771_a(RecipeItemHelper p_201771_1_);
