--- a/net/minecraft/data/RegistryDumpReport.java
+++ b/net/minecraft/data/RegistryDumpReport.java
@@ -34,7 +34,7 @@
          jsonobject.addProperty("default", resourcelocation.toString());
       }
 
-      int j = Registry.field_212617_f.func_148757_b(p_239828_0_);
+      int j = ((Registry)Registry.field_212617_f).func_148757_b(p_239828_0_);
       jsonobject.addProperty("protocol_id", j);
       JsonObject jsonobject1 = new JsonObject();
 
