--- a/net/minecraft/inventory/container/AbstractFurnaceContainer.java
+++ b/net/minecraft/inventory/container/AbstractFurnaceContainer.java
@@ -66,7 +66,7 @@
    }
 
    public void func_217056_a(boolean p_217056_1_, IRecipe<?> p_217056_2_, ServerPlayerEntity p_217056_3_) {
-      (new ServerRecipePlacerFurnace<>(this)).func_194327_a(p_217056_3_, p_217056_2_, p_217056_1_);
+      (new ServerRecipePlacerFurnace<>(this)).func_194327_a(p_217056_3_, (IRecipe<IInventory>) p_217056_2_, p_217056_1_);
    }
 
    public boolean func_201769_a(IRecipe<? super IInventory> p_201769_1_) {
@@ -143,7 +143,7 @@
    }
 
    protected boolean func_217057_a(ItemStack p_217057_1_) {
-      return this.field_217062_c.func_199532_z().func_215371_a(this.field_217065_f, new Inventory(p_217057_1_), this.field_217062_c).isPresent();
+      return this.field_217062_c.func_199532_z().func_215371_a((IRecipeType)this.field_217065_f, new Inventory(p_217057_1_), this.field_217062_c).isPresent();
    }
 
    protected boolean func_217058_b(ItemStack p_217058_1_) {
