--- a/net/minecraft/world/raid/Raid.java
+++ b/net/minecraft/world/raid/Raid.java
@@ -520,10 +520,9 @@
 
    public float func_221290_r() {
       float f = 0.0F;
-      Iterator iterator = this.field_221344_h.values().iterator();
 
-      while(iterator.hasNext()) {
-         for(AbstractRaiderEntity abstractraiderentity : (Set)iterator.next()) {
+      for (Set<AbstractRaiderEntity> subset : this.field_221344_h.values()) {
+         for(AbstractRaiderEntity abstractraiderentity : subset) {
             f += abstractraiderentity.func_110143_aJ();
          }
       }
