--- a/net/minecraft/block/state/BlockStateContainer.java
+++ b/net/minecraft/block/state/BlockStateContainer.java
@@ -78,7 +78,7 @@
          blockstatecontainer$stateimplementation1.func_177235_a(map2);
       }
 
-      this.field_177625_e = ImmutableList.copyOf(list1);
+      this.field_177625_e = ImmutableList.<IBlockState>copyOf(list1);
    }
 
    public static <T extends Comparable<T>> String func_185919_a(Block p_185919_0_, IProperty<T> p_185919_1_) {
@@ -108,7 +108,7 @@
 
       while(unmodifiableiterator.hasNext()) {
          IProperty<?> iproperty = (IProperty)unmodifiableiterator.next();
-         list.add(iproperty.func_177700_c());
+         list.add(((IProperty)iproperty).func_177700_c());
       }
 
       return list;
