--- a/net/minecraft/util/ExtraCodecs.java
+++ b/net/minecraft/util/ExtraCodecs.java
@@ -289,7 +289,7 @@
             P max = l.get(1);
             return makeInterval.apply(min, max);
         }), p -> ImmutableList.of(getMin.apply((I)p), getMax.apply((I)p)));
-        Codec<I> objectCodec = RecordCodecBuilder.<Pair>create(
+        Codec<I> objectCodec = RecordCodecBuilder.<Pair<P, P>>create(
                 i -> i.group(pointCodec.fieldOf(lowerBoundName).forGetter(Pair::getFirst), pointCodec.fieldOf(upperBoundName).forGetter(Pair::getSecond))
                     .apply(i, Pair::of)
             )
@@ -614,7 +614,7 @@
             public <T> RecordBuilder<T> encode(final V input, final DynamicOps<T> ops, final RecordBuilder<T> builder) {
                 K type = (K)typeGetter.apply(input);
                 builder.add(typeKey, typeCodec.encodeStart(ops, type));
-                DataResult<T> parameters = this.encode(valueCodec.apply(type), input, ops);
+                DataResult<T> parameters = this.encode((Codec)valueCodec.apply(type), input, ops);
                 if (parameters.result().isEmpty() || !Objects.equals(parameters.result().get(), ops.emptyMap())) {
                     builder.add(valueKey, parameters);
                 }
