--- a/net/minecraft/util/ExtraCodecs.java
+++ b/net/minecraft/util/ExtraCodecs.java
@@ -285,7 +285,7 @@
             P p1 = l.get(1);
             return makeInterval.apply(p, p1);
         }), p -> ImmutableList.of(getMin.apply((I)p), getMax.apply((I)p)));
-        Codec<I> codec1 = RecordCodecBuilder.<Pair>create(
+        Codec<I> codec1 = RecordCodecBuilder.<Pair<P, P>>create(
                 i -> i.group(pointCodec.fieldOf(lowerBoundName).forGetter(Pair::getFirst), pointCodec.fieldOf(upperBoundName).forGetter(Pair::getSecond))
                     .apply(i, Pair::of)
             )
@@ -607,7 +607,7 @@
             public <T> RecordBuilder<T> encode(final V input, final DynamicOps<T> ops, final RecordBuilder<T> builder) {
                 K k = (K)typeGetter.apply(input);
                 builder.add(typeKey, typeCodec.encodeStart(ops, k));
-                DataResult<T> dataresult = this.encode(valueCodec.apply(k), input, ops);
+                DataResult<T> dataresult = this.encode((Codec)valueCodec.apply(k), input, ops);
                 if (dataresult.result().isEmpty() || !Objects.equals(dataresult.result().get(), ops.emptyMap())) {
                     builder.add(valueKey, dataresult);
                 }
