--- a/net/minecraft/network/protocol/game/ClientboundLoginPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundLoginPacket.java
@@ -12,7 +12,7 @@
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.dimension.DimensionType;
 
-public record ClientboundLoginPacket(int f_132360_, boolean f_132362_, GameType f_132363_, GameType f_132364_, Set<ResourceKey<Level>> f_132365_, RegistryAccess.RegistryHolder f_132366_, DimensionType f_132367_, ResourceKey<Level> f_132368_, long f_132361_, int f_132369_, int f_132370_, int f_195761_, boolean f_132371_, boolean f_132372_, boolean f_132373_, boolean f_132374_) implements Packet {
+public record ClientboundLoginPacket(int f_132360_, boolean f_132362_, GameType f_132363_, GameType f_132364_, Set<ResourceKey<Level>> f_132365_, RegistryAccess.RegistryHolder f_132366_, DimensionType f_132367_, ResourceKey<Level> f_132368_, long f_132361_, int f_132369_, int f_132370_, int f_195761_, boolean f_132371_, boolean f_132372_, boolean f_132373_, boolean f_132374_) implements Packet<ClientGamePacketListener> {
    public ClientboundLoginPacket(FriendlyByteBuf p_178960_) {
       this(p_178960_.readInt(), p_178960_.readBoolean(), GameType.m_46393_(p_178960_.readByte()), GameType.m_151497_(p_178960_.readByte()), p_178960_.m_178371_(Sets::newHashSetWithExpectedSize, (p_178965_) -> {
          return ResourceKey.m_135785_(Registry.f_122819_, p_178965_.m_130281_());
