--- a/net/minecraft/world/entity/monster/piglin/PiglinAi.java
+++ b/net/minecraft/world/entity/monster/piglin/PiglinAi.java
@@ -100,27 +100,27 @@
    }
 
    private static void m_34820_(Brain<Piglin> p_34821_) {
-      p_34821_.m_21891_(Activity.f_37978_, 0, ImmutableList.of(new LookAtTargetSink(45, 90), new MoveToTargetSink(), new InteractWithDoor(), m_34937_(), m_34955_(), new StopHoldingItemIfNoLongerAdmiring(), new StartAdmiringItemIfSeen(120), new StartCelebratingIfTargetDead(300, PiglinAi::m_34810_), new StopBeingAngryIfTargetDead()));
+      p_34821_.m_21891_(Activity.f_37978_, 0, ImmutableList.of(new LookAtTargetSink(45, 90), new MoveToTargetSink(), new InteractWithDoor(), m_34937_(), m_34955_(), new StopHoldingItemIfNoLongerAdmiring<>(), new StartAdmiringItemIfSeen<>(120), new StartCelebratingIfTargetDead(300, PiglinAi::m_34810_), new StopBeingAngryIfTargetDead<>()));
    }
 
    private static void m_34891_(Brain<Piglin> p_34892_) {
-      p_34892_.m_21891_(Activity.f_37979_, 10, ImmutableList.of(new SetEntityLookTarget(PiglinAi::m_34883_, 14.0F), new StartAttacking<>(AbstractPiglin::m_34667_, PiglinAi::m_35000_), new RunIf(Piglin::m_7121_, new StartHuntingHoglin<>()), m_34917_(), m_34970_(), m_34805_(), m_34882_(), new SetLookAndInteract(EntityType.f_20532_, 4)));
+      p_34892_.m_21891_(Activity.f_37979_, 10, ImmutableList.of(new SetEntityLookTarget(PiglinAi::m_34883_, 14.0F), new StartAttacking<>(AbstractPiglin::m_34667_, PiglinAi::m_35000_), new RunIf<>(Piglin::m_7121_, new StartHuntingHoglin<>()), m_34917_(), m_34970_(), m_34805_(), m_34882_(), new SetLookAndInteract(EntityType.f_20532_, 4)));
    }
 
    private static void m_34903_(Piglin p_34904_, Brain<Piglin> p_34905_) {
-      p_34905_.m_21895_(Activity.f_37988_, 10, ImmutableList.of(new StopAttackingIfTargetInvalid<>((p_34981_) -> {
+      p_34905_.m_21895_(Activity.f_37988_, 10, ImmutableList.<net.minecraft.world.entity.ai.behavior.Behavior<? super Piglin>>of(new StopAttackingIfTargetInvalid<Piglin>((p_34981_) -> {
          return !m_34900_(p_34904_, p_34981_);
-      }), new RunIf(PiglinAi::m_34918_, new BackUpIfTooClose<>(5, 0.75F)), new SetWalkTargetFromAttackTargetIfTargetOutOfReach(1.0F), new MeleeAttack(20), new CrossbowAttack(), new RememberIfHoglinWasKilled(), new EraseMemoryIf(PiglinAi::m_34998_, MemoryModuleType.f_26372_)), MemoryModuleType.f_26372_);
+      }), new RunIf<>(PiglinAi::m_34918_, new BackUpIfTooClose<>(5, 0.75F)), new SetWalkTargetFromAttackTargetIfTargetOutOfReach(1.0F), new MeleeAttack(20), new CrossbowAttack(), new RememberIfHoglinWasKilled(), new EraseMemoryIf<>(PiglinAi::m_34998_, MemoryModuleType.f_26372_)), MemoryModuleType.f_26372_);
    }
 
    private static void m_34920_(Brain<Piglin> p_34921_) {
       p_34921_.m_21895_(Activity.f_37989_, 10, ImmutableList.of(m_34917_(), new SetEntityLookTarget(PiglinAi::m_34883_, 14.0F), new StartAttacking<Piglin>(AbstractPiglin::m_34667_, PiglinAi::m_35000_), new RunIf<Piglin>((p_34804_) -> {
          return !p_34804_.m_34771_();
-      }, new GoToCelebrateLocation<>(2, 1.0F)), new RunIf<Piglin>(Piglin::m_34771_, new GoToCelebrateLocation<>(4, 0.6F)), new RunOne(ImmutableList.of(Pair.of(new SetEntityLookTarget(EntityType.f_20511_, 8.0F), 1), Pair.of(new RandomStroll(0.6F, 2, 1), 1), Pair.of(new DoNothing(10, 20), 1)))), MemoryModuleType.f_26341_);
+      }, new GoToCelebrateLocation<>(2, 1.0F)), new RunIf<Piglin>(Piglin::m_34771_, new GoToCelebrateLocation<>(4, 0.6F)), new RunOne<>(ImmutableList.of(Pair.of(new SetEntityLookTarget(EntityType.f_20511_, 8.0F), 1), Pair.of(new RandomStroll(0.6F, 2, 1), 1), Pair.of(new DoNothing(10, 20), 1)))), MemoryModuleType.f_26341_);
    }
 
    private static void m_34940_(Brain<Piglin> p_34941_) {
-      p_34941_.m_21895_(Activity.f_37990_, 10, ImmutableList.of(new GoToWantedItem<>(PiglinAi::m_35028_, 1.0F, true, 9), new StopAdmiringIfItemTooFarAway(9), new StopAdmiringIfTiredOfTryingToReachItem(200, 200)), MemoryModuleType.f_26336_);
+      p_34941_.m_21895_(Activity.f_37990_, 10, ImmutableList.of(new GoToWantedItem<>(PiglinAi::m_35028_, 1.0F, true, 9), new StopAdmiringIfItemTooFarAway<>(9), new StopAdmiringIfTiredOfTryingToReachItem<>(200, 200)), MemoryModuleType.f_26336_);
    }
 
    private static void m_34958_(Brain<Piglin> p_34959_) {
@@ -128,7 +128,7 @@
    }
 
    private static void m_34973_(Brain<Piglin> p_34974_) {
-      p_34974_.m_21895_(Activity.f_37992_, 10, ImmutableList.of(new Mount<>(0.8F), new SetEntityLookTarget(PiglinAi::m_34883_, 8.0F), new RunIf(Entity::m_20159_, m_34805_()), new DismountOrSkipMounting(8, PiglinAi::m_34834_)), MemoryModuleType.f_26376_);
+      p_34974_.m_21895_(Activity.f_37992_, 10, ImmutableList.of(new Mount<>(0.8F), new SetEntityLookTarget(PiglinAi::m_34883_, 8.0F), new RunIf<>(Entity::m_20159_, m_34805_()), new DismountOrSkipMounting<>(8, PiglinAi::m_34834_)), MemoryModuleType.f_26376_);
    }
 
    private static RunOne<Piglin> m_34805_() {
