--- a/net/minecraft/world/level/pathfinder/PathFinder.java
+++ b/net/minecraft/world/level/pathfinder/PathFinder.java
@@ -75,7 +75,7 @@
         this.openSet.clear();
         this.openSet.insert(from);
         boolean captureDebug = this.captureDebug.getAsBoolean();
-        Set<Node> closedSet = (Set<Node>)(captureDebug ? new HashSet<>() : Set.of());
+        Set<Node> closedSet = (captureDebug ? new HashSet<>() : Set.of());
         int count = 0;
         Set<Target> reachedTargets = Sets.newHashSetWithExpectedSize(targets.size());
         int maxVisitedNodesAdjusted = (int)(this.maxVisitedNodes * maxVisitedNodesMultiplier);
