--- a/net/minecraft/core/RegistrySynchronization.java
+++ b/net/minecraft/core/RegistrySynchronization.java
@@ -45,7 +45,7 @@
 
    private static <E> DataResult<? extends Codec<E>> m_245699_(ResourceKey<? extends Registry<E>> p_252190_) {
       return Optional.ofNullable(f_244438_.get(p_252190_)).map((p_250582_) -> {
-         return p_250582_.f_244392_();
+         return (Codec<E>)p_250582_.f_244392_();
       }).map(DataResult::success).orElseGet(() -> {
          return DataResult.error(() -> {
             return "Unknown or not serializable registry: " + p_252190_;
@@ -69,9 +69,9 @@
    private static <K extends ResourceKey<? extends Registry<?>>, V extends Registry<?>> Codec<RegistryAccess> m_245940_(UnboundedMapCodec<K, V> p_249934_) {
       return p_249934_.xmap(RegistryAccess.ImmutableRegistryAccess::new, (p_251578_) -> {
          return m_247199_(p_251578_).collect(ImmutableMap.toImmutableMap((p_250395_) -> {
-            return p_250395_.f_206233_();
+            return (K)p_250395_.f_206233_();
          }, (p_248951_) -> {
-            return p_248951_.f_206234_();
+            return (V)p_248951_.f_206234_();
          }));
       });
    }
