--- a/net/minecraft/world/entity/animal/Sheep.java
+++ b/net/minecraft/world/entity/animal/Sheep.java
@@ -82,7 +82,7 @@
       p_29841_.put(DyeColor.RED, Blocks.f_50108_);
       p_29841_.put(DyeColor.BLACK, Blocks.f_50109_);
    });
-   private static final Map<DyeColor, float[]> f_29801_ = Maps.newEnumMap(Arrays.stream(DyeColor.values()).collect(Collectors.toMap((p_29868_) -> {
+   private static final Map<DyeColor, float[]> f_29801_ = Maps.<DyeColor, float[]>newEnumMap(Arrays.stream(DyeColor.values()).collect(Collectors.toMap((p_29868_) -> {
       return p_29868_;
    }, Sheep::m_29865_)));
    private int f_29802_;
