--- a/net/minecraft/world/entity/animal/allay/Allay.java
+++ b/net/minecraft/world/entity/animal/allay/Allay.java
@@ -78,7 +78,7 @@
     public static final int MAX_NOTEBLOCK_DISTANCE = 1024;
     private static final EntityDataAccessor<Boolean> DATA_DANCING = SynchedEntityData.defineId(Allay.class, EntityDataSerializers.BOOLEAN);
     private static final EntityDataAccessor<Boolean> DATA_CAN_DUPLICATE = SynchedEntityData.defineId(Allay.class, EntityDataSerializers.BOOLEAN);
-    private static final Brain.Provider<Allay> BRAIN_PROVIDER = Brain.provider(
+    private static final Brain.Provider<Allay> BRAIN_PROVIDER = Brain.<Allay>provider(
         List.of(MemoryModuleType.LIKED_PLAYER, MemoryModuleType.LIKED_NOTEBLOCK_POSITION, MemoryModuleType.LIKED_NOTEBLOCK_COOLDOWN_TICKS),
         List.of(SensorType.NEAREST_LIVING_ENTITIES, SensorType.NEAREST_PLAYERS, SensorType.HURT_BY, SensorType.NEAREST_ITEMS),
         var0 -> AllayAi.getActivities()
@@ -118,7 +118,7 @@
 
     @Override
     public Brain<Allay> getBrain() {
-        return super.getBrain();
+        return (Brain<Allay>)super.getBrain();
     }
 
     public static AttributeSupplier.Builder createAttributes() {
