--- a/net/minecraft/client/gui/font/FontSet.java
+++ b/net/minecraft/client/gui/font/FontSet.java
@@ -55,7 +55,7 @@
       }
 
       Set<GlyphProvider> set = Sets.newHashSet();
-      intset.forEach((p_222277_) -> {
+      intset.forEach((int p_222277_) -> {
          for(GlyphProvider glyphprovider1 : p_95072_) {
             GlyphInfo glyphinfo = glyphprovider1.m_213395_(p_222277_);
             if (glyphinfo != null) {
