--- a/net/minecraft/world/level/block/Block.java
+++ b/net/minecraft/world/level/block/Block.java
@@ -605,7 +605,7 @@
     }
 
     private static <S extends StateHolder<?, S>, T extends Comparable<T>> S setValueHelper(final S state, final Property<T> property, final Object value) {
-        return state.setValue(property, (Comparable)value);
+        return state.setValue(property, (T)value);
     }
 
     @Deprecated
