--- a/net/minecraft/world/item/HoneycombItem.java
+++ b/net/minecraft/world/item/HoneycombItem.java
@@ -71,7 +71,7 @@
         Level level = p_150869_.m_43725_();
         BlockPos blockpos = p_150869_.m_8083_();
         BlockState blockstate = level.m_8055_(blockpos);
-        return m_150878_(blockstate).map(p_359407_ -> {
+        return m_150878_(blockstate).<InteractionResult>map(p_359407_ -> {
             Player player = p_150869_.m_43723_();
             ItemStack itemstack = p_150869_.m_43722_();
             if (player instanceof ServerPlayer serverplayer) {
