--- a/net/minecraft/world/item/component/KineticWeapon.java
+++ b/net/minecraft/world/item/component/KineticWeapon.java
@@ -112,7 +112,7 @@
             for (EntityHitResult hitResult : ProjectileUtil.getHitEntitiesAlong(
                     livingEntity, attackRange, e -> PiercingWeapon.canHitEntity(livingEntity, e), ClipContext.Block.COLLIDER
                 )
-                .map(a -> List.of(), e -> (Collection)e)) {
+                .map(a -> List.<EntityHitResult>of(), e -> (Collection<EntityHitResult>)e)) {
                 Entity otherEntity = hitResult.getEntity();
                 if (otherEntity instanceof EnderDragonPart dragonPart) {
                     otherEntity = dragonPart.parentMob;
