--- a/net\minecraft\world\gen\FlatGenSettings.java
+++ b/net\minecraft\world\gen\FlatGenSettings.java
@@ -269,8 +269,8 @@
       T t = p_210834_1_.createList(this.field_82655_a.stream().<T>map((p_210837_1_) -> {
          return p_210834_1_.createMap(ImmutableMap.<T, T>of(p_210834_1_.createString("height"), p_210834_1_.createInt(p_210837_1_.func_82657_a()), p_210834_1_.createString("block"), p_210834_1_.createString(((ResourceLocation)Block.field_149771_c.func_177774_c(p_210837_1_.func_175900_c().func_177230_c())).toString())));
       }));
-      T t1 = p_210834_1_.createMap((Map)this.field_82653_b.entrySet().stream().map((p_210833_1_) -> {
-         return Pair.of(p_210834_1_.createString(((String)p_210833_1_.getKey()).toLowerCase(Locale.ROOT)), p_210834_1_.createMap((Map)((Map)p_210833_1_.getValue()).entrySet().stream().map((p_210836_1_) -> {
+      T t1 = p_210834_1_.createMap(this.field_82653_b.entrySet().stream().map((p_210833_1_) -> {
+         return Pair.of(p_210834_1_.createString(((String)p_210833_1_.getKey()).toLowerCase(Locale.ROOT)), p_210834_1_.createMap((p_210833_1_.getValue()).entrySet().stream().map((p_210836_1_) -> {
             return Pair.of(p_210834_1_.createString((String)p_210836_1_.getKey()), p_210834_1_.createString((String)p_210836_1_.getValue()));
          }).collect(Collectors.toMap(Pair::getFirst, Pair::getSecond))));
       }).collect(Collectors.toMap(Pair::getFirst, Pair::getSecond)));
@@ -279,7 +279,7 @@
 
    public static FlatGenSettings func_210835_a(Dynamic<?> p_210835_0_) {
       FlatGenSettings flatgensettings = ChunkGeneratorType.field_205489_f.func_205483_a();
-      List<Pair<Integer, Block>> list = (List)((Stream)p_210835_0_.get("layers").<Pair<Integer, Block>>flatMap(Dynamic::getStream).orElse(Stream.empty())).<Pair<Integer, Block>>map((p_210838_0_) -> {
+      List<Pair<Integer, Block>> list = (List)(p_210835_0_.get("layers").flatMap(Dynamic::getStream).orElse(Stream.empty())).<Pair<Integer, Block>>map((p_210838_0_) -> {
          return Pair.of(p_210838_0_.getInt("height", 1), Block.func_149684_b(p_210838_0_.getString("block")));
       }).collect(Collectors.toList());
       if (list.stream().anyMatch((p_211743_0_) -> {
