--- a/net/minecraft/world/entity/animal/camel/CamelAi.java
+++ b/net/minecraft/world/entity/animal/camel/CamelAi.java
@@ -47,7 +47,7 @@
     }
 
     private static ActivityData<Camel> initCoreActivity() {
-        return ActivityData.create(
+        return ActivityData.<Camel>create(
             Activity.CORE,
             0,
             ImmutableList.of(
@@ -62,7 +62,7 @@
     }
 
     private static ActivityData<Camel> initIdleActivity() {
-        return ActivityData.create(
+        return ActivityData.<Camel>create(
             Activity.IDLE,
             ImmutableList.of(
                 Pair.of(0, SetEntityLookTargetSometimes.create(EntityType.PLAYER, 6.0F, UniformInt.of(30, 60))),
