--- a/net/minecraft/client/main/Main.java
+++ b/net/minecraft/client/main/Main.java
@@ -187,7 +187,7 @@
       };
       thread.setUncaughtExceptionHandler(new DefaultUncaughtExceptionHandler(f_129630_));
       Runtime.getRuntime().addShutdownHook(thread);
-      final Minecraft minecraft = null;
+      Minecraft minecraft = null;
 
       try {
          Thread.currentThread().setName("Render thread");
@@ -208,14 +208,14 @@
          return;
       }
 
-      Minecraft minecraft1 = minecraft;
+      final Minecraft minecraft1 = minecraft;
       Thread thread1;
       if (minecraft.m_91267_()) {
          thread1 = new Thread("Game thread") {
             public void run() {
                try {
                   RenderSystem.initGameThread(true);
-                  minecraft.m_91374_();
+                  minecraft1.m_91374_();
                } catch (Throwable throwable3) {
                   Main.f_129630_.error("Exception in client thread", throwable3);
                }
