--- a/net/minecraft/world/entity/monster/piglin/StartHuntingHoglin.java
+++ b/net/minecraft/world/entity/monster/piglin/StartHuntingHoglin.java
@@ -18,12 +18,12 @@
                     i.registered(MemoryModuleType.NEAREST_VISIBLE_ADULT_PIGLINS)
                 )
                 .apply(i, (huntable, angryAt, huntedRecently, nearestPiglins) -> (level, body, timestamp) -> {
-                    if (!body.isBaby() && !i.<List>tryGet(nearestPiglins).filter(p -> p.stream().anyMatch(StartHuntingHoglin::hasHuntedRecently)).isPresent()) {
+                    if (!body.isBaby() && !i.tryGet(nearestPiglins).filter(p -> p.stream().anyMatch(StartHuntingHoglin::hasHuntedRecently)).isPresent()) {
                         Hoglin hoglin = i.get(huntable);
                         PiglinAi.setAngerTarget(level, body, hoglin);
                         PiglinAi.dontKillAnyMoreHoglinsForAWhile(body);
                         PiglinAi.broadcastAngerTarget(level, body, hoglin);
-                        i.<List>tryGet(nearestPiglins).ifPresent(p -> p.forEach(PiglinAi::dontKillAnyMoreHoglinsForAWhile));
+                        i.tryGet(nearestPiglins).ifPresent(p -> p.forEach(PiglinAi::dontKillAnyMoreHoglinsForAWhile));
                         return true;
                     } else {
                         return false;
