--- a/net/minecraft/entity/passive/SheepEntity.java
+++ b/net/minecraft/entity/passive/SheepEntity.java
@@ -81,7 +81,7 @@
       p_203402_0_.put(DyeColor.RED, Blocks.field_196570_aZ);
       p_203402_0_.put(DyeColor.BLACK, Blocks.field_196602_ba);
    });
-   private static final Map<DyeColor, float[]> field_175514_bm = Maps.newEnumMap(Arrays.stream(DyeColor.values()).collect(Collectors.toMap((p_200204_0_) -> {
+   private static final Map<DyeColor, float[]> field_175514_bm = Maps.newEnumMap(Arrays.stream(DyeColor.values()).collect(Collectors.toMap((DyeColor p_200204_0_) -> {
       return p_200204_0_;
    }, SheepEntity::func_192020_c)));
    private int field_70899_e;
