--- a/net/minecraft/client/gui/font/FontSet.java
+++ b/net/minecraft/client/gui/font/FontSet.java
@@ -58,7 +58,7 @@
       }
 
       Set<GlyphProvider> set = Sets.newHashSet();
-      intset.forEach((p_95076_) -> {
+      intset.forEach((int p_95076_) -> {
          for(GlyphProvider glyphprovider1 : p_95072_) {
             GlyphInfo glyphinfo = (GlyphInfo)(p_95076_ == 32 ? f_95049_ : glyphprovider1.m_7823_(p_95076_));
             if (glyphinfo != null) {
