--- a/net/minecraft/advancements/AdvancementList.java
+++ b/net/minecraft/advancements/AdvancementList.java
@@ -59,7 +59,7 @@
    }
 
    public void func_192083_a(Map<ResourceLocation, Advancement.Builder> p_192083_1_) {
-      Function<ResourceLocation, Advancement> function = Functions.<ResourceLocation, Advancement>forMap(this.field_192092_b, (Object)null);
+      Function<ResourceLocation, Advancement> function = Functions.<ResourceLocation, Advancement>forMap(this.field_192092_b, null);
 
       while(!p_192083_1_.isEmpty()) {
          boolean flag = false;
