--- a/net/minecraft/world/GameRules.java
+++ b/net/minecraft/world/GameRules.java
@@ -133,8 +133,8 @@
    }
 
    private static <T extends GameRules.RuleValue<T>> void func_234897_a_(GameRules.IRuleEntryVisitor p_234897_0_, GameRules.RuleKey<?> p_234897_1_, GameRules.RuleType<?> p_234897_2_) {
-      p_234897_0_.func_223481_a(p_234897_1_, p_234897_2_);
-      p_234897_2_.func_234914_a_(p_234897_0_, p_234897_1_);
+      p_234897_0_.func_223481_a((RuleKey)p_234897_1_, p_234897_2_);
+      p_234897_2_.func_234914_a_(p_234897_0_, (RuleKey)p_234897_1_);
    }
 
    public boolean func_223586_b(GameRules.RuleKey<GameRules.BooleanValue> p_223586_1_) {
