--- a/net/minecraft/world/level/biome/Climate.java
+++ b/net/minecraft/world/level/biome/Climate.java
@@ -202,7 +202,7 @@
                throw new IllegalStateException("Expecting parameter space to be 7, got " + i);
             } else {
                List<Climate.RTree.Leaf<T>> list = p_186936_.stream().map((p_186934_) -> {
-                  return new Climate.RTree.Leaf(p_186934_.getFirst(), p_186934_.getSecond());
+                  return new Climate.RTree.Leaf<T>(p_186934_.getFirst(), p_186934_.getSecond());
                }).collect(Collectors.toCollection(ArrayList::new));
                return new Climate.RTree<>(m_186920_(i, list));
             }
