--- a/net/minecraft/world/level/timers/TimerCallbacks.java
+++ b/net/minecraft/world/level/timers/TimerCallbacks.java
@@ -23,7 +23,7 @@
    }
 
    private <T extends TimerCallback<C>> TimerCallback.Serializer<C, T> m_82236_(Class<?> p_82237_) {
-      return this.f_82229_.get(p_82237_);
+      return (TimerCallback.Serializer<C, T>)this.f_82229_.get(p_82237_);
    }
 
    public <T extends TimerCallback<C>> CompoundTag m_82234_(T p_82235_) {
