--- a/net/minecraft/util/math/shapes/EntitySelectionContext.java
+++ b/net/minecraft/util/math/shapes/EntitySelectionContext.java
@@ -33,7 +33,7 @@
 
    @Deprecated
    protected EntitySelectionContext(Entity p_i51182_1_) {
-      this(p_i51182_1_.func_226274_bn_(), p_i51182_1_.func_226278_cu_(), p_i51182_1_ instanceof LivingEntity ? ((LivingEntity)p_i51182_1_).func_184614_ca().func_77973_b() : Items.field_190931_a, p_i51182_1_ instanceof LivingEntity ? (LivingEntity)p_i51182_1_::func_230285_a_ : (p_237494_0_) -> {
+      this(p_i51182_1_.func_226274_bn_(), p_i51182_1_.func_226278_cu_(), p_i51182_1_ instanceof LivingEntity ? ((LivingEntity)p_i51182_1_).func_184614_ca().func_77973_b() : Items.field_190931_a, p_i51182_1_ instanceof LivingEntity ? ((LivingEntity)p_i51182_1_)::func_230285_a_ : (p_237494_0_) -> {
          return false;
       });
    }
