--- a/net/minecraft/server/level/ChunkMap.java
+++ b/net/minecraft/server/level/ChunkMap.java
@@ -555,7 +555,7 @@
 
             return flag;
          });
-      }).thenApplyAsync((p_214891_) -> {
+      }).<Either<ChunkAccess, ChunkHolder.ChunkLoadingFailure>>thenApplyAsync((p_214891_) -> {
          this.f_140133_.m_46473_().m_6174_("chunkLoad");
          if (p_214891_.isPresent()) {
             ChunkAccess chunkaccess = ChunkSerializer.m_188230_(this.f_140133_, this.f_140138_, p_140418_, p_214891_.get());
