--- a/net/minecraft/advancements/predicates/entity/EntityFlagsPredicate.java
+++ b/net/minecraft/advancements/predicates/entity/EntityFlagsPredicate.java
@@ -70,7 +70,7 @@
         } else {
             return this.isFallFlying.isPresent() && entity instanceof LivingEntity living && living.isFallFlying() != this.isFallFlying.get()
                 ? false
-                : !(this.isBaby.isPresent() && entity instanceof LivingEntity living) || living.isBaby() == this.isBaby.get();
+                : !(this.isBaby.isPresent() && entity instanceof LivingEntity living2) || living2.isBaby() == this.isBaby.get();
         }
     }
 
