--- a/net\minecraft\util\Scheduler.java
+++ b/net\minecraft\util\Scheduler.java
@@ -31,8 +31,8 @@
    private final ExecutorService field_202857_c;
    private final AtomicInteger field_202858_d = new AtomicInteger(1);
    private final List<CompletableFuture<R>> field_202859_e = Lists.<CompletableFuture<R>>newArrayList();
-   private CompletableFuture<R> field_202860_f = CompletableFuture.<R>completedFuture((Object)null);
-   private CompletableFuture<R> field_202861_g = CompletableFuture.<R>completedFuture((Object)null);
+   private CompletableFuture<R> field_202860_f = CompletableFuture.<R>completedFuture(null);
+   private CompletableFuture<R> field_202861_g = CompletableFuture.<R>completedFuture(null);
    private final Supplier<Map<T, CompletableFuture<R>>> field_202862_h;
    private final Supplier<Map<T, CompletableFuture<Void>>> field_202863_i;
    private final T field_202864_j;
@@ -53,7 +53,7 @@
          this.field_202857_c = new ForkJoinPool(p_i48724_2_ - 1, (p_202853_2_) -> {
             return new ForkJoinWorkerThread(p_202853_2_) {
                {
-                  this.setName(p_202853_1_ + "-Worker-" + Scheduler.this.field_202858_d.getAndIncrement());
+                  this.setName(p_i48724_1_ + "-Worker-" + Scheduler.this.field_202858_d.getAndIncrement());
                }
             };
          }, (p_202848_0_, p_202848_1_) -> {
