--- a/net/minecraft/server/commands/data/DataCommands.java
+++ b/net/minecraft/server/commands/data/DataCommands.java
@@ -198,18 +198,7 @@
         Objects.requireNonNull(p_265255_);
 
         return switch (p_265255_) {
-            case StringTag stringtag -> {
-                StringTag stringtag1 = stringtag;
-
-                try {
-                    s1 = stringtag1.f_382711_();
-                } catch (Throwable throwable) {
-                    throw new MatchException(throwable.toString(), throwable);
-                }
-
-                String s = s1;
-                yield s;
-            }
+            case StringTag(String s) -> s;
             case PrimitiveTag primitivetag -> primitivetag.toString();
             default -> throw f_263767_.create(p_265255_);
         };
@@ -418,18 +407,7 @@
             case NumericTag numerictag -> Mth.m_14107_(numerictag.m_384364_());
             case CollectionTag collectiontag -> collectiontag.size();
             case CompoundTag compoundtag -> compoundtag.m_128440_();
-            case StringTag stringtag -> {
-                StringTag stringtag1 = stringtag;
-
-                try {
-                    s1 = stringtag1.f_382711_();
-                } catch (Throwable throwable) {
-                    throw new MatchException(throwable.toString(), throwable);
-                }
-
-                String s = s1;
-                yield s.length();
-            }
+            case StringTag(String s) -> s.length();
             case EndTag endtag -> throw f_139354_.create(p_139446_.toString());
             default -> throw new MatchException(null, null);
         };
