--- a/net/minecraft/profiler/LongTickDetector.java
+++ b/net/minecraft/profiler/LongTickDetector.java
@@ -16,6 +16,13 @@
    private final File field_233520_e_;
    private IResultableProfiler field_233521_f_;
 
+   private LongTickDetector() {
+      field_233517_b_ = null;
+      field_233518_c_ = 0L;
+      field_233520_e_ = null;
+      throw new IllegalStateException("This is a synthetic added by MCP do not call");
+   }
+
    public IProfiler func_233522_a_() {
       this.field_233521_f_ = new Profiler(this.field_233517_b_, () -> {
          return this.field_233519_d_;
