--- a/net/minecraft/client/gui/screens/worldselection/WorldGenSettingsComponent.java
+++ b/net/minecraft/client/gui/screens/worldselection/WorldGenSettingsComponent.java
@@ -186,7 +186,7 @@
    }
 
    private static Component m_229272_(Holder<WorldPreset> p_229273_) {
-      return p_229273_.m_203543_().map((p_216693_) -> {
+      return p_229273_.m_203543_().<Component>map((p_216693_) -> {
          return Component.m_229817_(p_216693_.m_135782_().m_222163_("generator"));
       }).orElse(f_101382_);
    }
