--- a/net/minecraft/world/IWorldReaderBase.java
+++ b/net/minecraft/world/IWorldReaderBase.java
@@ -198,8 +198,7 @@
          return ShapeUtils.func_197880_a();
       } else {
          voxelshape = voxelshape.func_197753_c();
-         blockpos$pooledmutableblockpos = new VoxelShapeInt(voxelshapepart, i, k, i1);
-         voxelshape = ShapeUtils.func_197882_b(voxelshape, blockpos$pooledmutableblockpos, IBooleanFunction.OR);
+         voxelshape = ShapeUtils.func_197882_b(voxelshape, new VoxelShapeInt(voxelshapepart, i, k, i1), IBooleanFunction.OR);
          return voxelshape;
       }
    }
@@ -213,7 +212,7 @@
       VoxelShape voxelshape = ShapeUtils.func_197881_a(p_211153_2_);
       VoxelShape voxelshape1 = ShapeUtils.func_197881_a(p_211153_2_.func_72317_d(p_211153_4_ > 0.0D ? -1.0E-7D : 1.0E-7D, p_211153_6_ > 0.0D ? -1.0E-7D : 1.0E-7D, p_211153_8_ > 0.0D ? -1.0E-7D : 1.0E-7D));
       VoxelShape voxelshape2 = ShapeUtils.func_197882_b(ShapeUtils.func_197881_a(p_211153_2_.func_72321_a(p_211153_4_, p_211153_6_, p_211153_8_).func_186662_g(1.0E-7D)), voxelshape1, IBooleanFunction.ONLY_FIRST);
-      return this.func_211154_a(p_211153_1_, voxelshape2, voxelshape, p_211153_3_, false);
+      return this.func_211154_a(p_211153_1_, voxelshape, voxelshape2, p_211153_3_, false); //MCP Note: Swapping these changes from strictly equal bytecode.. but it fixes falling through to world. Potential Vanilla bug.. or decompile bug elseware?
    }
 
    default VoxelShape func_195583_a(@Nullable Entity p_195583_1_, AxisAlignedBB p_195583_2_) {
