--- a/net/minecraft/resources/RegistryResourceAccess.java
+++ b/net/minecraft/resources/RegistryResourceAccess.java
@@ -151,7 +151,7 @@
          return this.f_195913_.entrySet().stream().filter((p_214277_) -> {
             return p_214277_.getKey().m_135783_(p_214274_);
          }).collect(Collectors.toMap((p_214287_) -> {
-            return p_214287_.getKey();
+            return (ResourceKey<E>)p_214287_.getKey();
          }, (p_214283_) -> {
             return p_214283_.getValue()::m_214288_;
          }));
