--- a/net/minecraft/client/renderer/tileentity/TileEntityRendererDispatcher.java
+++ b/net/minecraft/client/renderer/tileentity/TileEntityRendererDispatcher.java
@@ -79,11 +79,11 @@
    public <T extends TileEntity> TileEntityRenderer<T> func_147546_a(Class<? extends TileEntity> p_147546_1_) {
       TileEntityRenderer<? extends TileEntity> tileentityrenderer = this.field_147559_m.get(p_147546_1_);
       if (tileentityrenderer == null && p_147546_1_ != TileEntity.class) {
-         tileentityrenderer = this.func_147546_a(p_147546_1_.getSuperclass());
+         tileentityrenderer = this.func_147546_a((Class<? extends TileEntity>)p_147546_1_.getSuperclass());
          this.field_147559_m.put(p_147546_1_, tileentityrenderer);
       }
 
-      return tileentityrenderer;
+      return (TileEntityRenderer<T>)tileentityrenderer;
    }
 
    @Nullable
