--- a/net/minecraft/client/gui/Font.java
+++ b/net/minecraft/client/gui/Font.java
@@ -139,7 +139,7 @@
     ) {
         int i = m_92719_(p_168650_);
         Font.StringRenderOutput font$stringrenderoutput = new Font.StringRenderOutput(
-            this, p_168652_, 0.0F, 0.0F, i, false, p_254170_, Font.DisplayMode.NORMAL, p_168653_
+            p_168652_, 0.0F, 0.0F, i, false, p_254170_, Font.DisplayMode.NORMAL, p_168653_
         );
 
         for (int j = -1; j <= 1; j++) {
@@ -163,7 +163,7 @@
 
         font$stringrenderoutput.m_352913_();
         Font.StringRenderOutput font$stringrenderoutput1 = new Font.StringRenderOutput(
-            this, p_168652_, p_168647_, p_168648_, m_92719_(p_168649_), false, p_254170_, Font.DisplayMode.POLYGON_OFFSET, p_168653_
+            p_168652_, p_168647_, p_168648_, m_92719_(p_168649_), false, p_254170_, Font.DisplayMode.POLYGON_OFFSET, p_168653_
         );
         p_168646_.m_13731_(font$stringrenderoutput1);
         font$stringrenderoutput1.m_92961_(p_168647_);
@@ -235,7 +235,7 @@
         boolean p_369545_
     ) {
         Font.StringRenderOutput font$stringrenderoutput = new Font.StringRenderOutput(
-            this, p_272595_, p_273532_, p_272783_, p_273217_, p_273727_, p_273583_, p_272734_, p_273610_, p_273199_, p_369545_
+            p_272595_, p_273532_, p_272783_, p_273217_, p_273727_, p_273583_, p_272734_, p_273610_, p_273199_, p_369545_
         );
         StringDecomposer.m_14346_(p_273765_, Style.f_131099_, font$stringrenderoutput);
         return font$stringrenderoutput.m_92961_(p_273532_);
@@ -255,7 +255,7 @@
         boolean p_365371_
     ) {
         Font.StringRenderOutput font$stringrenderoutput = new Font.StringRenderOutput(
-            this, p_273392_, p_272632_, p_273541_, p_273200_, p_273774_, p_273312_, p_273276_, p_272625_, p_273371_, p_365371_
+            p_273392_, p_272632_, p_273541_, p_273200_, p_273774_, p_273312_, p_273276_, p_272625_, p_273371_, p_365371_
         );
         p_273322_.m_13731_(font$stringrenderoutput);
         return font$stringrenderoutput.m_92961_(p_272632_);
@@ -336,7 +336,6 @@
         }
 
         public StringRenderOutput(
-            final Font param1,
             final MultiBufferSource p_181365_,
             final float p_181366_,
             final float p_181367_,
@@ -346,11 +345,10 @@
             final Font.DisplayMode p_181371_,
             final int p_181372_
         ) {
-            this(var1, p_181365_, p_181366_, p_181367_, p_181368_, 0, p_181369_, p_254510_, p_181371_, p_181372_, true);
+            this(p_181365_, p_181366_, p_181367_, p_181368_, 0, p_181369_, p_254510_, p_181371_, p_181372_, true);
         }
 
         public StringRenderOutput(
-            final Font param1,
             final MultiBufferSource p_369661_,
             final float p_364376_,
             final float p_367232_,
@@ -362,7 +360,6 @@
             final int p_361543_,
             final boolean p_363799_
         ) {
-            this.f_92938_ = var1;
             this.f_347105_ = new ArrayList<>();
             this.f_92937_ = p_369661_;
             this.f_92948_ = p_364376_;
@@ -379,8 +376,8 @@
 
         @Override
         public boolean m_6411_(int p_92967_, Style p_92968_, int p_92969_) {
-            FontSet fontset = this.f_92938_.m_92863_(p_92968_.m_131192_());
-            GlyphInfo glyphinfo = fontset.m_243128_(p_92969_, this.f_92938_.f_242994_);
+            FontSet fontset = Font.this.m_92863_(p_92968_.m_131192_());
+            GlyphInfo glyphinfo = fontset.m_243128_(p_92969_, Font.this.f_242994_);
             BakedGlyph bakedglyph = p_92968_.m_131176_() && p_92969_ != 32 ? fontset.m_95067_(glyphinfo) : fontset.m_95078_(p_92969_);
             boolean flag = p_92968_.m_131154_();
             TextColor textcolor = p_92968_.m_131135_();
@@ -416,7 +413,7 @@
                 BakedGlyph.Effect bakedglyph$effect = new BakedGlyph.Effect(
                     p_92963_ - 1.0F, this.f_92949_ + 9.0F, this.f_92948_, this.f_92949_ - 1.0F, this.m_352635_(), this.f_347276_
                 );
-                bakedglyph = this.f_92938_.m_92863_(Style.f_131100_).m_95064_();
+                bakedglyph = Font.this.m_92863_(Style.f_131100_).m_95064_();
                 VertexConsumer vertexconsumer = this.f_92937_.m_6299_(bakedglyph.m_181387_(this.f_181362_));
                 bakedglyph.m_95220_(bakedglyph$effect, this.f_92945_, vertexconsumer, this.f_92947_);
             }
@@ -424,7 +421,7 @@
             this.m_352913_();
             if (this.f_92950_ != null) {
                 if (bakedglyph == null) {
-                    bakedglyph = this.f_92938_.m_92863_(Style.f_131100_).m_95064_();
+                    bakedglyph = Font.this.m_92863_(Style.f_131100_).m_95064_();
                 }
 
                 VertexConsumer vertexconsumer1 = this.f_92937_.m_6299_(bakedglyph.m_181387_(this.f_181362_));
