--- a/net/minecraft/util/datafix/fixes/EntityHorseSplitFix.java
+++ b/net/minecraft/util/datafix/fixes/EntityHorseSplitFix.java
@@ -38,7 +38,7 @@
 
          dynamic.remove("Type");
          Type<?> type = this.getOutputSchema().findChoiceType(References.f_16786_).types().get(s);
-         return Pair.of(s, (Typed)((Pair)p_15452_.write().flatMap(type::readTyped).result().orElseThrow(() -> {
+         return Pair.of(s, (Typed<?>)((Pair)((com.mojang.serialization.DataResult<Dynamic<?>>)p_15452_.write()).flatMap(type::readTyped).result().orElseThrow(() -> {
             return new IllegalStateException("Could not parse the new horse");
          })).getFirst());
       } else {
