--- a/net/minecraft/core/Registry.java
+++ b/net/minecraft/core/Registry.java
@@ -135,7 +135,7 @@
     Holder<T> wrapAsHolder(T value);
 
     default Iterable<Holder<T>> getTagOrEmpty(final TagKey<T> id) {
-        return DataFixUtils.orElse(this.get(id), List.<T>of());
+        return DataFixUtils.orElse((Optional<Iterable>)(Optional)this.get(id), List.<T>of());
     }
 
     Stream<HolderSet.Named<T>> getTags();
