--- a/net/minecraft/client/renderer/texture/TextureManager.java
+++ b/net/minecraft/client/renderer/texture/TextureManager.java
@@ -68,11 +68,12 @@
          this.field_110585_a.put(p_110579_1_, p_110579_2_);
          flag = false;
       } catch (Throwable throwable) {
+         ITextureObject p_110579_2_f = p_110579_2_;
          CrashReport crashreport = CrashReport.func_85055_a(throwable, "Registering texture");
          CrashReportCategory crashreportcategory = crashreport.func_85058_a("Resource location being registered");
          crashreportcategory.func_71507_a("Resource location", p_110579_1_);
          crashreportcategory.func_189529_a("Texture object class", () -> {
-            return p_110579_2_.getClass().getName();
+            return p_110579_2_f.getClass().getName();
          });
          throw new ReportedException(crashreport);
       }
