--- a/net/minecraft/world/gen/placement/Placement.java
+++ b/net/minecraft/world/gen/placement/Placement.java
@@ -46,7 +46,7 @@
 
    public Placement(Codec<DC> p_i232086_1_) {
       this.field_236961_M_ = p_i232086_1_.fieldOf("config").xmap((p_236966_1_) -> {
-         return new ConfiguredPlacement<>(this, p_236966_1_);
+         return new ConfiguredPlacement<DC>(this, p_236966_1_);
       }, ConfiguredPlacement::func_242877_b).codec();
    }
 
