--- a/net/minecraft/util/datafix/fixes/LevelFlatGeneratorInfoFix.java
+++ b/net/minecraft/util/datafix/fixes/LevelFlatGeneratorInfoFix.java
@@ -36,7 +36,7 @@
 
    private Dynamic<?> m_16352_(Dynamic<?> p_16353_) {
       return p_16353_.get("generatorName").asString("").equalsIgnoreCase("flat") ? p_16353_.update("generatorOptions", (p_16357_) -> {
-         return DataFixUtils.orElse(p_16357_.asString().map(this::m_16354_).map(p_16357_::createString).result(), p_16357_);
+         return DataFixUtils.orElse(p_16357_.asString().map(this::m_16354_).map(s -> p_16357_.createString(s)).result(), p_16357_);
       }) : p_16353_;
    }
 
