--- a/net/minecraft/world/entity/animal/axolotl/AxolotlAi.java
+++ b/net/minecraft/world/entity/animal/axolotl/AxolotlAi.java
@@ -73,7 +73,7 @@
             Activity.f_37988_,
             0,
             ImmutableList.of(
-                StopAttackingIfTargetInvalid.<Mob>m_257940_(Axolotl::m_218443_),
+                StopAttackingIfTargetInvalid.m_257940_(Axolotl::m_218443_),
                 SetWalkTargetFromAttackTargetIfTargetOutOfReach.m_257648_(AxolotlAi::m_149288_),
                 MeleeAttack.m_257733_(20),
                 EraseMemoryIf.<Mob>m_258093_(BehaviorUtils::m_217126_, MemoryModuleType.f_26372_)
