--- a/net/minecraft/world/chunk/storage/IOWorker.java
+++ b/net/minecraft/world/chunk/storage/IOWorker.java
@@ -45,7 +45,7 @@
                if (p_227098_2_ != null) {
                   p_227094_3_.completeExceptionally(p_227098_2_);
                } else {
-                  p_227094_3_.complete((Object)null);
+                  p_227094_3_.complete(null);
                }
 
             });
@@ -102,7 +102,7 @@
                return new CompletableFuture[p_227089_0_];
             }));
             completablefuture.whenComplete((p_227097_1_, p_227097_2_) -> {
-               p_227096_1_.complete((Object)null);
+               p_227096_1_.complete(null);
             });
          };
       });
