--- a/net/minecraft/world/item/AdventureModePredicate.java
+++ b/net/minecraft/world/item/AdventureModePredicate.java
@@ -114,7 +114,7 @@
         return p_328947_.stream()
             .flatMap(p_449774_ -> p_449774_.f_434497_().orElseThrow().m_203614_())
             .distinct()
-            .map(p_335858_ -> p_335858_.m_203334_().m_49954_().m_130940_(ChatFormatting.DARK_GRAY))
+            .map(p_335858_ -> (Component)p_335858_.m_203334_().m_49954_().m_130940_(ChatFormatting.DARK_GRAY))
             .toList();
     }
 
