--- a/net/minecraft/client/renderer/entity/EntityRendererManager.java
+++ b/net/minecraft/client/renderer/entity/EntityRendererManager.java
@@ -201,9 +201,9 @@
       if (p_78713_1_ instanceof AbstractClientPlayerEntity) {
          String s = ((AbstractClientPlayerEntity)p_78713_1_).func_175154_l();
          PlayerRenderer playerrenderer = this.field_178636_l.get(s);
-         return playerrenderer != null ? playerrenderer : this.field_178637_m;
+         return (EntityRenderer<? super T>) (playerrenderer != null ? playerrenderer : this.field_178637_m);
       } else {
-         return this.field_78729_o.get(p_78713_1_.func_200600_R());
+         return (EntityRenderer<? super T>) this.field_78729_o.get(p_78713_1_.func_200600_R());
       }
    }
 
