--- a/net/minecraft/client/gui/fonts/FontResourceManager.java
+++ b/net/minecraft/client/gui/fonts/FontResourceManager.java
@@ -104,7 +104,7 @@
                intset.addAll(iglyphprovider1.func_230428_a_());
             }
 
-            intset.forEach((p_238555_1_) -> {
+            intset.forEach((int p_238555_1_) -> {
                if (p_238555_1_ != 32) {
                   for(IGlyphProvider iglyphprovider2 : Lists.reverse(list)) {
                      if (iglyphprovider2.func_212248_a(p_238555_1_) != null) {
