--- a/net/minecraft/world/entity/monster/piglin/PiglinAi.java
+++ b/net/minecraft/world/entity/monster/piglin/PiglinAi.java
@@ -180,7 +180,7 @@
                 new SpearAttack(1.0, 1.0, 10.0F, 2.0F),
                 new SpearRetreat(1.0),
                 MeleeAttack.m_257733_(20),
-                new CrossbowAttack(),
+                new CrossbowAttack<>(),
                 RememberIfHoglinWasKilled.m_257745_(),
                 EraseMemoryIf.m_258093_(PiglinAi::m_34998_, MemoryModuleType.f_26372_)
             ),
@@ -201,7 +201,7 @@
                 new RunOne<LivingEntity>(
                     ImmutableList.of(
                         Pair.of(SetEntityLookTarget.m_258096_(EntityType.f_20511_, 8.0F), 1),
-                        Pair.of(RandomStroll.m_257960_(0.6F, 2, 1), 1),
+                        Pair.of((BehaviorControl<LivingEntity>)(BehaviorControl)RandomStroll.m_257960_(0.6F, 2, 1), 1),
                         Pair.of(new DoNothing(10, 20), 1)
                     )
                 )
@@ -231,7 +231,7 @@
                 SetWalkTargetAwayFrom.m_257370_(MemoryModuleType.f_26383_, 1.0F, 12, true),
                 m_34805_(),
                 m_34882_(),
-                EraseMemoryIf.<PathfinderMob>m_258093_(PiglinAi::m_35008_, MemoryModuleType.f_26383_)
+                EraseMemoryIf.m_258093_(PiglinAi::m_35008_, MemoryModuleType.f_26383_)
             ),
             MemoryModuleType.f_26383_
         );
@@ -253,7 +253,7 @@
                             .build()
                     )
                 ),
-                DismountOrSkipMounting.<LivingEntity>m_257459_(8, PiglinAi::m_34834_)
+                DismountOrSkipMounting.m_257459_(8, PiglinAi::m_34834_)
             ),
             MemoryModuleType.f_26376_
         );
