--- a/com/mojang/blaze3d/font/TrueTypeGlyphProvider.java
+++ b/com/mojang/blaze3d/font/TrueTypeGlyphProvider.java
@@ -94,7 +94,7 @@
                      break label62;
                   }
 
-                  glyphinfo = () -> {
+                  glyphinfo = (GlyphInfo.SpaceGlyphInfo)() -> {
                      return f / this.f_83839_;
                   };
                } catch (Throwable throwable1) {
