--- a/net/minecraft/world/level/levelgen/feature/MultifaceGrowthFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/MultifaceGrowthFeature.java
@@ -27,7 +27,7 @@
         } else if (!(config.placeBlock instanceof MultifaceSpreadeableBlock placerBlock)) {
             return false;
         } else {
-            List var14 = config.getShuffledDirections(random);
+            List<Direction> var14 = config.getShuffledDirections(random);
             if (placeGrowthIfPossible(placerBlock, level, origin, level.getBlockState(origin), config, random, var14)) {
                 return true;
             }
