--- a/net/minecraft/util/datafix/fixes/EntityHorseSplitFix.java
+++ b/net/minecraft/util/datafix/fixes/EntityHorseSplitFix.java
@@ -17,8 +17,7 @@
     @Override
     protected Pair<String, Typed<?>> fix(final String name, final Typed<?> entity) {
         if (Objects.equals("EntityHorse", name)) {
-            Dynamic<?> dynamic = entity.get(DSL.remainderFinder());
-            int i = dynamic.get("Type").asInt(0);
+            int i = entity.get(DSL.remainderFinder()).get("Type").asInt(0);
 
             String s = switch (i) {
                 case 1 -> "Donkey";
