--- a/net/minecraft/client/particle/ParticleManager.java
+++ b/net/minecraft/client/particle/ParticleManager.java
@@ -163,7 +163,7 @@
          p_215226_3_.func_76319_b();
          p_215226_3_.func_219897_b();
          return atlastexture$sheetdata;
-      }, p_215226_5_).<Void>thenCompose(p_215226_1_::func_216872_a).thenAcceptAsync((p_215229_3_) -> {
+      }, p_215226_5_).thenCompose(p_215226_1_::func_216872_a).thenAcceptAsync((p_215229_3_) -> {
          p_215226_4_.func_219894_a();
          p_215226_4_.func_76320_a("upload");
          this.field_215243_j.func_215260_a(p_215229_3_);
@@ -230,7 +230,7 @@
 
    @Nullable
    private <T extends IParticleData> Particle func_199927_b(T p_199927_1_, double p_199927_2_, double p_199927_4_, double p_199927_6_, double p_199927_8_, double p_199927_10_, double p_199927_12_) {
-      IParticleFactory<T> iparticlefactory = this.field_178932_g.get(Registry.field_212632_u.func_148757_b(p_199927_1_.func_197554_b()));
+      IParticleFactory<T> iparticlefactory = (IParticleFactory<T>) this.field_178932_g.get(Registry.field_212632_u.func_148757_b(p_199927_1_.func_197554_b()));
       return iparticlefactory == null ? null : iparticlefactory.func_199234_a(p_199927_1_, this.field_78878_a, p_199927_2_, p_199927_4_, p_199927_6_, p_199927_8_, p_199927_10_, p_199927_12_);
    }
 
