--- a/net/minecraft/world/Teleporter.java
+++ b/net/minecraft/world/Teleporter.java
@@ -32,7 +32,7 @@
       pointofinterestmanager.func_226347_a_(this.field_85192_a, p_242957_1_, i);
       Optional<PointOfInterest> optional = pointofinterestmanager.func_226353_b_((p_242952_0_) -> {
          return p_242952_0_ == PointOfInterestType.field_226358_u_;
-      }, p_242957_1_, i, PointOfInterestManager.Status.ANY).sorted(Comparator.comparingDouble((p_242954_1_) -> {
+      }, p_242957_1_, i, PointOfInterestManager.Status.ANY).sorted(Comparator.<PointOfInterest>comparingDouble((p_242954_1_) -> {
          return p_242954_1_.func_218261_f().func_177951_i(p_242957_1_);
       }).thenComparingInt((p_242959_0_) -> {
          return p_242959_0_.func_218261_f().func_177956_o();
