--- a/net/minecraft/world/entity/ai/behavior/SetLookAndInteract.java
+++ b/net/minecraft/world/entity/ai/behavior/SetLookAndInteract.java
@@ -19,7 +19,7 @@
                     ix.present(MemoryModuleType.NEAREST_VISIBLE_LIVING_ENTITIES)
                 )
                 .apply(ix, (lookTarget, interactionTarget, nearestEntities) -> (level, body, timestamp) -> {
-                    Optional<LivingEntity> optional = i.<NearestVisibleLivingEntities>get(nearestEntities)
+                    Optional<LivingEntity> optional = ix.<NearestVisibleLivingEntities>get(nearestEntities)
                         .findClosest(e -> e.distanceToSqr(body) <= i && e.is(type));
                     if (optional.isEmpty()) {
                         return false;
