--- a/net/minecraft/world/phys/shapes/CollisionContext.java
+++ b/net/minecraft/world/phys/shapes/CollisionContext.java
@@ -38,7 +38,7 @@
             true,
             p_397199_ != null ? p_397199_.m_20186_() : -Double.MAX_VALUE,
             p_397199_ instanceof LivingEntity livingentity ? livingentity.m_21205_() : ItemStack.f_41583_,
-            p_397199_ instanceof LivingEntity livingentity1 ? p_394132_ -> livingentity1.m_203441_(p_394132_) : p_397477_ -> false,
+            p_397199_ instanceof LivingEntity livingentity1 ? p_394132_ -> ((LivingEntity)p_397199_).m_203441_(p_394132_) : p_397477_ -> false,
             p_397199_
         );
     }
