--- a/net/minecraft/client/multiplayer/ClientPacketListener.java
+++ b/net/minecraft/client/multiplayer/ClientPacketListener.java
@@ -508,7 +508,7 @@
 
    public void m_182047_(ClientboundRemoveEntitiesPacket p_182633_) {
       PacketUtils.m_131363_(p_182633_, this, this.f_104888_);
-      p_182633_.m_182730_().forEach((p_205521_) -> {
+      p_182633_.m_182730_().forEach((int p_205521_) -> {
          this.f_104889_.m_171642_(p_205521_, Entity.RemovalReason.DISCARDED);
       });
    }
@@ -1265,7 +1265,7 @@
             return new IllegalStateException("Unknown registry " + p_205561_);
          });
          Map<TagKey<T>, List<Holder<T>>> map = new HashMap<>();
-         TagNetworkSerialization.m_203952_(p_205561_, registry, p_205562_, map::put);
+         TagNetworkSerialization.m_203952_((ResourceKey<? extends Registry<T>>)p_205561_, registry, p_205562_, map::put);
          registry.m_203652_(map);
       }
    }
