--- a/net/minecraft/client/multiplayer/ClientPacketListener.java
+++ b/net/minecraft/client/multiplayer/ClientPacketListener.java
@@ -1270,7 +1270,7 @@
             return new IllegalStateException("Unknown registry " + p_205561_);
          });
          Map<TagKey<T>, List<Holder<T>>> map = new HashMap<>();
-         TagNetworkSerialization.m_203952_(p_205561_, registry, p_205562_, map::put);
+         TagNetworkSerialization.m_203952_((ResourceKey<? extends Registry<T>>)p_205561_, registry, p_205562_, map::put);
          registry.m_203652_(map);
       }
    }
