--- a/net/minecraft/realms/action/ConnectingToRealmsAction.java
+++ b/net/minecraft/realms/action/ConnectingToRealmsAction.java
@@ -91,6 +91,7 @@
             if (realmsserveraddress.field_230602_b_ != null && realmsserveraddress.field_230603_c_ != null) {
                ITextComponent itextcomponent = new TranslationTextComponent("mco.configure.world.resourcepack.question.line1");
                ITextComponent itextcomponent1 = new TranslationTextComponent("mco.configure.world.resourcepack.question.line2");
+               final RealmsServerAddress realmsserveraddress_f = realmsserveraddress;
                func_238127_a_(new RealmsLongConfirmationScreen((p_238121_2_) -> {
                   try {
                      if (p_238121_2_) {
@@ -102,8 +103,8 @@
                         };
 
                         try {
-                           Minecraft.func_71410_x().func_195541_I().func_217818_a(realmsserveraddress.field_230602_b_, realmsserveraddress.field_230603_c_).thenRun(() -> {
-                              this.func_224987_a(new RealmsLongRunningMcoTaskScreen(this.field_238117_d_, new ConnectedToRealmsAction(this.field_238117_d_, realmsserveraddress)));
+                           Minecraft.func_71410_x().func_195541_I().func_217818_a(realmsserveraddress_f.field_230602_b_, realmsserveraddress_f.field_230603_c_).thenRun(() -> {
+                              this.func_224987_a(new RealmsLongRunningMcoTaskScreen(this.field_238117_d_, new ConnectedToRealmsAction(this.field_238117_d_, realmsserveraddress_f)));
                            }).exceptionally(function);
                         } catch (Exception exception1) {
                            function.apply(exception1);
