--- a/net/minecraft/data/loot/LootTableProvider.java
+++ b/net/minecraft/data/loot/LootTableProvider.java
@@ -85,7 +85,7 @@
             problemreporter$collector.forEach((id, problem) -> LOGGER.warn("Found validation problem in {}: {}", id, problem.description()));
             throw new IllegalStateException("Failed to validate loot tables, see logs");
         } else {
-            return CompletableFuture.allOf(writableregistry.entrySet().stream().map(entry -> {
+            return CompletableFuture.allOf(writableregistry.entrySet().stream().<CompletableFuture<?>>map(entry -> {
                 ResourceKey<LootTable> resourcekey1 = entry.getKey();
                 LootTable loottable = entry.getValue();
                 Path path = this.pathProvider.json(resourcekey1.identifier());
