--- a/com/mojang/blaze3d/platform/InputConstants.java
+++ b/com/mojang/blaze3d/platform/InputConstants.java
@@ -200,7 +200,7 @@
 
    public static boolean m_84826_() {
       try {
-         return f_166354_ != null && f_166354_.invokeExact();
+         return f_166354_ != null && (boolean)f_166354_.invokeExact();
       } catch (Throwable throwable) {
          throw new RuntimeException(throwable);
       }
@@ -222,7 +222,7 @@
       try {
          methodhandle = lookup.findStatic(GLFW.class, "glfwRawMouseMotionSupported", methodtype);
          MethodHandle methodhandle1 = lookup.findStaticGetter(GLFW.class, "GLFW_RAW_MOUSE_MOTION", Integer.TYPE);
-         i = methodhandle1.invokeExact();
+         i = (int)methodhandle1.invokeExact();
       } catch (NoSuchFieldException | NoSuchMethodException nosuchmethodexception) {
       } catch (Throwable throwable) {
          throw new RuntimeException(throwable);
