--- a/net\minecraft\world\gen\feature\structure\OceanMonumentPieces.java
+++ b/net\minecraft\world\gen\feature\structure\OceanMonumentPieces.java
@@ -1262,7 +1262,7 @@
       protected static final IBlockState field_175824_d = field_175826_b;
       protected static final IBlockState field_175825_e = Blocks.field_180398_cJ.func_176223_P();
       protected static final IBlockState field_175822_f = Blocks.field_150355_j.func_176223_P();
-      protected static final Set<Block> field_212180_g = ImmutableSet.builder().add(Blocks.field_150432_aD).add(Blocks.field_150403_cj).add(Blocks.field_205164_gk).add(field_175822_f.func_177230_c()).build();
+      protected static final Set<Block> field_212180_g = ImmutableSet.<Block>builder().add(Blocks.field_150432_aD).add(Blocks.field_150403_cj).add(Blocks.field_205164_gk).add(field_175822_f.func_177230_c()).build();
       protected static final int field_175823_g = func_175820_a(2, 0, 0);
       protected static final int field_175831_h = func_175820_a(2, 2, 0);
       protected static final int field_175832_i = func_175820_a(0, 1, 0);
