--- a/net/minecraft/util/datafix/fixes/AddNewChoices.java
+++ b/net/minecraft/util/datafix/fixes/AddNewChoices.java
@@ -28,10 +28,10 @@
       } else {
          return this.fixTypeEverywhere(p_14638_, p_14639_, p_14640_, (p_14636_) -> {
             return (p_14633_) -> {
-               if (!p_14640_.hasType(p_14633_.getFirst())) {
+               if (!((TaggedChoiceType<K>)p_14640_).hasType(p_14633_.getFirst())) {
                   throw new IllegalArgumentException(String.format("Unknown type %s in %s ", p_14633_.getFirst(), this.f_14626_));
                } else {
-                  return p_14633_;
+                  return (com.mojang.datafixers.util.Pair)p_14633_;
                }
             };
          });
