--- a/net/minecraft/util/datafix/fixes/EntityRenameFix.java
+++ b/net/minecraft/util/datafix/fixes/EntityRenameFix.java
@@ -18,8 +18,8 @@
    }
 
    public TypeRewriteRule makeRule() {
-      TaggedChoiceType<String> taggedchoicetype = this.getInputSchema().findChoiceType(References.f_16786_);
-      TaggedChoiceType<String> taggedchoicetype1 = this.getOutputSchema().findChoiceType(References.f_16786_);
+      TaggedChoiceType<String> taggedchoicetype = (TaggedChoiceType<String>)this.getInputSchema().findChoiceType(References.f_16786_);
+      TaggedChoiceType<String> taggedchoicetype1 = (TaggedChoiceType<String>)this.getOutputSchema().findChoiceType(References.f_16786_);
       return this.fixTypeEverywhere(this.f_15616_, taggedchoicetype, taggedchoicetype1, (p_15624_) -> {
          return (p_15629_) -> {
             String s = p_15629_.getFirst();
