--- a/net/minecraft/util/datafix/fixes/EntityMinecartIdentifiersFix.java
+++ b/net/minecraft/util/datafix/fixes/EntityMinecartIdentifiersFix.java
@@ -20,8 +20,8 @@
    }
 
    public TypeRewriteRule makeRule() {
-      TaggedChoiceType<String> taggedchoicetype = this.getInputSchema().findChoiceType(References.f_16786_);
-      TaggedChoiceType<String> taggedchoicetype1 = this.getOutputSchema().findChoiceType(References.f_16786_);
+      TaggedChoiceType<String> taggedchoicetype = (TaggedChoiceType<String>)this.getInputSchema().findChoiceType(References.f_16786_);
+      TaggedChoiceType<String> taggedchoicetype1 = (TaggedChoiceType<String>)this.getOutputSchema().findChoiceType(References.f_16786_);
       return this.fixTypeEverywhere("EntityMinecartIdentifiersFix", taggedchoicetype, taggedchoicetype1, (p_15485_) -> {
          return (p_15490_) -> {
             if (!Objects.equals(p_15490_.getFirst(), "Minecart")) {
@@ -37,7 +37,7 @@
                   s = "MinecartRideable";
                }
 
-               return Pair.of(s, typed.write().map((p_15494_) -> {
+               return Pair.of(s, typed.write().<com.mojang.serialization.DataResult>map((p_15494_) -> {
                   return taggedchoicetype1.types().get(s).read(p_15494_);
                }).result().orElseThrow(() -> {
                   return new IllegalStateException("Could not read the new minecart.");
