--- a/net/minecraft/world/level/biome/Climate.java
+++ b/net/minecraft/world/level/biome/Climate.java
@@ -269,7 +269,7 @@
                     throw new IllegalStateException("Expecting parameter space to be 7, got " + i);
                 } else {
                     List<Climate.RTree.Leaf<T>> list = values.stream()
-                        .map(p -> new Climate.RTree.Leaf(p.getFirst(), p.getSecond()))
+                        .map(p -> new Climate.RTree.Leaf<T>(p.getFirst(), p.getSecond()))
                         .collect(Collectors.toCollection(ArrayList::new));
                     return new Climate.RTree<>(build(i, list));
                 }
