--- a/net/minecraft/world/SerializableTickList.java
+++ b/net/minecraft/world/SerializableTickList.java
@@ -15,7 +15,7 @@
 
    public SerializableTickList(Function<T, ResourceLocation> p_i231603_1_, List<NextTickListEntry<T>> p_i231603_2_, long p_i231603_3_) {
       this(p_i231603_1_, p_i231603_2_.stream().map((p_234854_2_) -> {
-         return new SerializableTickList.TickHolder(p_234854_2_.func_151351_a(), p_234854_2_.field_180282_a, (int)(p_234854_2_.field_235017_b_ - p_i231603_3_), p_234854_2_.field_82754_f);
+         return new SerializableTickList.TickHolder<>(p_234854_2_.func_151351_a(), p_234854_2_.field_180282_a, (int)(p_234854_2_.field_235017_b_ - p_i231603_3_), p_234854_2_.field_82754_f);
       }).collect(Collectors.toList()));
    }
 
