--- a/net/minecraft/world/entity/monster/Zoglin.java
+++ b/net/minecraft/world/entity/monster/Zoglin.java
@@ -89,7 +89,7 @@
     }
 
     private static ActivityData<Zoglin> initIdleActivity() {
-        return ActivityData.create(
+        return ActivityData.<Zoglin>create(
             Activity.IDLE,
             10,
             ImmutableList.of(
@@ -105,7 +105,7 @@
     }
 
     private static ActivityData<Zoglin> initFightActivity() {
-        return ActivityData.create(
+        return ActivityData.<Zoglin>create(
             Activity.FIGHT,
             10,
             ImmutableList.of(
@@ -208,7 +208,7 @@
 
     @Override
     public Brain<Zoglin> getBrain() {
-        return super.getBrain();
+        return (Brain<Zoglin>)super.getBrain();
     }
 
     protected void updateActivity() {
