--- a/net/minecraft/world/entity/projectile/AbstractArrow.java
+++ b/net/minecraft/world/entity/projectile/AbstractArrow.java
@@ -567,9 +567,9 @@
         super.m_5602_(p_36770_);
 
         this.f_36705_ = switch (p_36770_) {
-            case null, default -> this.f_36705_;
             case Player player when this.f_36705_ == AbstractArrow.Pickup.DISALLOWED -> AbstractArrow.Pickup.ALLOWED;
             case OminousItemSpawner ominousitemspawner -> AbstractArrow.Pickup.DISALLOWED;
+            case null, default -> this.f_36705_;
         };
     }
 
