--- a/net/minecraft/commands/arguments/EntityArgument.java
+++ b/net/minecraft/commands/arguments/EntityArgument.java
@@ -111,10 +111,10 @@
    }
 
    public <S> CompletableFuture<Suggestions> listSuggestions(CommandContext<S> p_91482_, SuggestionsBuilder p_91483_) {
-      StringReader stringreader = p_91482_.getSource();
-      if (stringreader instanceof SharedSuggestionProvider) {
-         SharedSuggestionProvider sharedsuggestionprovider = (SharedSuggestionProvider)stringreader;
-         stringreader = new StringReader(p_91483_.getInput());
+      S s = p_91482_.getSource();
+      if (s instanceof SharedSuggestionProvider) {
+         SharedSuggestionProvider sharedsuggestionprovider = (SharedSuggestionProvider)s;
+         StringReader stringreader = new StringReader(p_91483_.getInput());
          stringreader.setCursor(p_91483_.getStart());
          EntitySelectorParser entityselectorparser = new EntitySelectorParser(stringreader, sharedsuggestionprovider.m_6761_(2));
 
